// @fluentui/react-northstar/dist/dts/src/utils/mergeProviderContexts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var styles_1 = require("@fluentui/styles");
var felaRenderer_1 = require("./felaRenderer");
var isBrowser_1 = require("./isBrowser");
var registeredRenderers = new WeakMap();
exports.mergeRenderers = function (current, next, target) {
    if (next) {
        return next;
    }
    // A valid comparisons, default renderer will be used
    if (!isBrowser_1.default() || typeof target === 'undefined') {
        return felaRenderer_1.felaRenderer;
    }
    // SSR logic will be handled by condition above
    // eslint-disable-next-line no-undef
    if (target === document) {
        return felaRenderer_1.felaRenderer;
    }
    if (registeredRenderers.has(target)) {
        return registeredRenderers.get(target);
    }
    var createdRenderer = felaRenderer_1.createRenderer();
    registeredRenderers.set(target, createdRenderer);
    return createdRenderer;
};
exports.mergePerformanceOptions = function (target) {
    var sources = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        sources[_i - 1] = arguments[_i];
    }
    return Object.assign.apply(Object, tslib_1.__spreadArrays([target], sources));
};
exports.mergeBooleanValues = function (target) {
    var sources = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        sources[_i - 1] = arguments[_i];
    }
    return sources.reduce(function (acc, next) {
        return typeof next === 'boolean' ? next : acc;
    }, target);
};
var mergeProviderContexts = function () {
    var contexts = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        contexts[_i] = arguments[_i];
    }
    var emptyContext = {
        theme: {
            siteVariables: {
                fontSizes: {},
            },
            componentVariables: {},
            componentStyles: {},
            fontFaces: [],
            staticStyles: [],
            animations: {},
        },
        rtl: false,
        disableAnimations: false,
        target: isBrowser_1.default() ? document : undefined,
        performance: {
            enableSanitizeCssPlugin: process.env.NODE_ENV !== 'production',
            enableStylesCaching: true,
            enableVariablesCaching: true,
            enableBooleanVariablesCaching: false,
        },
        telemetry: undefined,
        renderer: undefined,
    };
    return contexts.reduce(function (acc, next) {
        if (!next)
            return acc;
        acc.theme = styles_1.mergeThemes(acc.theme, next.theme);
        // Latest RTL value wins
        var mergedRTL = exports.mergeBooleanValues(acc.rtl, next.rtl);
        if (typeof mergedRTL === 'boolean') {
            acc.rtl = mergedRTL;
        }
        // Use provided renderer if it is defined
        acc.target = next.target || acc.target;
        acc.renderer = exports.mergeRenderers(acc.renderer, next.renderer, acc.target);
        // Latest disableAnimations value wins
        var mergedDisableAnimations = exports.mergeBooleanValues(acc.disableAnimations, next.disableAnimations);
        if (typeof mergedDisableAnimations === 'boolean') {
            acc.disableAnimations = mergedDisableAnimations;
        }
        acc.performance = exports.mergePerformanceOptions(acc.performance, next.performance || {});
        acc.telemetry = next.telemetry || acc.telemetry;
        return acc;
    }, emptyContext);
};
exports.default = mergeProviderContexts;
