// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Carousel/carouselNavigationStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _colors = require("../../colors");

var carouselNavigationStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    var iconOnly = p.iconOnly,
        primary = p.primary,
        vertical = p.vertical,
        thumbnails = p.thumbnails;
    var colors = (0, _colors.getColorScheme)(v.colorScheme, null, primary);
    return Object.assign({
      display: 'flex',
      minHeight: (0, _utils.pxToRem)(24),
      margin: 0,
      padding: 0,
      color: v.color,
      backgroundColor: v.backgroundColor || 'inherit',
      listStyleType: 'none',
      justifyContent: 'center'
    }, !vertical && thumbnails && {
      justifyContent: 'start',
      transform: "translateX(".concat((0, _utils.pxToRem)(v.width / 2 - v.thumbnailWidth / 2 - +p.activeIndex * v.thumbnailWidth), ")"),
      transition: 'transform .5s ease'
    }, {}, iconOnly && {
      alignItems: 'center'
    }, {}, vertical && Object.assign({
      flexDirection: 'column',
      backgroundColor: v.verticalBackgroundColor,
      width: 'fit-content',
      padding: "".concat((0, _utils.pxToRem)(8), " 0")
    }, iconOnly && {
      display: 'inline-block',
      width: 'auto'
    }), {}, !iconOnly && !vertical && {
      // primary has hardcoded grey border color
      border: "".concat(v.borderWidth, " solid ").concat(primary ? v.primaryBorderColor : v.borderColor || colors.border),
      borderRadius: (0, _utils.pxToRem)(4)
    });
  }
};
var _default = carouselNavigationStyles;
exports.default = _default;
