// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Tree/treeTitleStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _utils = require("../../../../utils");

var _checkboxIndicatorUrl = _interopRequireDefault(require("../Checkbox/checkboxIndicatorUrl"));

var _checkboxIndicatorIndeterminatedUrl = _interopRequireDefault(require("./checkboxIndicatorIndeterminatedUrl"));

var treeTitleStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    return Object.assign({
      padding: v.padding,
      cursor: 'pointer',
      color: v.color,
      position: 'relative'
    }, p.selectable && {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center'
    }, {}, (0, _getBorderFocusStyles.default)({
      variables: siteVariables
    }));
  },
  selectionIndicator: function selectionIndicator(_ref2) {
    var p = _ref2.props,
        v = _ref2.variables;
    return Object.assign({
      display: 'inline-block',
      float: 'right',
      verticalAlign: 'middle',
      boxShadow: 'unset',
      width: (0, _utils.pxToRem)(16),
      height: (0, _utils.pxToRem)(16),
      borderColor: v.borderColor,
      borderStyle: v.borderStyle,
      borderRadius: v.borderRadius,
      borderWidth: v.borderWidth,
      color: v.indicatorColor,
      margin: v.selectionIndicatorMargin,
      padding: v.padding,
      userSelect: 'none',
      backgroundImage: (0, _checkboxIndicatorUrl.default)(v.indicatorColor, v.background),
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat'
    }, p.selected && {
      borderColor: v.checkedBorderColor,
      backgroundImage: (0, _checkboxIndicatorUrl.default)(v.checkedIndicatorColor, v.checkedBackground)
    }, {}, p.indeterminate && {
      borderColor: v.checkedBorderColor,
      backgroundImage: (0, _checkboxIndicatorIndeterminatedUrl.default)(v.checkedIndicatorColor, v.checkedBackground)
    }, {}, p.disabled && {
      background: v.disabledBackground,
      borderColor: v.disabledBorderColor
    }, {}, p.disabled && p.selected && {
      color: v.disabledCheckedIndicatorColor,
      borderColor: v.disabledBackgroundChecked,
      backgroundImage: (0, _checkboxIndicatorUrl.default)(v.disabledCheckedIndicatorColor, v.disabledBackgroundChecked)
    });
  }
};
var _default = treeTitleStyles;
exports.default = _default;
