// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Tree/treeItemStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _utils = require("../../../../utils");

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _TreeTitle = require("../../../../components/Tree/TreeTitle");

var treeItemStyles = {
  root: function root(_ref) {
    var siteVariables = _ref.theme.siteVariables,
        p = _ref.props;
    return Object.assign({
      listStyleType: 'none',
      padding: "0 0 0 ".concat((0, _utils.pxToRem)(1 + (p.level - 1) * 10))
    }, (0, _getBorderFocusStyles.default)({
      variables: siteVariables
    })[':focus'], {
      ':focus-visible': (0, _defineProperty2.default)({
        outline: 0
      }, "> .".concat(_TreeTitle.treeTitleClassName), Object.assign({
        position: 'relative'
      }, (0, _getBorderFocusStyles.default)({
        variables: siteVariables
      })[':focus-visible']))
    });
  }
};
var _default = treeItemStyles;
exports.default = _default;
