// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Tree/treeTitleVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _default = function _default(siteVars) {
  return {
    color: siteVars.colorScheme.default.foreground,
    padding: "".concat((0, _utils.pxToRem)(1), " 0"),
    borderColor: siteVars.colorScheme.default.foreground1,
    borderStyle: 'solid',
    borderRadius: (0, _utils.pxToRem)(3),
    borderWidth: (0, _utils.pxToRem)(1),
    indicatorColor: 'transparent',
    selectionIndicatorMargin: "0 0 0 0",
    background: 'transparent',
    checkedBackground: siteVars.colorScheme.brand.backgroundActive1,
    checkedBorderColor: siteVars.colorScheme.brand.backgroundActive1,
    checkedIndicatorColor: siteVars.colorScheme.default.background,
    disabledBackground: siteVars.colorScheme.default.background,
    disabledBackgroundChecked: siteVars.colorScheme.default.backgroundDisabled,
    disabledBorderColor: siteVars.colorScheme.default.foregroundDisabled1,
    disabledCheckedIndicatorColor: siteVars.colorScheme.default.foregroundDisabled
  };
};

exports.default = _default;
