// @fluentui/react-northstar/dist/es/components/Card/CardTopControls.js
import { getElementType, useUnhandledProps, useAccessibility, useStyles, useTelemetry } from '@fluentui/react-bindings';
import * as React from 'react'; // @ts-ignore

import { ThemeContext } from 'react-fela';
import { withSafeTypeForAs } from '../../types';
import { commonPropTypes, createShorthandFactory } from '../../utils';
export var cardTopControlsClassName = 'ui-card__topcontrols';

var CardTopControls = function CardTopControls(props) {
  var context = React.useContext(ThemeContext);

  var _useTelemetry = useTelemetry(CardTopControls.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children;
  var ElementType = getElementType(props);
  var unhandledProps = useUnhandledProps(CardTopControls.handledProps, props);
  var getA11yProps = useAccessibility(props.accessibility, {
    debugName: CardTopControls.displayName,
    rtl: context.rtl
  });

  var _useStyles = useStyles(CardTopControls.displayName, {
    className: cardTopControlsClassName,
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardTopControls.displayName = 'CardTopControls';
CardTopControls.propTypes = Object.assign({}, commonPropTypes.createCommon());
CardTopControls.handledProps = Object.keys(CardTopControls.propTypes);
CardTopControls.create = createShorthandFactory({
  Component: CardTopControls
});
/**
 * A CardTopControls is used to render control elements in the top of a Card component.
 */

export default withSafeTypeForAs(CardTopControls);
