// @fluentui/react-northstar/dist/es/components/Card/CardBody.js
import * as React from 'react';
import { withSafeTypeForAs } from '../../types';
import { commonPropTypes, createShorthandFactory } from '../../utils';
import { useTelemetry, useStyles, getElementType, useUnhandledProps, useAccessibility } from '@fluentui/react-bindings';
import * as PropTypes from 'prop-types'; // @ts-ignore

import { ThemeContext } from 'react-fela';
export var cardBodyClassName = 'ui-card__body';

var CardBody = function CardBody(props) {
  var context = React.useContext(ThemeContext);

  var _useTelemetry = useTelemetry(CardBody.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children,
      fitted = props.fitted;
  var ElementType = getElementType(props);
  var unhandledProps = useUnhandledProps(CardBody.handledProps, props);
  var getA11yProps = useAccessibility(props.accessibility, {
    debugName: CardBody.displayName,
    rtl: context.rtl
  });

  var _useStyles = useStyles(CardBody.displayName, {
    className: cardBodyClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        fitted: fitted
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardBody.displayName = 'CardBody';
CardBody.propTypes = Object.assign({}, commonPropTypes.createCommon(), {
  fitted: PropTypes.bool
});
CardBody.handledProps = Object.keys(CardBody.propTypes);
CardBody.create = createShorthandFactory({
  Component: CardBody
});
/**
 * A CardBody is used to display data in Card body.
 */

export default withSafeTypeForAs(CardBody);
