// @fluentui/react-northstar/dist/commonjs/themes/teams/components/TextArea/textAreaVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _default = function _default(siteVars) {
  return {
    margin: '0',
    padding: "".concat((0, _utils.pxToRem)(7), " ").concat((0, _utils.pxToRem)(12)),
    borderColor: 'transparent',
    borderRadius: "".concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(2), " ").concat((0, _utils.pxToRem)(2)),
    borderWidth: "0 0 ".concat((0, _utils.pxToRem)(2), " 0"),
    backgroundColor: siteVars.colorScheme.default.background2,
    invertedBackgroundColor: siteVars.colorScheme.default.background,
    placeholderColor: siteVars.colorScheme.default.foreground1,
    disabledBackgroundColor: siteVars.colorScheme.default.backgroundColorDisabled,
    disabledColor: siteVars.colorScheme.default.colorDisabled,
    fontColor: siteVars.colorScheme.default.foreground,
    fontSize: siteVars.fontSizes.medium,
    borderColorFocus: "transparent transparent ".concat(siteVars.colorScheme.brand.borderFocus1, " transparent"),
    height: 'auto'
  };
};

exports.default = _default;
