// @fluentui/styles/dist/commonjs/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  callable: true,
  createTheme: true,
  isDebugEnabled: true,
  deepmerge: true,
  mergeThemes: true,
  emptyTheme: true,
  mergeComponentVariables: true,
  mergeComponentStyles: true,
  mergeFontFaces: true,
  mergeStaticStyles: true,
  mergeStyles: true,
  mergeThemeStyles: true,
  mergeThemeVariables: true,
  mergeSiteVariables: true,
  objectKeyToValues: true,
  withDebugId: true
};
Object.defineProperty(exports, "callable", {
  enumerable: true,
  get: function get() {
    return _callable.default;
  }
});
Object.defineProperty(exports, "createTheme", {
  enumerable: true,
  get: function get() {
    return _createTheme.default;
  }
});
Object.defineProperty(exports, "isDebugEnabled", {
  enumerable: true,
  get: function get() {
    return _debugEnabled.isEnabled;
  }
});
Object.defineProperty(exports, "deepmerge", {
  enumerable: true,
  get: function get() {
    return _deepmerge.default;
  }
});
Object.defineProperty(exports, "mergeThemes", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.default;
  }
});
Object.defineProperty(exports, "emptyTheme", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.emptyTheme;
  }
});
Object.defineProperty(exports, "mergeComponentVariables", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeComponentVariables;
  }
});
Object.defineProperty(exports, "mergeComponentStyles", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeComponentStyles;
  }
});
Object.defineProperty(exports, "mergeFontFaces", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeFontFaces;
  }
});
Object.defineProperty(exports, "mergeStaticStyles", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeStaticStyles;
  }
});
Object.defineProperty(exports, "mergeStyles", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeStyles;
  }
});
Object.defineProperty(exports, "mergeThemeStyles", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeThemeStyles;
  }
});
Object.defineProperty(exports, "mergeThemeVariables", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeThemeVariables;
  }
});
Object.defineProperty(exports, "mergeSiteVariables", {
  enumerable: true,
  get: function get() {
    return _mergeThemes.mergeSiteVariables;
  }
});
Object.defineProperty(exports, "objectKeyToValues", {
  enumerable: true,
  get: function get() {
    return _objectKeysToValues.default;
  }
});
Object.defineProperty(exports, "withDebugId", {
  enumerable: true,
  get: function get() {
    return _withDebugId.default;
  }
});

var _callable = _interopRequireDefault(require("./callable"));

var _createTheme = _interopRequireDefault(require("./createTheme"));

var _debugEnabled = require("./debugEnabled");

var _deepmerge = _interopRequireDefault(require("./deepmerge"));

var _mergeThemes = _interopRequireWildcard(require("./mergeThemes"));

var _objectKeysToValues = _interopRequireDefault(require("./objectKeysToValues"));

var _withDebugId = _interopRequireDefault(require("./withDebugId"));

var _types = require("./types");

Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _types[key];
    }
  });
});
