// @fluentui/react-northstar/dist/commonjs/components/Segment/Segment.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.segmentClassName = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _utils = require("../../utils");

var _types = require("../../types");

var _Box = _interopRequireDefault(require("../Box/Box"));

var segmentClassName = 'ui-segment';
exports.segmentClassName = segmentClassName;

var Segment =
/*#__PURE__*/
function (_UIComponent) {
  (0, _inherits2.default)(Segment, _UIComponent);

  function Segment() {
    (0, _classCallCheck2.default)(this, Segment);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Segment).apply(this, arguments));
  }

  (0, _createClass2.default)(Segment, [{
    key: "renderComponent",
    value: function renderComponent(_ref) {
      var accessibility = _ref.accessibility,
          ElementType = _ref.ElementType,
          classes = _ref.classes,
          unhandledProps = _ref.unhandledProps;
      var _this$props = this.props,
          children = _this$props.children,
          content = _this$props.content;
      return React.createElement(ElementType, (0, _extends2.default)({}, _utils.rtlTextContainer.getAttributes({
        forElements: [children]
      }), accessibility.attributes.root, unhandledProps, {
        className: classes.root
      }), (0, _utils.childrenExist)(children) ? children : _Box.default.create(content));
    }
  }]);
  return Segment;
}(_utils.UIComponent);
/**
 * A Segment visually groups related content.
 */


(0, _defineProperty2.default)(Segment, "deprecated_className", segmentClassName);
(0, _defineProperty2.default)(Segment, "displayName", 'Segment');
(0, _defineProperty2.default)(Segment, "propTypes", Object.assign({}, _utils.commonPropTypes.createCommon({
  content: 'shorthand',
  color: true
}), {
  disabled: PropTypes.bool,
  inverted: PropTypes.bool,
  rtlAttributes: PropTypes.func
}));
(0, _defineProperty2.default)(Segment, "defaultProps", {
  as: 'div'
});

var _default = (0, _types.withSafeTypeForAs)(Segment);

exports.default = _default;
