// @fluentui/react-bindings/dist/dts/src/styles/getStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var styles_1 = require("@fluentui/styles");
var _ = require("lodash");
var resolveVariables_1 = require("./resolveVariables");
var resolveStyles_1 = require("./resolveStyles");
var getStyles = function (options) {
    //
    // To compute styles we are going through three stages:
    // - resolve variables (siteVariables => componentVariables + props.variables)
    // - resolve styles (with resolvedVariables & props.styles & props.design)
    // - compute classes (with resolvedStyles)
    // - conditionally add sources for evaluating debug information to component
    var resolvedVariables = resolveVariables_1.default(options.displayNames, options.theme, options.props.variables, options.performance.enableVariablesCaching);
    var _a = resolveStyles_1.default(options, resolvedVariables), classes = _a.classes, resolvedStyles = _a.resolvedStyles, resolvedStylesDebug = _a.resolvedStylesDebug;
    // conditionally add sources for evaluating debug information to component
    if (process.env.NODE_ENV !== 'production' && styles_1.isDebugEnabled) {
        options.saveDebug({
            componentName: options.displayNames.join(':'),
            componentVariables: _.filter(resolvedVariables._debug, function (variables) { return !_.isEmpty(variables.resolved); }),
            componentStyles: resolvedStylesDebug,
            siteVariables: _.filter(options.theme.siteVariables._debug, function (siteVars) {
                if (_.isEmpty(siteVars) || _.isEmpty(siteVars.resolved)) {
                    return false;
                }
                var keys = Object.keys(siteVars.resolved);
                if (keys.length === 1 && keys.pop() === 'fontSizes' && _.isEmpty(siteVars.resolved['fontSizes'])) {
                    return false;
                }
                return true;
            }),
        });
    }
    return {
        classes: classes,
        variables: resolvedVariables,
        styles: resolvedStyles,
        theme: options.theme,
    };
};
exports.default = getStyles;
