// @fluentui/accessibility/dist/commonjs/behaviors/Toolbar/toolbarMenuItemCheckboxBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

/**
 * @specification
 * Adds attribute 'aria-checked=true' based on the property 'active'.
 * Adds attribute 'aria-disabled=true' based on the property 'disabled'.
 * Adds role='menuitemcheckbox'.
 * Adds role 'presentation' to 'wrapper' slot.
 * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
 */
var toolbarMenuItemCheckboxBehavior = function toolbarMenuItemCheckboxBehavior(props) {
  return {
    attributes: {
      wrapper: {
        role: 'presentation'
      },
      root: {
        'aria-checked': props.active,
        'aria-disabled': props.disabled,
        role: 'menuitemcheckbox'
      }
    },
    keyActions: {
      root: {
        performClick: {
          keyCombinations: [{
            keyCode: keyboardKey.Enter
          }, {
            keyCode: keyboardKey.Spacebar
          }]
        }
      }
    }
  };
};

var _default = toolbarMenuItemCheckboxBehavior;
exports.default = _default;
