// @fluentui/react-northstar/dist/commonjs/themes/teams/siteVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "colors", {
  enumerable: true,
  get: function get() {
    return _colors.colors;
  }
});
Object.defineProperty(exports, "contextualColors", {
  enumerable: true,
  get: function get() {
    return _colors.contextualColors;
  }
});
Object.defineProperty(exports, "naturalColors", {
  enumerable: true,
  get: function get() {
    return _colors.naturalColors;
  }
});
Object.defineProperty(exports, "primitiveColors", {
  enumerable: true,
  get: function get() {
    return _colors.primitiveColors;
  }
});
Object.defineProperty(exports, "colorScheme", {
  enumerable: true,
  get: function get() {
    return _colors.colorScheme;
  }
});
Object.defineProperty(exports, "transparentColors", {
  enumerable: true,
  get: function get() {
    return _colors.transparentColors;
  }
});
Object.defineProperty(exports, "categoryColors", {
  enumerable: true,
  get: function get() {
    return _categoryColors.categoryColors;
  }
});
Object.defineProperty(exports, "categoryColorScheme", {
  enumerable: true,
  get: function get() {
    return _categoryColors.categoryColorScheme;
  }
});
exports.bodyLineHeight = exports.bodyColor = exports.bodyBackground = exports.bodyFontSize = exports.bodyFontFamily = exports.bodyMargin = exports.bodyPadding = exports.zIndexes = exports.lineHeightLarger = exports.lineHeightLarge = exports.lineHeightMedium = exports.lineHeightSmall = exports.lineHeightSmaller = exports.lineHeightDefault = exports.fontWeightBold = exports.fontWeightSemibold = exports.fontWeightRegular = exports.fontWeightSemilight = exports.fontWeightLight = exports.fontSizes = exports.shadowLevel1Dark = exports.shadowLevel4 = exports.shadowLevel3 = exports.shadowLevel2 = exports.shadowLevel1 = exports.focusOuterBorderColor = exports.focusInnerBorderColor = exports.borderRadius = exports.borderWidth = void 0;

var _utils = require("../../utils");

var _colors = require("./colors");

var _categoryColors = require("./categoryColors");

//
// COLORS
//
//
// BORDER STYLES
//
var borderWidth = '1px';
exports.borderWidth = borderWidth;
var borderRadius = '2px';
exports.borderRadius = borderRadius;
var focusInnerBorderColor = _colors.colors.white;
exports.focusInnerBorderColor = focusInnerBorderColor;
var focusOuterBorderColor = _colors.colors.black; //
// SHADOW LEVELS
//

exports.focusOuterBorderColor = focusOuterBorderColor;
var shadowLevel1 = '0 .2rem .4rem -.075rem rgba(0, 0, 0, .1)';
exports.shadowLevel1 = shadowLevel1;
var shadowLevel2 = '0 .4rem .7rem -.1rem rgba(0, 0, 0, .1)';
exports.shadowLevel2 = shadowLevel2;
var shadowLevel3 = '0 .8rem 1rem -.2rem rgba(0, 0, 0, .1)';
exports.shadowLevel3 = shadowLevel3;
var shadowLevel4 = '0 1.6rem 1.8rem -.4rem rgba(0, 0, 0, .1)';
exports.shadowLevel4 = shadowLevel4;
var shadowLevel1Dark = '0 .2rem .4rem -.075rem rgba(0, 0, 0, .25)'; //
// FONT SIZES
//

exports.shadowLevel1Dark = shadowLevel1Dark;
var fontSizes = {
  smaller: (0, _utils.pxToRem)(10),
  small: (0, _utils.pxToRem)(12),
  medium: (0, _utils.pxToRem)(14),
  large: (0, _utils.pxToRem)(18),
  larger: (0, _utils.pxToRem)(24)
}; //
// FONT WEIGHTS
//

exports.fontSizes = fontSizes;
var fontWeightLight = 200;
exports.fontWeightLight = fontWeightLight;
var fontWeightSemilight = 300;
exports.fontWeightSemilight = fontWeightSemilight;
var fontWeightRegular = 400;
exports.fontWeightRegular = fontWeightRegular;
var fontWeightSemibold = 600;
exports.fontWeightSemibold = fontWeightSemibold;
var fontWeightBold = 700; //
// LINE HEIGHTS
//

exports.fontWeightBold = fontWeightBold;
var lineHeightDefault = 1;
exports.lineHeightDefault = lineHeightDefault;
var lineHeightSmaller = 1.2;
exports.lineHeightSmaller = lineHeightSmaller;
var lineHeightSmall = 1.3333;
exports.lineHeightSmall = lineHeightSmall;
var lineHeightMedium = 1.4286;
exports.lineHeightMedium = lineHeightMedium;
var lineHeightLarge = 1.3333;
exports.lineHeightLarge = lineHeightLarge;
var lineHeightLarger = 1.3333; //
// Z-INDEX
// Used to maintain proper stack order of components
//

exports.lineHeightLarger = lineHeightLarger;
var zIndexes = {
  background: 0,
  // Default value
  foreground: 1,
  // Put a component in front
  menuItem: 2,
  // Currently used only for menu item beak element
  overlay: 1000,
  // Dialog/popup/menu overlays
  overlayPriority: 1001,
  // for nested overlays, like tooltip in dialog.
  debug: 999999999 // for debug purposes

}; //
// SEMANTIC ASSIGNMENTS
//

exports.zIndexes = zIndexes;
var bodyPadding = 0;
exports.bodyPadding = bodyPadding;
var bodyMargin = 0;
exports.bodyMargin = bodyMargin;
var bodyFontFamily = '"Segoe UI", "Helvetica Neue", "Apple Color Emoji", "Segoe UI Emoji", Helvetica, Arial, sans-serif';
exports.bodyFontFamily = bodyFontFamily;
var bodyFontSize = fontSizes.medium;
exports.bodyFontSize = bodyFontSize;
var bodyBackground = _colors.colors.white;
exports.bodyBackground = bodyBackground;
var bodyColor = _colors.colors.grey[750];
exports.bodyColor = bodyColor;
var bodyLineHeight = lineHeightMedium;
exports.bodyLineHeight = bodyLineHeight;
