// @fluentui/react-bindings/dist/dts/test/styles/resolveStyles-test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var _ = require("lodash");
var styles_1 = require("@fluentui/styles");
var resolveStyles_1 = require("../../src/styles/resolveStyles");
var testComponentStyles = {
    root: function (_a) {
        var v = _a.variables, rtl = _a.rtl;
        return ({
            color: v.color,
            content: "\"rtl:" + rtl.toString() + "\"",
        });
    },
};
var resolvedVariables = {
    color: 'red',
};
var defaultPerformanceOptions = {
    enableSanitizeCssPlugin: true,
    enableStylesCaching: true,
    enableVariablesCaching: true,
    enableBooleanVariablesCaching: false,
};
var resolveStylesOptions = function (options) {
    var _a;
    var _b = options || {}, componentStyles = _b.componentStyles, _c = _b.displayNames, displayNames = _c === void 0 ? ['Test'] : _c, performance = _b.performance, _d = _b.props, props = _d === void 0 ? {} : _d, _e = _b.rtl, rtl = _e === void 0 ? false : _e;
    var theme = tslib_1.__assign(tslib_1.__assign({}, styles_1.emptyTheme), { componentStyles: tslib_1.__assign((_a = {}, _a[displayNames[0]] = testComponentStyles, _a), componentStyles) });
    return {
        theme: theme,
        displayNames: displayNames,
        props: props,
        rtl: rtl,
        disableAnimations: false,
        renderer: {
            renderRule: function () { return ''; },
        },
        performance: tslib_1.__assign(tslib_1.__assign({}, defaultPerformanceOptions), performance),
        saveDebug: function () { },
    };
};
describe('resolveStyles', function () {
    test('resolves styles', function () {
        var resolvedStyles = resolveStyles_1.default(resolveStylesOptions(), resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject({ color: 'red' });
    });
    test('caches resolved styles', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var resolvedStyles = resolveStyles_1.default(resolveStylesOptions(), resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject({ color: 'red' });
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
        expect(resolvedStyles.root).toMatchObject({ color: 'red' });
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
    });
    test('does not render classes if not fetched', function () {
        var renderStyles = jest.fn();
        var resolvedStyles = resolveStyles_1.default(resolveStylesOptions(), resolvedVariables, renderStyles).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject({ color: 'red' });
        expect(renderStyles).not.toBeCalled();
    });
    test('renders classes when slot classes getter is accessed', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var classes = resolveStyles_1.default(resolveStylesOptions(), resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
    });
    test('caches rendered classes', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var classes = resolveStyles_1.default(resolveStylesOptions(), resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledTimes(1);
    });
    test('caches resolved styles for no props', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var options = resolveStylesOptions();
        var resolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        var secondResolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
        expect(secondResolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
    });
    test('caches classes for no props', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var options = resolveStylesOptions({ displayNames: ['Test1'] });
        var classes = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        var secondClasses = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
        expect(secondClasses['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledTimes(1);
    });
    test('caches resolved styles for the same props', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var options = resolveStylesOptions({
            displayNames: ['Test2'],
            props: { primary: true },
        });
        var resolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        var secondResolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
        expect(secondResolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
    });
    test('caches classes for the same props', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var options = resolveStylesOptions({
            displayNames: ['Test3'],
            props: { primary: true },
        });
        var classes = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        var secondClasses = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
        expect(secondClasses['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledTimes(1);
    });
    test('handles multiple displayNames', function () {
        var options = resolveStylesOptions({
            displayNames: ['Test', 'Other'],
            componentStyles: {
                Other: { root: function () { return ({ background: 'pink' }); } },
            },
        });
        expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root', expect.objectContaining({
            color: 'red',
            background: 'pink',
        }));
    });
    test('considers props when caching resolved styles', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var options = resolveStylesOptions({
            displayNames: ['Test4'],
            props: { primary: true },
        });
        var resolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        var secondResolvedStyles = resolveStyles_1.default(tslib_1.__assign(tslib_1.__assign({}, options), { props: { primary: false } }), resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
        expect(secondResolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(2);
    });
    test('considers props when caching classes', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var options = resolveStylesOptions({
            displayNames: ['Test5'],
            props: { primary: true },
        });
        var classes = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        options.props = { primary: false };
        var secondClasses = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
        expect(secondClasses['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledTimes(2);
    });
    test('does not cache styles if caching is disabled', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var options = resolveStylesOptions({
            performance: { enableStylesCaching: false },
        });
        var resolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        var secondResolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
        expect(resolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(secondResolvedStyles.root).toMatchObject(expect.objectContaining({ color: 'red' }));
        expect(testComponentStyles.root).toHaveBeenCalledTimes(2);
    });
    test('does not cache classes if caching is disabled', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var options = resolveStylesOptions({
            performance: { enableStylesCaching: false },
        });
        var classes = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        var secondClasses = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
        expect(classes['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledWith(expect.objectContaining({ color: 'red' }));
        expect(secondClasses['root']).toBeDefined();
        expect(renderStyles).toHaveBeenCalledTimes(2);
    });
    test('does not cache styles if there are inline overrides', function () {
        spyOn(testComponentStyles, 'root').and.callThrough();
        var propsInlineOverrides = [
            { styles: { fontSize: '10px' } },
            { design: { left: '10px' } },
            { variables: { backgroundColor: 'yellow' } },
        ];
        var propsInlineOverridesResolvedStyles = [
            { color: 'red', fontSize: '10px' },
            { color: 'red', left: '10px' },
            { color: 'red' },
        ];
        var propsInlineOverridesSize = propsInlineOverrides.length;
        _.forEach(propsInlineOverrides, function (props, idx) {
            var options = resolveStylesOptions({
                props: props,
                performance: { enableStylesCaching: false },
            });
            var resolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
            var secondResolvedStyles = resolveStyles_1.default(options, resolvedVariables).resolvedStyles;
            expect(resolvedStyles.root).toMatchObject(propsInlineOverridesResolvedStyles[idx]);
            expect(secondResolvedStyles.root).toMatchObject(propsInlineOverridesResolvedStyles[idx]);
            resolveStyles_1.default(options, resolvedVariables);
        });
        expect(testComponentStyles.root).toHaveBeenCalledTimes(propsInlineOverridesSize * 2);
    });
    test('does not cache classes if there are inline overrides', function () {
        var renderStyles = jest.fn().mockReturnValue('a');
        var propsInlineOverrides = [
            { styles: { fontSize: '10px' } },
            { design: { left: '10px' } },
            { variables: { backgroundColor: 'yellow' } },
        ];
        var propsInlineOverridesSize = propsInlineOverrides.length;
        _.forEach(propsInlineOverrides, function (props) {
            var options = resolveStylesOptions({
                props: props,
                performance: { enableStylesCaching: false },
            });
            var classes = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
            var secondClasses = resolveStyles_1.default(options, resolvedVariables, renderStyles).classes;
            expect(classes['root']).toBeDefined();
            expect(secondClasses['root']).toBeDefined();
        });
        expect(renderStyles).toHaveBeenCalledTimes(propsInlineOverridesSize * 2);
    });
    test('computes new styles when "rtl" changes', function () {
        var renderStyles = jest.fn().mockImplementation(function (style) { return style.content; });
        var ltrOptions = resolveStylesOptions({ rtl: false });
        var rtlOptions = resolveStylesOptions({ rtl: true });
        var ltrStyles = resolveStyles_1.default(ltrOptions, resolvedVariables, renderStyles);
        var rtlStyles = resolveStyles_1.default(rtlOptions, resolvedVariables, renderStyles);
        expect(ltrStyles).toHaveProperty('resolvedStyles.root.content', expect.stringMatching(/rtl:false/));
        expect(ltrStyles).toHaveProperty('classes.root', expect.stringMatching(/rtl:false/));
        expect(renderStyles).toHaveBeenCalledTimes(1);
        expect(rtlStyles).toHaveProperty('resolvedStyles.root.content', expect.stringMatching(/rtl:true/));
        expect(rtlStyles).toHaveProperty('classes.root', expect.stringMatching(/rtl:true/));
        expect(renderStyles).toHaveBeenCalledTimes(2);
    });
    describe('enableBooleanVariablesCaching', function () {
        test('avoids "classes" computation when enabled', function () {
            var renderStyles = jest.fn().mockReturnValue('a');
            var options = resolveStylesOptions({
                props: { variables: { isFoo: true, isBar: null, isBaz: undefined } },
                performance: { enableBooleanVariablesCaching: true },
            });
            expect(resolveStyles_1.default(options, resolvedVariables, renderStyles)).toHaveProperty('classes.root', 'a');
            expect(resolveStyles_1.default(options, resolvedVariables, renderStyles)).toHaveProperty('classes.root', 'a');
            expect(renderStyles).toHaveBeenCalledTimes(1);
        });
        test('forces "classes" computation when disabled', function () {
            var renderStyles = jest.fn().mockReturnValue('a');
            var options = resolveStylesOptions({
                props: { variables: { isFoo: true, isBar: null, isBaz: undefined } },
                performance: { enableBooleanVariablesCaching: false },
            });
            expect(resolveStyles_1.default(options, resolvedVariables, renderStyles)).toHaveProperty('classes.root', 'a');
            expect(resolveStyles_1.default(options, resolvedVariables, renderStyles)).toHaveProperty('classes.root', 'a');
            expect(renderStyles).toHaveBeenCalledTimes(2);
        });
        test('avoids "styles" computation when enabled', function () {
            spyOn(testComponentStyles, 'root').and.callThrough();
            var options = resolveStylesOptions({
                props: { variables: { isFoo: true, isBar: null, isBaz: undefined } },
                performance: { enableBooleanVariablesCaching: true },
            });
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(testComponentStyles.root).toHaveBeenCalledTimes(1);
        });
        test('requires "enableStylesCaching" to be enabled', function () {
            var options = resolveStylesOptions({
                performance: { enableStylesCaching: false, enableBooleanVariablesCaching: true },
            });
            expect(function () { return resolveStyles_1.default(options, resolvedVariables); }).toThrowError(/Please check your "performance" settings on "Provider"/);
        });
        test('when enabled only "variables" as plain objects can be cached', function () {
            spyOn(testComponentStyles, 'root').and.callThrough();
            var options = resolveStylesOptions({
                props: { variables: function () { } },
                performance: { enableBooleanVariablesCaching: true },
            });
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(testComponentStyles.root).toHaveBeenCalledTimes(2);
        });
        test('when enabled only "variables" as boolean or nil properties can be cached', function () {
            spyOn(testComponentStyles, 'root').and.callThrough();
            var options = resolveStylesOptions({
                props: { variables: { foo: 'bar' } },
                performance: { enableBooleanVariablesCaching: true },
            });
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(resolveStyles_1.default(options, resolvedVariables)).toHaveProperty('resolvedStyles.root');
            expect(testComponentStyles.root).toHaveBeenCalledTimes(2);
        });
    });
});
