// @fluentui/react-northstar/dist/es/themes/teams/components/Input/inputVariables.js
import { pxToRem } from '../../../../utils';
export default (function (siteVars) {
  return {
    colorDisabled: siteVars.colorScheme.brand.foregroundDisabled,
    iconPosition: 'absolute',
    iconRight: pxToRem(10),
    iconLeft: pxToRem(6),
    inputPaddingWithIconAtStart: "".concat(pxToRem(5), " ").concat(pxToRem(12), " ").concat(pxToRem(5), " ").concat(pxToRem(24)),
    inputPaddingWithIconAtEnd: "".concat(pxToRem(5), " ").concat(pxToRem(24), " ").concat(pxToRem(5), " ").concat(pxToRem(12)),
    inputPadding: "".concat(pxToRem(5), " ").concat(pxToRem(12)),
    borderColor: 'transparent',
    borderRadius: "".concat(pxToRem(3), " ").concat(pxToRem(3), " ").concat(pxToRem(2), " ").concat(pxToRem(2)),
    borderWidth: "0 0 ".concat(pxToRem(2), " 0"),
    backgroundColor: siteVars.colorScheme.default.background2,
    backgroundColorInverted: siteVars.colorScheme.default.background,
    fontColor: siteVars.colorScheme.default.foreground,
    fontSize: siteVars.fontSizes.medium,
    iconColor: siteVars.colorScheme.default.foreground,
    inputFocusBorderColor: "transparent transparent ".concat(siteVars.colorScheme.brand.borderFocus1, " transparent"),
    inputFocusBorderRadius: "".concat(pxToRem(3), " ").concat(pxToRem(3), " ").concat(pxToRem(2), " ").concat(pxToRem(2)),
    placeholderColor: siteVars.colorScheme.default.foreground1
  };
});
