// @fluentui/react-northstar/dist/commonjs/themes/teams-dark/componentVariables.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Attachment", {
  enumerable: true,
  get: function get() {
    return _attachmentVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentAction", {
  enumerable: true,
  get: function get() {
    return _attachmentActionVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentBody", {
  enumerable: true,
  get: function get() {
    return _attachmentBodyVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentDescription", {
  enumerable: true,
  get: function get() {
    return _attachmentDescriptionVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentHeader", {
  enumerable: true,
  get: function get() {
    return _attachmentHeaderVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentIcon", {
  enumerable: true,
  get: function get() {
    return _attachmentIconVariables.default;
  }
});
Object.defineProperty(exports, "Avatar", {
  enumerable: true,
  get: function get() {
    return _avatarVariables.default;
  }
});
Object.defineProperty(exports, "Chat", {
  enumerable: true,
  get: function get() {
    return _chatVariables.default;
  }
});
Object.defineProperty(exports, "ChatMessage", {
  enumerable: true,
  get: function get() {
    return _chatMessageVariables.default;
  }
});
Object.defineProperty(exports, "Dialog", {
  enumerable: true,
  get: function get() {
    return _dialogVariables.default;
  }
});
Object.defineProperty(exports, "Divider", {
  enumerable: true,
  get: function get() {
    return _dividerVariables.default;
  }
});
Object.defineProperty(exports, "Header", {
  enumerable: true,
  get: function get() {
    return _headerVariables.default;
  }
});
Object.defineProperty(exports, "HeaderDescription", {
  enumerable: true,
  get: function get() {
    return _headerDescriptionVariables.default;
  }
});
Object.defineProperty(exports, "ListItem", {
  enumerable: true,
  get: function get() {
    return _listItemVariables.default;
  }
});
Object.defineProperty(exports, "RadioGroupItem", {
  enumerable: true,
  get: function get() {
    return _radioGroupItemVariables.default;
  }
});
Object.defineProperty(exports, "Segment", {
  enumerable: true,
  get: function get() {
    return _segmentVariables.default;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _textVariables.default;
  }
});
Object.defineProperty(exports, "HierarchicalTreeTitle", {
  enumerable: true,
  get: function get() {
    return _hierarchicalTreeTitleVariables.default;
  }
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function get() {
    return _menuVariables.default;
  }
});
Object.defineProperty(exports, "SvgIcon", {
  enumerable: true,
  get: function get() {
    return _svgIconVariables.default;
  }
});
Object.defineProperty(exports, "Reaction", {
  enumerable: true,
  get: function get() {
    return _reactionVariables.default;
  }
});
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function get() {
    return _alertVariables.default;
  }
});
Object.defineProperty(exports, "AlertDismissAction", {
  enumerable: true,
  get: function get() {
    return _alertDismissActionVariables.default;
  }
});
Object.defineProperty(exports, "Provider", {
  enumerable: true,
  get: function get() {
    return _providerVariables.default;
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function get() {
    return _labelVariables.default;
  }
});

var _attachmentVariables = _interopRequireDefault(require("./components/Attachment/attachmentVariables"));

var _attachmentActionVariables = _interopRequireDefault(require("./components/Attachment/attachmentActionVariables"));

var _attachmentBodyVariables = _interopRequireDefault(require("./components/Attachment/attachmentBodyVariables"));

var _attachmentDescriptionVariables = _interopRequireDefault(require("./components/Attachment/attachmentDescriptionVariables"));

var _attachmentHeaderVariables = _interopRequireDefault(require("./components/Attachment/attachmentHeaderVariables"));

var _attachmentIconVariables = _interopRequireDefault(require("./components/Attachment/attachmentIconVariables"));

var _avatarVariables = _interopRequireDefault(require("./components/Avatar/avatarVariables"));

var _chatVariables = _interopRequireDefault(require("./components/Chat/chatVariables"));

var _chatMessageVariables = _interopRequireDefault(require("./components/Chat/chatMessageVariables"));

var _dialogVariables = _interopRequireDefault(require("./components/Dialog/dialogVariables"));

var _dividerVariables = _interopRequireDefault(require("./components/Divider/dividerVariables"));

var _headerVariables = _interopRequireDefault(require("./components/Header/headerVariables"));

var _headerDescriptionVariables = _interopRequireDefault(require("./components/Header/headerDescriptionVariables"));

var _listItemVariables = _interopRequireDefault(require("./components/List/listItemVariables"));

var _radioGroupItemVariables = _interopRequireDefault(require("./components/RadioGroup/radioGroupItemVariables"));

var _segmentVariables = _interopRequireDefault(require("./components/Segment/segmentVariables"));

var _textVariables = _interopRequireDefault(require("./components/Text/textVariables"));

var _hierarchicalTreeTitleVariables = _interopRequireDefault(require("./components/HierarchicalTree/hierarchicalTreeTitleVariables"));

var _menuVariables = _interopRequireDefault(require("./components/Menu/menuVariables"));

var _svgIconVariables = _interopRequireDefault(require("./components/SvgIcon/svgIconVariables"));

var _reactionVariables = _interopRequireDefault(require("./components/Reaction/reactionVariables"));

var _alertVariables = _interopRequireDefault(require("./components/Alert/alertVariables"));

var _alertDismissActionVariables = _interopRequireDefault(require("./components/Alert/alertDismissActionVariables"));

var _providerVariables = _interopRequireDefault(require("./components/Provider/providerVariables"));

var _labelVariables = _interopRequireDefault(require("./components/Label/labelVariables"));
