// @fluentui/react-northstar/dist/dts/src/components/Header/HeaderDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var utils_1 = require("../../utils");
var types_1 = require("../../types");
exports.headerDescriptionClassName = 'ui-header__description';
var HeaderDescription = /** @class */ (function (_super) {
    tslib_1.__extends(HeaderDescription, _super);
    function HeaderDescription() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    HeaderDescription.prototype.renderComponent = function (_a) {
        var accessibility = _a.accessibility, ElementType = _a.ElementType, classes = _a.classes, unhandledProps = _a.unhandledProps;
        var _b = this.props, children = _b.children, content = _b.content;
        return (React.createElement(ElementType, tslib_1.__assign({}, utils_1.rtlTextContainer.getAttributes({ forElements: [children, content] }), accessibility.attributes.root, unhandledProps, { className: classes.root }), utils_1.childrenExist(children) ? children : content));
    };
    HeaderDescription.deprecated_className = exports.headerDescriptionClassName;
    HeaderDescription.displayName = 'HeaderDescription';
    HeaderDescription.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({ color: true }));
    HeaderDescription.defaultProps = {
        as: 'p',
    };
    return HeaderDescription;
}(utils_1.UIComponent));
HeaderDescription.create = utils_1.createShorthandFactory({
    Component: HeaderDescription,
    mappedProp: 'content',
});
/**
 * A HeaderDescription provides more detailed information about the Header.
 */
exports.default = types_1.withSafeTypeForAs(HeaderDescription);
