// @fluentui/react-northstar/dist/dts/src/components/Video/Video.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_component_ref_1 = require("@fluentui/react-component-ref");
var PropTypes = require("prop-types");
var React = require("react");
var accessibility_1 = require("@fluentui/accessibility");
var utils_1 = require("../../utils");
var types_1 = require("../../types");
var react_bindings_1 = require("@fluentui/react-bindings");
// @ts-ignore
var react_fela_1 = require("react-fela");
exports.videoClassName = 'ui-video';
exports.Video = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(exports.Video.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var controls = props.controls, autoPlay = props.autoPlay, loop = props.loop, poster = props.poster, src = props.src, muted = props.muted, variables = props.variables, className = props.className, design = props.design, styles = props.styles;
    var ElementType = react_bindings_1.getElementType(props);
    var videoRef = React.createRef();
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.Video.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: exports.Video.displayName,
    });
    React.useEffect(function () {
        // React doesn't guaranty that props will be set:
        // https://github.com/facebook/react/issues/10389
        if (videoRef.current) {
            videoRef.current.muted = !!muted;
        }
    }, [muted]);
    var classes = react_bindings_1.useStyles(exports.Video.displayName, {
        className: exports.videoClassName,
        mapPropsToInlineStyles: function () { return ({ className: className, variables: variables, design: design, styles: styles }); },
    }).classes;
    var element = (React.createElement(react_component_ref_1.Ref, { innerRef: videoRef },
        React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root, autoPlay: autoPlay,
            controls: controls,
            loop: loop,
            poster: poster,
            src: src }, unhandledProps))))));
    setEnd();
    return element;
};
exports.Video.create = utils_1.createShorthandFactory({ Component: exports.Video, mappedProp: 'src' });
exports.Video.displayName = 'Video';
exports.Video.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({
    children: false,
    content: false,
})), { autoPlay: PropTypes.bool, controls: PropTypes.bool, loop: PropTypes.bool, muted: PropTypes.bool, poster: PropTypes.string, src: PropTypes.string });
exports.Video.defaultProps = {
    as: 'video',
    accessibility: accessibility_1.videoBehavior,
    controls: true,
    autoPlay: false,
    muted: false,
};
exports.Video.handledProps = Object.keys(exports.Video.propTypes);
/**
 * A Video provides ability to embed video content.
 */
exports.default = types_1.withSafeTypeForAs(exports.Video);
