// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Button/buttonStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _isNil2 = _interopRequireDefault(require("lodash/isNil"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../../../utils");

var _Loader = require("../../../../components/Loader/Loader");

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _getIconFillOrOutlineStyles = _interopRequireDefault(require("../../getIconFillOrOutlineStyles"));

var buttonStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        theme = _ref.theme;
    var siteVariables = theme.siteVariables;
    var borderWidth = siteVariables.borderWidth;
    var borderFocusStyles = (0, _getBorderFocusStyles.default)(Object.assign({
      variables: siteVariables,
      borderPadding: borderWidth
    }, p.circular && {
      borderPadding: (0, _utils.pxToRem)(4)
    }));
    return Object.assign({
      height: v.height,
      minWidth: (0, _isNil2.default)(p.loading) ? v.minWidth : v.loadingMinWidth,
      maxWidth: v.maxWidth,
      color: v.color,
      backgroundColor: v.backgroundColor,
      borderRadius: v.borderRadius,
      display: 'inline-flex',
      justifyContent: 'center',
      alignItems: 'center',
      position: 'relative',
      padding: v.padding,
      verticalAlign: 'middle',
      cursor: 'pointer'
    }, p.size === 'small' && {
      padding: v.sizeSmallPadding,
      height: v.sizeSmallHeight,
      minWidth: v.sizeSmallMinWidth
    }, {}, !p.text && Object.assign({
      outline: 0,
      borderWidth: borderWidth,
      borderStyle: 'solid',
      borderColor: v.borderColor,
      boxShadow: v.boxShadow,
      ':hover': {
        color: v.colorHover,
        backgroundColor: v.backgroundColorHover,
        borderColor: v.borderColorHover
      },
      ':active': Object.assign({}, (0, _reactBindings.unstable_createAnimationStyles)('scaleDownSoft', theme), {
        color: v.colorActive,
        backgroundColor: v.backgroundColorActive,
        borderColor: v.borderColorActive,
        boxShadow: 'none'
      }),
      ':focus': borderFocusStyles[':focus'],
      ':focus-visible': Object.assign({}, borderFocusStyles[':focus-visible'], {
        backgroundColor: v.backgroundColorFocus,
        borderColor: v.borderColorFocus,
        color: v.colorFocus,
        borderWidth: borderWidth,
        ':hover': {
          borderColor: v.borderColorHover
        }
      })
    }, p.size === 'small' && {
      boxShadow: 'none'
    }), {}, p.circular && !p.text && Object.assign({
      minWidth: v.height,
      padding: 0,
      borderRadius: v.circularBorderRadius
    }, p.size === 'small' && {
      minWidth: v.sizeSmallHeight
    }), {}, p.text && Object.assign({
      color: v.textColor,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      padding: "0 ".concat((0, _utils.pxToRem)(8))
    }, (0, _getIconFillOrOutlineStyles.default)({
      outline: true
    }), {
      ':hover': Object.assign({
        color: v.textColorHover
      }, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      })),
      ':focus': Object.assign({
        boxShadow: 'none'
      }, borderFocusStyles[':focus']),
      ':focus-visible': borderFocusStyles[':focus-visible']
    }, p.primary && {
      color: v.textPrimaryColor
    }), {}, p.primary && !p.text && {
      color: v.primaryColor,
      backgroundColor: v.primaryBackgroundColor,
      borderColor: v.primaryBorderColor,
      boxShadow: v.primaryBoxShadow,
      ':active': Object.assign({}, (0, _reactBindings.unstable_createAnimationStyles)('scaleDownSoft', theme), {
        backgroundColor: v.primaryBackgroundColorActive,
        boxShadow: 'none'
      }),
      ':focus': borderFocusStyles[':focus'],
      ':focus-visible': Object.assign({}, borderFocusStyles[':focus-visible'], {
        backgroundColor: v.primaryBackgroundColorFocus
      }),
      ':hover': {
        color: v.primaryColorHover,
        backgroundColor: v.primaryBackgroundColorHover
      }
    }, {}, p.inverted && {
      backgroundColor: siteVariables.colorScheme.silver.background,
      borderColor: siteVariables.colorScheme.silver.border,
      color: siteVariables.colorScheme.silver.foreground,
      ':active': Object.assign({}, (0, _reactBindings.unstable_createAnimationStyles)('scaleDownSoft', theme), {
        backgroundColor: siteVariables.colorScheme.silver.backgroundPressed,
        color: siteVariables.colorScheme.silver.foregroundHover
      }),
      ':hover': {
        backgroundColor: siteVariables.colorScheme.silver.backgroundHover,
        color: siteVariables.colorScheme.silver.foregroundHover
      },
      ':focus': Object.assign({}, borderFocusStyles[':focus'], {
        boxShadow: 'none'
      }),
      ':focus-visible': Object.assign({}, borderFocusStyles[':focus-visible'], {
        backgroundColor: siteVariables.colorScheme.silver.backgroundPressed,
        color: siteVariables.colorScheme.silver.foregroundHover
      })
    }, {}, p.disabled && Object.assign({
      cursor: 'default',
      color: v.colorDisabled,
      boxShadow: 'none',
      pointerEvents: 'none',
      ':hover': {
        color: v.colorDisabled
      }
    }, p.text && {
      color: v.textColorDisabled,
      backgroundColor: 'transparent',
      ':hover': {
        color: v.textColorDisabled
      }
    }, {}, !p.text && {
      backgroundColor: v.backgroundColorDisabled,
      borderColor: v.borderColorDisabled
    }), {}, p.fluid && {
      width: '100%',
      maxWidth: '100%'
    }, {}, p.iconOnly && Object.assign({
      minWidth: v.height,
      padding: 0,
      ':hover': Object.assign({}, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      }), {
        color: v.textColorIconOnlyHover,
        background: v.backgroundColorIconOnlyHover
      })
    }, p.size === 'small' && {
      minWidth: v.sizeSmallHeight
    }));
  },
  icon: function icon(_ref2) {
    var p = _ref2.props,
        v = _ref2.variables;
    return Object.assign({
      display: 'inline-flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: v.iconSize,
      height: v.iconSize
    }, p.loading && {
      margin: 0,
      opacity: 0,
      width: 0
    }, {}, p.hasContent && Object.assign({
      margin: "0 ".concat((0, _utils.pxToRem)(10), " 0 0")
    }, p.iconPosition === 'after' && {
      margin: "0 0 0 ".concat((0, _utils.pxToRem)(10))
    }));
  },
  loader: function loader(_ref3) {
    var _Object$assign;

    var p = _ref3.props,
        v = _ref3.variables;
    return Object.assign((_Object$assign = {}, (0, _defineProperty2.default)(_Object$assign, "& .".concat(_Loader.loaderSlotClassNames.indicator), {
      width: p.size === 'small' ? v.sizeSmallLoaderSize : v.loaderSize,
      height: p.size === 'small' ? v.sizeSmallLoaderSize : v.loaderSize
    }), (0, _defineProperty2.default)(_Object$assign, "& .".concat(_Loader.loaderSlotClassNames.svg), {
      ':before': {
        animationName: {
          to: {
            transform: "translate3d(0, ".concat(p.size === 'small' ? v.sizeSmallLoaderSvgAnimationHeight : v.loaderSvgAnimationHeight, ", 0)")
          }
        },
        borderWidth: p.size === 'small' ? v.sizeSmallLoaderBorderSize : v.loaderBorderSize,
        width: p.size === 'small' ? v.sizeSmallLoaderSize : v.loaderSize,
        height: p.size === 'small' ? v.sizeSmallLoaderSvgHeight : v.loaderSvgHeight
      }
    }), _Object$assign), p.hasContent && {
      marginRight: (0, _utils.pxToRem)(4)
    });
  }
};
var _default = buttonStyles;
exports.default = _default;
