// @fluentui/accessibility/dist/commonjs/behaviors/Card/cardsContainerBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _types = require("../../focusZone/types");

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

/**
 * @description
 * Provides navigation between focusable children of Grid component with arrow keys in 4 directions.
 * Provides navigation inside the inner focus zone using Enter key
 *
 * @specification
 * Provides arrow key navigation in bidirectional direction.
 * Focus can be moved inside a child component with embeded inner FocusZone by pressing a specified key.
 */
var cardsContainerBehavior = function cardsContainerBehavior() {
  return {
    attributes: {},
    focusZone: {
      props: {
        shouldEnterInnerZone: function shouldEnterInnerZone(event) {
          return keyboardKey.getCode(event) === keyboardKey.Enter;
        },
        direction: _types.FocusZoneDirection.bidirectional
      }
    }
  };
};

var _default = cardsContainerBehavior;
exports.default = _default;
