// @fluentui/react-northstar/dist/dts/src/themes/teams/animations/slide.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var timingFunctions_1 = require("./timingFunctions");
var slideAnimations = {
    // Slides object down to original position
    slideDownEnterFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from top to bottom
    slideDownEnterMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from top to bottom
    slideDownEnterSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '700ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from top to bottom
    slideUpEnterFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from down to up
    slideUpEnterMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from bottom to top
    slideUpEnterSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '700ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // ---Horizontal Slide Animations---- //
    // Slides object in from right to left
    slideLeftEnterFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from right to left
    slideLeftEnterMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from right to left
    slideLeftEnterSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '700ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from left to right
    slideRightEnterFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from left to right
    slideRightEnterMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slides object in from left to right
    slideRightEnterSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
                '100%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '700ms',
        timingFunction: timingFunctions_1.easeOut,
        direction: 'forward',
        fillMode: 'forwards',
    },
    // Slide Exit Animation///
    slideDownExitFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideDownExitMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideDownExitSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideUpExitFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '300ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideUpExitMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideUpExitSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateY(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateY(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideRightExitFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '300ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideRightExitMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideRightExitSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideLeftExitFast: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '300ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideLeftExitMedium: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '400ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
    slideLeftExitSlow: {
        keyframe: function (_a) {
            var distance = _a.distance;
            return ({
                '0%': {
                    transform: 'translateX(0px)',
                    opacity: 1,
                },
                '100%': {
                    transform: "translateX(-" + distance + ")",
                    opacity: 0,
                },
            });
        },
        keyframeParams: { distance: '20px' },
        duration: '500ms',
        timingFunction: timingFunctions_1.easeIn,
        direction: 'forward',
        fillMode: 'forwards',
    },
};
exports.default = slideAnimations;
