// @fluentui/react-northstar/dist/dts/src/themes/teams/animations/timingFunctions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.easeIn = 'cubic-bezier(1.00,0.00,0.78,1.00)';
exports.easeOut = 'cubic-bezier(0.33,0.00,0.00,1.00)';
exports.easeEasy = 'cubic-bezier(0.33,0.00,0.67,1.00)';
exports.shotgun = 'cubic-bezier(0.00,0.00,0.00,1.00)';
exports.easeIn90 = 'cubic-bezier(0.90,0.00,0.78,1.00)';
exports.easeOut90 = 'cubic-bezier(0.33,0.00,0.10,1.00)';
exports.easeIn80 = 'cubic-bezier(0.80,0.00,0.78,1.0)';
exports.easeOut80 = 'cubic-bezier(0.33,0.00,0.20,1.00)';
exports.easeIn70 = 'cubic-bezier(0.70,0.00,0.78,1.00)';
exports.easeOut70 = 'cubic-bezier(0.33,0.00,0.30,1.00)';
exports.easeIn60 = 'cubic-bezier(0.60,0.00,0.78,1.00)';
exports.easeOut60 = 'cubic-bezier(0.33,0.00,0.40,1.00)';
exports.easeInAlt = 'cubic-bezier(1.00,0.00,0.67,1.00)';
exports.easeInSmooth = 'cubic-bezier(0.60,0.00,0.67,1.00)';
exports.easeOutSmooth = 'cubic-bezier(0.60,0.00,0.00,1.00)';
exports.expoIn5 = 'cubic-bezier(0.67,0.01,0.83,0.21)';
exports.expoOut5 = 'cubic-bezier(0.17,0.8,0.33,0.99)';
exports.expoInOut5 = 'cubic-bezier(0.78,0,0.22,1)';
exports.bounce1 = 'cubic-bezier(0.65, 0.00, 0.26, 1.31)';
exports.bounce2 = 'cubic-bezier(1.00, 0.00, 0.35, 1.36)';
