// @fluentui/react-northstar/dist/commonjs/components/Attachment/AttachmentDescription.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.attachmentDescriptionClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _Box = _interopRequireDefault(require("../Box/Box"));

var attachmentDescriptionClassName = 'ui-attachment__description';
/**
 * A AttachmentDescription provides more detailed information about the Attachment.
 */

exports.attachmentDescriptionClassName = attachmentDescriptionClassName;
var AttachmentDescription = (0, _reactBindings.compose)(_Box.default, {
  className: attachmentDescriptionClassName,
  displayName: 'AttachmentDescription',
  overrideStyles: true
});
AttachmentDescription.defaultProps = {
  as: 'span'
};
AttachmentDescription.propTypes = _utils.commonPropTypes.createCommon();
AttachmentDescription.shorthandConfig = {
  mappedProp: 'content'
};
var _default = AttachmentDescription;
exports.default = _default;
