// @fluentui/accessibility/dist/dts/src/behaviors/Table/gridCellMultipleFocusableBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var attributes_1 = require("../../attributes");
var types_1 = require("../../focusZone/types");
var keyboardKey = require("keyboard-key");
/**
 * @specification
 * Adds role='gridcell'.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 * Provides arrow key navigation in bidirectional direction.
 * Triggers 'focusCell' action with 'Escape' on 'root'.
 */
var gridCellMultipleFocusableBehavior = function (props) {
    var _a;
    return ({
        attributes: {
            root: (_a = {
                    role: 'gridcell'
                },
                _a[attributes_1.IS_FOCUSABLE_ATTRIBUTE] = true,
                _a),
        },
        focusZone: {
            props: {
                direction: types_1.FocusZoneDirection.bidirectional,
            },
        },
        keyActions: {
            root: {
                focusCell: {
                    keyCombinations: [{ keyCode: keyboardKey.Escape }],
                },
            },
        },
    });
};
exports.default = gridCellMultipleFocusableBehavior;
