// @fluentui/react-compose/lib/utils.js
import { __assign, __spreadArrays } from "tslib";
import * as ReactIs from 'react-is';
function computeDisplayNames(inputOptions, parentOptions) {
    if (inputOptions.overrideStyles) {
        return [inputOptions.displayName].filter(Boolean);
    }
    // To support styles composition we need to properly pick up display names
    return inputOptions.displayName
        ? parentOptions.displayNames.concat(inputOptions.displayName)
        : parentOptions.displayNames;
}
export var defaultComposeOptions = {
    className: process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉',
    displayName: '',
    displayNames: [],
    mapPropsToStylesPropsChain: [],
    render: function () { return null; },
    handledProps: [],
    overrideStyles: false,
    slots: {},
    mapPropsToSlotPropsChain: [],
    resolveSlotProps: function () { return ({}); },
};
export function mergeComposeOptions(input, inputOptions, parentOptions) {
    if (parentOptions === void 0) { parentOptions = defaultComposeOptions; }
    var mapPropsToSlotPropsChain = inputOptions.mapPropsToSlotProps
        ? __spreadArrays(parentOptions.mapPropsToSlotPropsChain, [inputOptions.mapPropsToSlotProps]) : parentOptions.mapPropsToSlotPropsChain;
    var resolveSlotProps = function (props) {
        return mapPropsToSlotPropsChain.reduce(function (acc, definition) {
            var nextProps = __assign({}, definition(props));
            var slots = __spreadArrays(Object.keys(acc), Object.keys(nextProps));
            var mergedSlotProps = {};
            slots.forEach(function (slot) {
                if (!mergedSlotProps[slot]) {
                    mergedSlotProps[slot] = __assign(__assign({}, acc[slot]), nextProps[slot]);
                }
            });
            return mergedSlotProps;
        }, {});
    };
    return {
        className: inputOptions.className || parentOptions.className,
        displayName: inputOptions.displayName || parentOptions.displayName,
        displayNames: computeDisplayNames(inputOptions, parentOptions),
        mapPropsToStylesPropsChain: inputOptions.mapPropsToStylesProps
            ? __spreadArrays(parentOptions.mapPropsToStylesPropsChain, [inputOptions.mapPropsToStylesProps]) : parentOptions.mapPropsToStylesPropsChain,
        render: typeof input === 'function' ? input : parentOptions.render,
        handledProps: __spreadArrays(parentOptions.handledProps, (inputOptions.handledProps || [])),
        overrideStyles: inputOptions.overrideStyles || false,
        slots: __assign(__assign({}, parentOptions.slots), inputOptions.slots),
        mapPropsToSlotPropsChain: mapPropsToSlotPropsChain,
        resolveSlotProps: resolveSlotProps,
    };
}
/**
 * compose() allows you to pass two inputs:
 * - React.forwardRef + static fluentComposeConfig, i.e. previously composed component
 * - a function
 */
export function wasComposedPreviously(input) {
    var isForwardRef = input.$$typeof === ReactIs.ForwardRef;
    return isForwardRef && !!input.fluentComposeConfig;
}
