// @fluentui/react-northstar/dist/dts/src/components/Design/Design.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var customPropTypes = require("@fluentui/react-proptypes");
// @ts-ignore
var react_fela_1 = require("react-fela");
var React = require("react");
var PropTypes = require("prop-types");
/**
 * The Design component provides a theme safe subset of CSS for designing layouts.
 */
function Design(_a) {
    var config = _a.config, children = _a.children;
    var context = React.useContext(react_fela_1.ThemeContext);
    var getConfig = React.useCallback(function () { return config; }, [config]);
    // Heads Up! Keep in sync with renderComponent.tsx
    var styleParam = {
        displayName: Design.displayName,
        disableAnimations: context.disableAnimations,
        theme: { direction: context.rtl ? 'rtl' : 'ltr' },
        sanitizeCss: context.performance.enableSanitizeCssPlugin,
    };
    var className = context.renderer.renderRule(getConfig, styleParam);
    return children({ className: className });
}
Design.displayName = 'Design';
Design.propTypes = {
    children: PropTypes.func.isRequired,
    config: customPropTypes.design.isRequired,
};
exports.default = Design;
