// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Embed/embedStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _utils = require("../../../../utils");

var _Embed = require("../../../../components/Embed/Embed");

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _playIndicatorUrl = _interopRequireDefault(require("./playIndicatorUrl"));

var _pauseIndicatorUrl = _interopRequireDefault(require("./pauseIndicatorUrl"));

var _default = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    var borderFocusStyles = (0, _getBorderFocusStyles.default)({
      variables: siteVariables
    });
    return Object.assign({
      display: 'inline-block',
      verticalAlign: 'middle',
      position: 'relative',
      cursor: 'pointer',
      width: v.width,
      height: v.height || 'auto',
      outline: 0
    }, borderFocusStyles[':focus'], {
      ':focus-visible': (0, _defineProperty2.default)({}, "& .".concat(_Embed.embedSlotClassNames.control), Object.assign({
        borderColor: v.focusBorderColor,
        opacity: 1
      }, borderFocusStyles[':focus-visible'])),
      ':hover': (0, _defineProperty2.default)({}, "& .".concat(_Embed.embedSlotClassNames.control), {
        opacity: 1,
        zIndex: v.zIndex
      })
    });
  },
  control: function control(_ref2) {
    var p = _ref2.props,
        v = _ref2.variables;
    return Object.assign({
      background: "0 no-repeat ".concat(v.controlBackgroundColor),
      backgroundPositionX: (0, _utils.pxToRem)(3),
      width: (0, _utils.pxToRem)(48),
      height: (0, _utils.pxToRem)(48),
      color: v.controlColor,
      padding: (0, _utils.pxToRem)(4),
      borderRadius: '50%',
      backgroundImage: (0, _playIndicatorUrl.default)(v.controlColor),
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center'
    }, p.active && {
      backgroundImage: (0, _pauseIndicatorUrl.default)(v.controlColor)
    }, {
      opacity: p.active ? 0 : 1,
      pointerEvents: 'none',
      transition: 'opacity .22s ease-in-out',
      left: '50%',
      position: 'absolute',
      top: '50%',
      transform: 'translate(-50%, -50%)'
    });
  },
  iframe: function iframe(_ref3) {
    var p = _ref3.props;
    return Object.assign({
      display: 'block'
    }, !p.iframeLoaded && {
      display: 'none'
    });
  }
};
exports.default = _default;
