// @fluentui/state/dist/commonjs/managers/carouselManager.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createCarouselManager = void 0;

var _createManager = _interopRequireDefault(require("../createManager"));

var createCarouselManager = function createCarouselManager() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return (0, _createManager.default)(Object.assign({}, config, {
    state: Object.assign({
      activeIndex: 0,
      prevActiveIndex: -1,
      ariaLiveOn: false,
      shouldFocusContainer: false,
      isFromKeyboard: false
    }, config.state),
    actions: Object.assign({
      setIndexes: function setIndexes(activeIndex, prevActiveIndex) {
        return function () {
          return {
            activeIndex: activeIndex,
            prevActiveIndex: prevActiveIndex
          };
        };
      },
      setAriaLiveOn: function setAriaLiveOn(ariaLiveOn) {
        return function () {
          return {
            ariaLiveOn: ariaLiveOn
          };
        };
      },
      setShouldFocusContainer: function setShouldFocusContainer(shouldFocusContainer) {
        return function () {
          return {
            shouldFocusContainer: shouldFocusContainer
          };
        };
      },
      setIsFromKeyboard: function setIsFromKeyboard(isFromKeyboard) {
        return function () {
          return {
            isFromKeyboard: isFromKeyboard
          };
        };
      }
    }, config.actions)
  }));
};

exports.createCarouselManager = createCarouselManager;
