// @fluentui/accessibility/dist/commonjs/behaviors/Checkbox/checkboxBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

var _attributes = require("../../attributes");

/**
 * @specification
 * Adds role='checkbox'. This allows screen readers to handle the component as a checkbox button.
 * Adds attribute 'aria-checked=true' based on the property 'checked'.
 * Adds attribute 'aria-disabled=true' based on the property 'disabled'.
 * Adds attribute 'tabIndex=0' to 'root' slot.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 */
var checkboxBehavior = function checkboxBehavior(props) {
  return {
    attributes: {
      root: (0, _defineProperty2.default)({
        'aria-checked': !!props.checked,
        'aria-disabled': props.disabled,
        role: 'checkbox',
        tabIndex: 0
      }, _attributes.IS_FOCUSABLE_ATTRIBUTE, true)
    },
    keyActions: {
      root: {
        performClick: {
          keyCombinations: [{
            keyCode: keyboardKey.Spacebar
          }]
        }
      }
    }
  };
};

var _default = checkboxBehavior;
exports.default = _default;
