// @fluentui/react-northstar/dist/commonjs/utils/felaStylisEnhancer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _felaUtils = require("fela-utils");

var _stylis = _interopRequireDefault(require("stylis"));

// @ts-ignore
// `stylis@3` is a CJS library, there are known issues with them:
// https://github.com/rollup/rollup/issues/1267#issuecomment-446681320
var Stylis = _stylis.default.default || _stylis.default; // We use Stylis only for vendor prefixing, all other capabilities are disabled

var stylis = new Stylis({
  cascade: false,
  compress: false,
  global: false,
  keyframe: false,
  preserve: false,
  semicolon: false
});

var felaStylisEnhancer = function felaStylisEnhancer(renderer) {
  return Object.assign({}, renderer, {
    _emitChange: function _emitChange(change) {
      if (change.type === _felaUtils.RULE_TYPE) {
        var prefixed = stylis('', change.declaration); // Fela uses objects by references, it's safe to override properties

        change.declaration = prefixed.slice(1, -1);
      }

      renderer._emitChange(change);
    }
  });
};

var _default = felaStylisEnhancer;
exports.default = _default;
