// @fluentui/react-northstar/dist/commonjs/utils/felaExpandCssShorthandsPlugin.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _inlineStyleExpandShorthand = require("inline-style-expand-shorthand");

// https://jsperf.com/array-indexof-vs-object-key-lookup2/12
var handledCssProps = {
  // 'font', Oops, is not supported by inline-style-expand-shorthand
  padding: true,
  margin: true,
  border: true,
  borderWidth: true,
  borderStyle: true,
  borderColor: true,
  borderTop: true,
  borderRight: true,
  borderBottom: true,
  borderLeft: true,
  borderRadius: true,
  background: true,
  outline: true
};

var _default = function _default() {
  var expandCssShorthands = function expandCssShorthands(styles) {
    return Object.keys(styles).reduce(function (acc, cssPropertyName) {
      var cssPropertyValue = styles[cssPropertyName];

      if (cssPropertyValue === null || typeof cssPropertyValue === 'undefined') {
        return Object.assign({}, acc, (0, _defineProperty2.default)({}, cssPropertyName, cssPropertyValue));
      }

      if (handledCssProps[cssPropertyName]) {
        var expandedProps = (0, _inlineStyleExpandShorthand.expandProperty)(cssPropertyName, cssPropertyValue);

        if (expandedProps) {
          return Object.assign({}, acc, {}, expandedProps);
        }
      }

      if (Array.isArray(cssPropertyValue)) {
        return Object.assign({}, acc, (0, _defineProperty2.default)({}, cssPropertyName, cssPropertyValue));
      }

      if ((0, _typeof2.default)(cssPropertyValue) === 'object') {
        return Object.assign({}, acc, (0, _defineProperty2.default)({}, cssPropertyName, expandCssShorthands(cssPropertyValue)));
      }

      return Object.assign({}, acc, (0, _defineProperty2.default)({}, cssPropertyName, cssPropertyValue));
    }, {});
  };

  return expandCssShorthands;
};

exports.default = _default;
