// @fluentui/react-northstar/dist/commonjs/utils/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  commonPropTypes: true,
  applyAccessibilityKeyHandlers: true,
  AutoControlledComponent: true,
  childrenExist: true,
  UIComponent: true,
  createRenderer: true,
  felaRenderer: true,
  rtlTextContainer: true,
  stringLiteralsArray: true,
  getOrGenerateIdFromShorthand: true,
  constants: true,
  mergeProviderContexts: true,
  renderComponent: true,
  htmlImageProps: true,
  htmlInputAttrs: true,
  htmlInputEvents: true,
  htmlInputProps: true,
  partitionHTMLProps: true,
  isBrowser: true,
  doesNodeContainClick: true,
  pxToRem: true,
  createComponent: true,
  getKindProp: true,
  screenReaderContainerStyles: true
};
Object.defineProperty(exports, "applyAccessibilityKeyHandlers", {
  enumerable: true,
  get: function get() {
    return _applyAccessibilityKeyHandlers.default;
  }
});
Object.defineProperty(exports, "AutoControlledComponent", {
  enumerable: true,
  get: function get() {
    return _AutoControlledComponent.default;
  }
});
Object.defineProperty(exports, "childrenExist", {
  enumerable: true,
  get: function get() {
    return _childrenExist.default;
  }
});
Object.defineProperty(exports, "UIComponent", {
  enumerable: true,
  get: function get() {
    return _UIComponent.default;
  }
});
Object.defineProperty(exports, "createRenderer", {
  enumerable: true,
  get: function get() {
    return _felaRenderer.createRenderer;
  }
});
Object.defineProperty(exports, "felaRenderer", {
  enumerable: true,
  get: function get() {
    return _felaRenderer.felaRenderer;
  }
});
Object.defineProperty(exports, "rtlTextContainer", {
  enumerable: true,
  get: function get() {
    return _rtlTextContainer.default;
  }
});
Object.defineProperty(exports, "stringLiteralsArray", {
  enumerable: true,
  get: function get() {
    return _stringLiteralsArray.default;
  }
});
Object.defineProperty(exports, "getOrGenerateIdFromShorthand", {
  enumerable: true,
  get: function get() {
    return _getOrGenerateIdFromShorthand.default;
  }
});
Object.defineProperty(exports, "constants", {
  enumerable: true,
  get: function get() {
    return _constants.default;
  }
});
Object.defineProperty(exports, "mergeProviderContexts", {
  enumerable: true,
  get: function get() {
    return _mergeProviderContexts.default;
  }
});
Object.defineProperty(exports, "renderComponent", {
  enumerable: true,
  get: function get() {
    return _renderComponent.default;
  }
});
Object.defineProperty(exports, "htmlImageProps", {
  enumerable: true,
  get: function get() {
    return _htmlPropsUtils.htmlImageProps;
  }
});
Object.defineProperty(exports, "htmlInputAttrs", {
  enumerable: true,
  get: function get() {
    return _htmlPropsUtils.htmlInputAttrs;
  }
});
Object.defineProperty(exports, "htmlInputEvents", {
  enumerable: true,
  get: function get() {
    return _htmlPropsUtils.htmlInputEvents;
  }
});
Object.defineProperty(exports, "htmlInputProps", {
  enumerable: true,
  get: function get() {
    return _htmlPropsUtils.htmlInputProps;
  }
});
Object.defineProperty(exports, "partitionHTMLProps", {
  enumerable: true,
  get: function get() {
    return _htmlPropsUtils.partitionHTMLProps;
  }
});
Object.defineProperty(exports, "isBrowser", {
  enumerable: true,
  get: function get() {
    return _isBrowser.default;
  }
});
Object.defineProperty(exports, "doesNodeContainClick", {
  enumerable: true,
  get: function get() {
    return _doesNodeContainClick.default;
  }
});
Object.defineProperty(exports, "pxToRem", {
  enumerable: true,
  get: function get() {
    return _fontSizeUtility.pxToRem;
  }
});
Object.defineProperty(exports, "createComponent", {
  enumerable: true,
  get: function get() {
    return _createComponent.default;
  }
});
Object.defineProperty(exports, "getKindProp", {
  enumerable: true,
  get: function get() {
    return _getKindProp.getKindProp;
  }
});
Object.defineProperty(exports, "screenReaderContainerStyles", {
  enumerable: true,
  get: function get() {
    return _accessibilityStyles.screenReaderContainerStyles;
  }
});
exports.commonPropTypes = void 0;

var _applyAccessibilityKeyHandlers = _interopRequireDefault(require("./applyAccessibilityKeyHandlers"));

var _AutoControlledComponent = _interopRequireDefault(require("./AutoControlledComponent"));

var _childrenExist = _interopRequireDefault(require("./childrenExist"));

var _UIComponent = _interopRequireDefault(require("./UIComponent"));

var _felaRenderer = require("./felaRenderer");

var _rtlTextContainer = _interopRequireDefault(require("./rtlTextContainer"));

var _stringLiteralsArray = _interopRequireDefault(require("./stringLiteralsArray"));

var _getOrGenerateIdFromShorthand = _interopRequireDefault(require("./getOrGenerateIdFromShorthand"));

var _factories = require("./factories");

Object.keys(_factories).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _factories[key];
    }
  });
});

var _constants = _interopRequireDefault(require("./constants"));

var _mergeProviderContexts = _interopRequireDefault(require("./mergeProviderContexts"));

var _renderComponent = _interopRequireWildcard(require("./renderComponent"));

Object.keys(_renderComponent).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _renderComponent[key];
    }
  });
});

var _htmlPropsUtils = require("./htmlPropsUtils");

var _isBrowser = _interopRequireDefault(require("./isBrowser"));

var _doesNodeContainClick = _interopRequireDefault(require("./doesNodeContainClick"));

var _fontSizeUtility = require("./fontSizeUtility");

var _createComponent = _interopRequireDefault(require("./createComponent"));

var _getKindProp = require("./getKindProp");

var _whatInput = require("./whatInput");

Object.keys(_whatInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _whatInput[key];
    }
  });
});

var _commonPropInterfaces = require("./commonPropInterfaces");

Object.keys(_commonPropInterfaces).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _commonPropInterfaces[key];
    }
  });
});

var _accessibilityStyles = require("./accessibility/Styles/accessibilityStyles");

var _commonPropTypes2 = require("./commonPropTypes");

// work around api-extractor limitation
var commonPropTypes;
exports.commonPropTypes = commonPropTypes;

(function (_commonPropTypes) {
  var createCommon = _commonPropTypes.createCommon = _commonPropTypes2.createCommon;
})(commonPropTypes || (exports.commonPropTypes = commonPropTypes = {}));
