// @fluentui/react-northstar/dist/commonjs/themes/teams-high-contrast/componentVariables.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Attachment", {
  enumerable: true,
  get: function get() {
    return _attachmentVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentAction", {
  enumerable: true,
  get: function get() {
    return _attachmentActionVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentBody", {
  enumerable: true,
  get: function get() {
    return _attachmentBodyVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentDescription", {
  enumerable: true,
  get: function get() {
    return _attachmentDescriptionVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentHeader", {
  enumerable: true,
  get: function get() {
    return _attachmentHeaderVariables.default;
  }
});
Object.defineProperty(exports, "AttachmentIcon", {
  enumerable: true,
  get: function get() {
    return _attachmentIconVariables.default;
  }
});
Object.defineProperty(exports, "Avatar", {
  enumerable: true,
  get: function get() {
    return _avatarVariables.default;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _buttonVariables.default;
  }
});
Object.defineProperty(exports, "Chat", {
  enumerable: true,
  get: function get() {
    return _chatVariables.default;
  }
});
Object.defineProperty(exports, "ChatMessage", {
  enumerable: true,
  get: function get() {
    return _chatMessageVariables.default;
  }
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function get() {
    return _checkboxVariables.default;
  }
});
Object.defineProperty(exports, "Dialog", {
  enumerable: true,
  get: function get() {
    return _dialogVariables.default;
  }
});
Object.defineProperty(exports, "Divider", {
  enumerable: true,
  get: function get() {
    return _dividerVariables.default;
  }
});
Object.defineProperty(exports, "Header", {
  enumerable: true,
  get: function get() {
    return _headerVariables.default;
  }
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function get() {
    return _inputVariables.default;
  }
});
Object.defineProperty(exports, "ListItem", {
  enumerable: true,
  get: function get() {
    return _listItemVariables.default;
  }
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function get() {
    return _menuVariables.default;
  }
});
Object.defineProperty(exports, "MenuItem", {
  enumerable: true,
  get: function get() {
    return _menuItemVariables.default;
  }
});
Object.defineProperty(exports, "RadioGroupItem", {
  enumerable: true,
  get: function get() {
    return _radioGroupItemVariables.default;
  }
});
Object.defineProperty(exports, "Segment", {
  enumerable: true,
  get: function get() {
    return _segmentVariables.default;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _textVariables.default;
  }
});
Object.defineProperty(exports, "TextArea", {
  enumerable: true,
  get: function get() {
    return _textAreaVariables.default;
  }
});
Object.defineProperty(exports, "Toolbar", {
  enumerable: true,
  get: function get() {
    return _toolbarVariables.default;
  }
});
Object.defineProperty(exports, "ToolbarItem", {
  enumerable: true,
  get: function get() {
    return _toolbarItemVariables.default;
  }
});
Object.defineProperty(exports, "ToolbarMenuItem", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuItemVariables.default;
  }
});
Object.defineProperty(exports, "HierarchicalTreeTitle", {
  enumerable: true,
  get: function get() {
    return _hierarchicalTreeTitleVariables.default;
  }
});
Object.defineProperty(exports, "Reaction", {
  enumerable: true,
  get: function get() {
    return _reactionVariables.default;
  }
});
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function get() {
    return _alertVariables.default;
  }
});
Object.defineProperty(exports, "AlertDismissAction", {
  enumerable: true,
  get: function get() {
    return _alertDismissActionVariables.default;
  }
});
Object.defineProperty(exports, "Provider", {
  enumerable: true,
  get: function get() {
    return _providerVariables.default;
  }
});
Object.defineProperty(exports, "Dropdown", {
  enumerable: true,
  get: function get() {
    return _dropdownVariables.default;
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function get() {
    return _labelVariables.default;
  }
});
Object.defineProperty(exports, "TooltipContent", {
  enumerable: true,
  get: function get() {
    return _tooltipContentVariables.default;
  }
});

var _attachmentVariables = _interopRequireDefault(require("./components/Attachment/attachmentVariables"));

var _attachmentActionVariables = _interopRequireDefault(require("./components/Attachment/attachmentActionVariables"));

var _attachmentBodyVariables = _interopRequireDefault(require("./components/Attachment/attachmentBodyVariables"));

var _attachmentDescriptionVariables = _interopRequireDefault(require("./components/Attachment/attachmentDescriptionVariables"));

var _attachmentHeaderVariables = _interopRequireDefault(require("./components/Attachment/attachmentHeaderVariables"));

var _attachmentIconVariables = _interopRequireDefault(require("./components/Attachment/attachmentIconVariables"));

var _avatarVariables = _interopRequireDefault(require("./components/Avatar/avatarVariables"));

var _buttonVariables = _interopRequireDefault(require("./components/Button/buttonVariables"));

var _chatVariables = _interopRequireDefault(require("./components/Chat/chatVariables"));

var _chatMessageVariables = _interopRequireDefault(require("./components/Chat/chatMessageVariables"));

var _checkboxVariables = _interopRequireDefault(require("./components/Checkbox/checkboxVariables"));

var _dialogVariables = _interopRequireDefault(require("./components/Dialog/dialogVariables"));

var _dividerVariables = _interopRequireDefault(require("./components/Divider/dividerVariables"));

var _headerVariables = _interopRequireDefault(require("./components/Header/headerVariables"));

var _inputVariables = _interopRequireDefault(require("./components/Input/inputVariables"));

var _listItemVariables = _interopRequireDefault(require("./components/List/listItemVariables"));

var _menuVariables = _interopRequireDefault(require("./components/Menu/menuVariables"));

var _menuItemVariables = _interopRequireDefault(require("./components/Menu/menuItemVariables"));

var _radioGroupItemVariables = _interopRequireDefault(require("./components/RadioGroup/radioGroupItemVariables"));

var _segmentVariables = _interopRequireDefault(require("./components/Segment/segmentVariables"));

var _textVariables = _interopRequireDefault(require("./components/Text/textVariables"));

var _textAreaVariables = _interopRequireDefault(require("./components/TextArea/textAreaVariables"));

var _toolbarVariables = _interopRequireDefault(require("./components/Toolbar/toolbarVariables"));

var _toolbarItemVariables = _interopRequireDefault(require("./components/Toolbar/toolbarItemVariables"));

var _toolbarMenuItemVariables = _interopRequireDefault(require("./components/Toolbar/toolbarMenuItemVariables"));

var _hierarchicalTreeTitleVariables = _interopRequireDefault(require("./components/HierarchicalTree/hierarchicalTreeTitleVariables"));

var _reactionVariables = _interopRequireDefault(require("./components/Reaction/reactionVariables"));

var _alertVariables = _interopRequireDefault(require("./components/Alert/alertVariables"));

var _alertDismissActionVariables = _interopRequireDefault(require("./components/Alert/alertDismissActionVariables"));

var _providerVariables = _interopRequireDefault(require("./components/Provider/providerVariables"));

var _dropdownVariables = _interopRequireDefault(require("./components/Dropdown/dropdownVariables"));

var _labelVariables = _interopRequireDefault(require("./components/Label/labelVariables"));

var _tooltipContentVariables = _interopRequireDefault(require("./components/Tooltip/tooltipContentVariables"));
