// @fluentui/accessibility/dist/dts/src/behaviors/HierarchicalTree/hierarchicalTreeTitleBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var keyboardKey = require("keyboard-key");
var attributes_1 = require("../../attributes");
/**
 * @description
 * Adds role 'treeitem' if the title is a leaf node inside the tree.
 * Adds 'tabIndex' as '-1' if the title is a leaf node inside the tree.
 *
 * @specification
 * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
 */
var hierarchicalTreeTitleBehavior = function (props) {
    var _a;
    return ({
        attributes: {
            root: tslib_1.__assign({}, (!props.hasSubtree && (_a = {
                    tabIndex: -1
                },
                _a[attributes_1.IS_FOCUSABLE_ATTRIBUTE] = true,
                _a.role = 'treeitem',
                _a))),
        },
        keyActions: {
            root: {
                performClick: {
                    keyCombinations: [{ keyCode: keyboardKey.Enter }, { keyCode: keyboardKey.Spacebar }],
                },
            },
        },
    });
};
exports.default = hierarchicalTreeTitleBehavior;
