// @fluentui/react-northstar/dist/dts/test/specs/components/Input/Input-test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var faker = require("faker");
var React = require("react");
var keyboardKey = require("keyboard-key");
var utils_1 = require("test/utils");
var commonTests_1 = require("test/specs/commonTests");
var Input_1 = require("src/components/Input/Input");
var Box_1 = require("src/components/Box/Box");
var testValue = 'test value';
var htmlInputAttrs = ['id', 'name', 'pattern', 'placeholder', 'type', 'value'];
var getInputAttrsObject = function (value) {
    return htmlInputAttrs.reduce(function (acc, attr) {
        acc[attr] = value;
        return acc;
    }, {});
};
var getInputDomNode = function (inputComp) {
    return inputComp.find('input').getDOMNode();
};
var setUserInputValue = function (inputComp, value) {
    inputComp.find('input').simulate('change', { target: { value: value } });
};
describe('Input', function () {
    commonTests_1.isConformant(Input_1.default, {
        eventTargets: {
            onChange: 'input',
            onKeyDown: 'input',
            onKeyPress: 'input',
            onKeyUp: 'input',
        },
        autoControlledProps: ['value'],
    });
    commonTests_1.implementsShorthandProp(Input_1.default)('input', Box_1.default, { mapsValueToProp: 'type' });
    describe('wrapper', function () {
        commonTests_1.implementsShorthandProp(Input_1.default)('wrapper', Box_1.default, { mapsValueToProp: 'children' });
        commonTests_1.implementsWrapperProp(Input_1.default, { wrapppedComponentSelector: 'input' });
    });
    it('renders a text <input> by default', function () {
        var inputComp = utils_1.mountWithProvider(React.createElement(Input_1.default, null));
        expect(inputComp.find('input[type="text"]').length).toBeGreaterThan(0);
    });
    describe('input related HTML attribute', function () {
        var inputAttrsObject = getInputAttrsObject(testValue);
        var domNode = getInputDomNode(utils_1.mountWithProvider(React.createElement(Input_1.default, tslib_1.__assign({}, inputAttrsObject))));
        htmlInputAttrs.forEach(function (attr) {
            it("'" + attr + "' is set correctly to '" + testValue + "'", function () {
                expect(domNode[attr]).toEqual(testValue);
            });
        });
    });
    describe('clearable', function () {
        it('calls onChange on Clearable icon click with an `empty` value', function () {
            var onChange = jest.fn();
            var wrapper = utils_1.mountWithProvider(React.createElement(Input_1.default, { clearable: true, defaultValue: faker.lorem.word(), onChange: onChange }));
            wrapper
                .find("." + Input_1.inputSlotClassNames.icon)
                .first()
                .simulate('click');
            expect(onChange).toBeCalledTimes(1);
            expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ type: 'click' }), expect.objectContaining({ value: '' }));
        });
        it('calls onChange on Escape key with an `empty` value and stops propagation when has content', function () {
            var onChange = jest.fn();
            var stopPropagation = jest.fn();
            var nativeEventStopPropagation = jest.fn();
            var wrapper = utils_1.mountWithProvider(React.createElement(Input_1.default, { clearable: true, defaultValue: faker.lorem.word(), onChange: onChange }));
            wrapper.find('input').simulate('keydown', {
                keyCode: keyboardKey.Escape,
                key: 'Escape',
                stopPropagation: stopPropagation,
                nativeEvent: { stopPropagation: nativeEventStopPropagation },
            });
            expect(onChange).toBeCalledTimes(1);
            expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ type: 'keydown' }), expect.objectContaining({ value: '' }));
            expect(stopPropagation).toHaveBeenCalledTimes(1);
            expect(nativeEventStopPropagation).toHaveBeenCalledTimes(1);
        });
        it('does not call onChange and does not stop propagation if is already empty', function () {
            var onChange = jest.fn();
            var stopPropagation = jest.fn();
            var nativeEventStopPropagation = jest.fn();
            var wrapper = utils_1.mountWithProvider(React.createElement(Input_1.default, { clearable: true, defaultValue: '', onChange: onChange }));
            wrapper.find('input').simulate('keydown', {
                keyCode: keyboardKey.Escape,
                key: 'Escape',
                stopPropagation: stopPropagation,
                nativeEvent: { stopPropagation: nativeEventStopPropagation },
            });
            expect(onChange).not.toBeCalled();
            expect(stopPropagation).not.toBeCalled();
            expect(nativeEventStopPropagation).not.toBeCalled();
        });
    });
    describe('icon', function () {
        var SearchIcon = function () { return React.createElement("span", null); };
        it('creates the Icon component when the icon shorthand is provided', function () {
            var inputComp = utils_1.mountWithProvider(React.createElement(Input_1.default, { icon: React.createElement(SearchIcon, null) }));
            expect(inputComp.find('SearchIcon').length).toBeGreaterThan(0);
        });
        it('creates an empty Icon component when the clearable prop is provided and the input has content, removes the icon and value when the icon is clicked', function () {
            var inputComp = utils_1.mountWithProvider(React.createElement(Input_1.default, { clearable: true }));
            var domNode = getInputDomNode(inputComp);
            setUserInputValue(inputComp, testValue); // user types into the input
            var iconComp = inputComp.find("Box[className~=\"" + Input_1.inputSlotClassNames.icon + "\"]");
            expect(domNode.value).toEqual(testValue); // input value is the one typed by the user
            expect(iconComp.length).toBeGreaterThan(0); // the 'x' icon appears
            iconComp.simulate('click'); // user clicks on 'x' icon
            expect(domNode.value).toEqual(''); // input value gets cleared
            expect(inputComp.find("Box[className~=\"" + Input_1.inputSlotClassNames.icon + "\"]").length).toEqual(0); // the 'x' icon disappears
        });
    });
    it('disabled prop makes the input un-actionable', function () {
        var inputComp = utils_1.mountWithProvider(React.createElement(Input_1.default, { disabled: true }));
        var domNode = getInputDomNode(inputComp);
        expect(domNode).toHaveAttribute('disabled');
        setUserInputValue(inputComp, testValue); // user types into the input
        expect(domNode.value).toEqual(''); // but nothing happens
        domNode.focus();
        expect(domNode).not.toHaveFocus();
        domNode.click();
        expect(domNode).not.toHaveFocus();
    });
});
