// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Card/cardVariables.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var utils_1 = require("../../../../utils");
exports.default = (function (siteVars) {
    return {
        backgroundColor: siteVars.colorScheme.default.background,
        backgroundColorDisabled: siteVars.colorScheme.default.backgroundDisabled,
        backgroundColorFocus: siteVars.colorScheme.default.backgroundFocus,
        backgroundColorHover: siteVars.colorScheme.default.backgroundHover,
        backgroundColorPressed: siteVars.colorScheme.default.backgroundPressed,
        borderStyle: 'solid',
        borderColor: siteVars.colorScheme.default.border,
        borderColorDisabled: siteVars.colorScheme.default.borderDisabled,
        borderColorHover: siteVars.colorScheme.default.borderHover,
        borderColorPressed: siteVars.colorScheme.default.borderPressed,
        borderWidth: siteVars.borderWidth,
        borderRadius: utils_1.pxToRem(4),
        boxShadow: 'none',
        boxShadowDisabled: 'none',
        boxShadowHover: siteVars.shadowLevel3,
        boxShadowFocus: siteVars.shadowLevel2,
        boxShadowPressed: siteVars.shadowLevel1,
        padding: utils_1.pxToRem(10),
        colorDisabled: siteVars.colorScheme.default.foregroundDisabled,
        compactPadding: utils_1.pxToRem(0),
        previewMargin: "0 0 " + utils_1.pxToRem(10) + " 0",
        headerMargin: "0 0 " + utils_1.pxToRem(10) + " 0",
        bodyMargin: "0 0 " + utils_1.pxToRem(10) + " 0",
        footerMargin: "0 0 " + utils_1.pxToRem(10) + " 0",
        fittedPreviewMargin: utils_1.pxToRem(0),
        fittedHeaderMargin: utils_1.pxToRem(0),
        fittedBodyMargin: utils_1.pxToRem(0),
        fittedFooterMargin: utils_1.pxToRem(0),
        previewMarginHorizontal: "0 " + utils_1.pxToRem(10) + " 0 0",
        topControlsTop: utils_1.pxToRem(10),
        topControlsRight: utils_1.pxToRem(0),
        // TODO: update with latest values from design
        width: utils_1.pxToRem(300),
        height: '100%',
        sizeSmallWidth: utils_1.pxToRem(200),
        sizeSmallHeight: '100%',
        sizeSmallPadding: utils_1.pxToRem(5),
        sizeLargeWidth: utils_1.pxToRem(500),
        sizeLargeHeight: '100%',
        sizeLargePadding: utils_1.pxToRem(20),
        fluidWidth: '100%',
        fluidHeight: '100%',
    };
});
