// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Card/cardStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var cardStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        theme = _ref.theme;
    var siteVariables = theme.siteVariables;
    var borderFocusStyles = (0, _getBorderFocusStyles.default)({
      variables: siteVariables,
      borderRadius: v.borderRadius
    });
    return Object.assign({
      display: 'flex',
      flexDirection: 'column',
      position: 'relative',
      padding: v.padding,
      width: v.width,
      height: v.height,
      backgroundColor: v.backgroundColor,
      boxShadow: v.boxShadow,
      ':hover': {
        backgroundColor: v.backgroundColorHover,
        borderColor: v.borderColorHover,
        boxShadow: v.boxShadowHover
      },
      borderWidth: v.borderWidth,
      borderStyle: v.borderStyle,
      borderColor: v.borderColor,
      borderRadius: v.borderRadius
    }, p.size === 'small' && {
      width: v.sizeSmallWidth,
      height: v.sizeSmallHeight,
      padding: v.sizeSmallPadding
    }, {}, p.size === 'large' && {
      width: v.sizeLargeWidth,
      height: v.sizeLargeHeight,
      padding: v.sizeLargePadding
    }, {}, p.fluid && {
      width: v.fluidWidth,
      height: v.fluidHeight
    }, {}, p.horizontal && {
      flexDirection: 'row'
    }, {}, p.compact && {
      padding: v.compactPadding
    }, {}, p.centered && {
      alignItems: 'center'
    }, {}, p.actionable && {
      cursor: 'pointer',
      ':focus-visible': {
        backgroundColor: v.backgroundColorFocus,
        boxShadow: v.boxShadowFocus
      },
      ':active': {
        backgroundColor: v.backgroundColorPressed,
        borderColor: v.borderColorPressed,
        boxShadow: v.boxShadowPressed
      }
    }, {}, p.disabled && {
      cursor: 'not-allowed',
      color: v.colorDisabled,
      backgroundColor: v.backgroundColorDisabled,
      borderColor: v.borderColorDisabled,
      boxShadow: v.boxShadowDisabled,
      ':hover': {
        boxShadow: v.boxShadowDisabled
      },
      ':focus-visible': {
        boxShadow: v.boxShadowDisabled
      },
      ':active': {
        boxShadow: v.boxShadowDisabled
      }
    }, {}, borderFocusStyles);
  }
};
var _default = cardStyles;
exports.default = _default;
