// @fluentui/react-northstar/dist/dts/src/components/Debug/DebugRect.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var DebugRect = /** @class */ (function (_super) {
    tslib_1.__extends(DebugRect, _super);
    function DebugRect() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.selectorRef = React.createRef();
        _this.setDebugSelectorPosition = function () {
            var fiberNav = _this.props.fiberNav;
            if (fiberNav &&
                fiberNav.domNode &&
                fiberNav.domNode.getBoundingClientRect &&
                typeof fiberNav.domNode.getBoundingClientRect === 'function' &&
                _this.selectorRef.current) {
                var rect = fiberNav.domNode.getBoundingClientRect();
                _this.selectorRef.current.style.top = rect.top + "px";
                _this.selectorRef.current.style.left = rect.left + "px";
                _this.selectorRef.current.style.width = rect.width + "px";
                _this.selectorRef.current.style.height = rect.height + "px";
                requestAnimationFrame(_this.setDebugSelectorPosition);
            }
        };
        return _this;
    }
    DebugRect.prototype.componentDidMount = function () {
        this.setDebugSelectorPosition();
    };
    DebugRect.prototype.componentDidUpdate = function (prevProps, prevState, snapshot) {
        this.setDebugSelectorPosition();
    };
    DebugRect.prototype.render = function () {
        var fiberNav = this.props.fiberNav;
        if (!fiberNav) {
            return null;
        }
        return (React.createElement("pre", { ref: this.selectorRef, style: {
                position: 'fixed',
                padding: 0,
                margin: 0,
                background: '#6495ed22',
                border: '1px solid #6495edcc',
                zIndex: 99999999,
                pointerEvents: 'none',
                userSelect: 'none',
            } },
            React.createElement("div", { style: {
                    position: 'absolute',
                    padding: '2px 4px',
                    margin: '-1px 0 0 -1px',
                    bottom: '100%',
                    left: 0,
                    color: '#fff',
                    background: '#6495ed',
                } },
                React.createElement("span", { style: { fontWeight: 'bold' } }, "<" + fiberNav.name + " />")),
            fiberNav.domNode && (React.createElement("div", { style: {
                    fontSize: '0.9em',
                    position: 'absolute',
                    padding: '2px 4px',
                    margin: '0 0 1px -1px',
                    top: '100%',
                    left: 0,
                    background: '#6495ed',
                } },
                React.createElement("strong", { style: { fontWeight: 'bold', color: 'hsl(160, 100%, 80%)' } }, fiberNav.domNode.tagName && fiberNav.domNode.tagName.toLowerCase()),
                fiberNav.domNode.hasAttribute &&
                    typeof fiberNav.domNode.hasAttribute === 'function' &&
                    fiberNav.domNode.hasAttribute('class') && (React.createElement("span", { style: { color: 'rgba(255, 255, 255, 0.75)' } },
                    ".",
                    (fiberNav.domNode.getAttribute('class') || '').replace(/ +/g, '.')))))));
    };
    return DebugRect;
}(React.Component));
exports.default = DebugRect;
