// @fluentui/react-northstar/dist/dts/src/components/Card/CardPreview.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var PropTypes = require("prop-types");
var React = require("react");
// @ts-ignore
var react_fela_1 = require("react-fela");
var types_1 = require("../../types");
var utils_1 = require("../../utils");
exports.cardPreviewClassName = 'ui-card__preview';
var CardPreview = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(CardPreview.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var className = props.className, design = props.design, styles = props.styles, variables = props.variables, children = props.children, horizontal = props.horizontal, fitted = props.fitted;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(CardPreview.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: CardPreview.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(CardPreview.displayName, {
        className: exports.cardPreviewClassName,
        mapPropsToStyles: function () { return ({ horizontal: horizontal, fitted: fitted }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), children));
    setEnd();
    return element;
};
CardPreview.displayName = 'CardPreview';
CardPreview.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { horizontal: PropTypes.bool, fitted: PropTypes.bool });
CardPreview.handledProps = Object.keys(CardPreview.propTypes);
CardPreview.create = utils_1.createShorthandFactory({ Component: CardPreview });
/**
 * A CardPreview is used to display data Card preview.
 */
exports.default = types_1.withSafeTypeForAs(CardPreview);
