// @fluentui/react-northstar/dist/dts/src/components/Card/CardHeader.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var PropTypes = require("prop-types");
var React = require("react");
// @ts-ignore
var react_fela_1 = require("react-fela");
var types_1 = require("../../types");
var utils_1 = require("../../utils");
exports.cardHeaderClassName = 'ui-card__header';
var CardHeader = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(CardHeader.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var className = props.className, design = props.design, styles = props.styles, variables = props.variables, children = props.children, fitted = props.fitted;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(CardHeader.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: CardHeader.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(CardHeader.displayName, {
        className: exports.cardHeaderClassName,
        mapPropsToStyles: function () { return ({ fitted: fitted }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), children));
    setEnd();
    return element;
};
CardHeader.displayName = 'CardHeader';
CardHeader.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { fitted: PropTypes.bool });
CardHeader.handledProps = Object.keys(CardHeader.propTypes);
CardHeader.create = utils_1.createShorthandFactory({ Component: CardHeader });
/**
 * A CardHeader is used to display data in Card header.
 */
exports.default = types_1.withSafeTypeForAs(CardHeader);
