// @fluentui/react-northstar/dist/dts/src/components/Card/CardTopControls.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var React = require("react");
// @ts-ignore
var react_fela_1 = require("react-fela");
var types_1 = require("../../types");
var utils_1 = require("../../utils");
exports.cardTopControlsClassName = 'ui-card__topcontrols';
var CardTopControls = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(CardTopControls.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var className = props.className, design = props.design, styles = props.styles, variables = props.variables, children = props.children;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(CardTopControls.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: CardTopControls.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(CardTopControls.displayName, {
        className: exports.cardTopControlsClassName,
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), children));
    setEnd();
    return element;
};
CardTopControls.displayName = 'CardTopControls';
CardTopControls.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon());
CardTopControls.handledProps = Object.keys(CardTopControls.propTypes);
CardTopControls.create = utils_1.createShorthandFactory({ Component: CardTopControls });
/**
 * A CardTopControls is used to render control elements in the top of a Card component.
 */
exports.default = types_1.withSafeTypeForAs(CardTopControls);
