// @fluentui/react-northstar/dist/dts/src/components/Card/CardColumn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var types_1 = require("../../types");
var utils_1 = require("../../utils");
var react_bindings_1 = require("@fluentui/react-bindings");
// @ts-ignore
var react_fela_1 = require("react-fela");
exports.cardColumnClassName = 'ui-card__column';
var CardColumn = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(CardColumn.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var className = props.className, design = props.design, styles = props.styles, variables = props.variables, children = props.children;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(CardColumn.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: CardColumn.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(CardColumn.displayName, {
        className: exports.cardColumnClassName,
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), children));
    setEnd();
    return element;
};
CardColumn.displayName = 'CardColumn';
CardColumn.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon());
CardColumn.handledProps = Object.keys(CardColumn.propTypes);
CardColumn.create = utils_1.createShorthandFactory({ Component: CardColumn });
/**
 * A CardColumn is used to display content in card as column
 */
exports.default = types_1.withSafeTypeForAs(CardColumn);
