// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Input/inputVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _default = function _default(siteVars) {
  return {
    colorDisabled: siteVars.colorScheme.brand.foregroundDisabled,
    iconPosition: 'absolute',
    iconRight: (0, _utils.pxToRem)(10),
    iconLeft: (0, _utils.pxToRem)(6),
    inputPaddingWithIconAtStart: "".concat((0, _utils.pxToRem)(5), " ").concat((0, _utils.pxToRem)(12), " ").concat((0, _utils.pxToRem)(5), " ").concat((0, _utils.pxToRem)(24)),
    inputPaddingWithIconAtEnd: "".concat((0, _utils.pxToRem)(5), " ").concat((0, _utils.pxToRem)(24), " ").concat((0, _utils.pxToRem)(5), " ").concat((0, _utils.pxToRem)(12)),
    inputPadding: "".concat((0, _utils.pxToRem)(5), " ").concat((0, _utils.pxToRem)(12)),
    borderColor: 'transparent',
    borderRadius: "".concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(2), " ").concat((0, _utils.pxToRem)(2)),
    borderWidth: "0 0 ".concat((0, _utils.pxToRem)(2), " 0"),
    backgroundColor: siteVars.colorScheme.default.background2,
    backgroundColorInverted: siteVars.colorScheme.default.background,
    fontColor: siteVars.colorScheme.default.foreground,
    fontSize: siteVars.fontSizes.medium,
    iconColor: siteVars.colorScheme.default.foreground,
    inputFocusBorderColor: "transparent transparent ".concat(siteVars.colorScheme.brand.borderFocus1, " transparent"),
    inputFocusBorderRadius: "".concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(2), " ").concat((0, _utils.pxToRem)(2)),
    placeholderColor: siteVars.colorScheme.default.foreground1
  };
};

exports.default = _default;
