// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Dropdown/dropdownVariables.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _utils = require("../../../../utils");

var _map = [3, 12].map(function (v) {
  return (0, _utils.pxToRem)(v);
}),
    _map2 = (0, _slicedToArray2.default)(_map, 2),
    cornerRadius = _map2[0],
    _12px_asRem = _map2[1];

var _default = function _default(siteVars) {
  return {
    backgroundColor: siteVars.colorScheme.default.background2,
    backgroundColorHover: siteVars.colorScheme.default.backgroundHover3,
    invertedBackgroundColor: siteVars.colorScheme.default.background,
    invertedBackgroundColorHover: siteVars.colorScheme.default.backgroundHover4,
    borderColor: 'transparent',
    borderColorHover: 'transparent',
    borderColorFocus: siteVars.colorScheme.brand.borderFocus1,
    borderWidth: '0px',
    openBorderColorHover: undefined,
    containerBorderRadius: "".concat(cornerRadius),
    disabledColor: siteVars.colorScheme.default.foregroundDisabled,
    openAboveContainerBorderRadius: "0 0 ".concat(cornerRadius, " ").concat(cornerRadius),
    openBelowContainerBorderRadius: "".concat(cornerRadius, " ").concat(cornerRadius, " 0 0"),
    searchBorderBottomWidth: (0, _utils.pxToRem)(2),
    color: siteVars.colorScheme.default.foreground1,
    comboboxPaddingButton: "0 ".concat(_12px_asRem),
    comboboxFlexBasis: (0, _utils.pxToRem)(50),
    aboveListBorderRadius: "".concat(cornerRadius, " ").concat(cornerRadius, " 0 0"),
    belowListBorderRadius: "0 0 ".concat(cornerRadius, " ").concat(cornerRadius),
    listBackgroundColor: siteVars.colorScheme.default.background,
    listBorderColor: 'transparent',
    listBorderWidth: '0px',
    listPadding: "".concat((0, _utils.pxToRem)(8), " 0 ").concat((0, _utils.pxToRem)(6)),
    listBoxShadow: siteVars.shadowLevel3,
    listMaxHeight: (0, _utils.pxToRem)(296),
    listItemFocusBorderWidth: (0, _utils.pxToRem)(1),
    listItemBackgroundColor: 'transparent',
    listItemBackgroundColorActive: siteVars.colorScheme.default.backgroundActive,
    listItemBackgroundColorHover: siteVars.colorScheme.default.backgroundHover,
    listItemColorActive: siteVars.colorScheme.default.backgroundFocus3,
    listItemColorHover: siteVars.colorScheme.default.foregroundHover,
    listItemSelectedColor: siteVars.colorScheme.default.foreground,
    listItemSelectedFontWeight: siteVars.fontWeightSemibold,
    // TODO: prod app uses 17.5px here, it should be 16px per the design guide!
    listItemHeaderLineHeight: siteVars.lineHeightSmall,
    listItemContentLineHeight: siteVars.lineHeightSmall,
    selectedItemBackgroundColor: siteVars.colorScheme.default.background,
    selectedItemBackgroundColorHover: siteVars.colorScheme.brand.backgroundHover2,
    selectedItemBorder: 'none',
    selectedItemColor: siteVars.colorScheme.default.foreground,
    selectedItemColorHover: siteVars.colorScheme.default.foregroundHover,
    selectedItemIconColor: siteVars.colorScheme.default.foreground1,
    selectedItemIconColorHover: siteVars.colorScheme.brand.foregroundHover,
    selectedItemColorFocus: siteVars.bodyColor,
    selectedItemsMaxHeight: (0, _utils.pxToRem)(82),
    toggleIndicatorSize: (0, _utils.pxToRem)(32),
    triggerButtonColorFocusActive: undefined,
    triggerButtonColorHover: siteVars.bodyColor,
    width: (0, _utils.pxToRem)(356),
    overlayZIndex: siteVars.zIndexes.overlay,
    // disabled state
    disabledBorderColorHover: 'transparent',
    disabledTriggerColorHover: siteVars.colorScheme.brand.foregroundDisabled,
    disabledBackgroundColorHover: siteVars.colorScheme.brand.backgroundDisabled,
    // these should only apply when there is content in the image/media slot:
    listItemHeaderFontSize: siteVars.fontSizes.medium,
    listItemHeaderColor: siteVars.colorScheme.default.foreground1,
    listItemContentFontSize: siteVars.fontSizes.small,
    listItemContentColor: siteVars.colorScheme.default.foreground2,
    headerMessageBackgroundColor: siteVars.colors.white,
    noResultsMessageBackgroundColor: siteVars.colors.white,
    loadingMessageBackgroundColor: siteVars.colors.white
  };
};

exports.default = _default;
