// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Dropdown/dropdownItemStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _DropdownItem = require("../../../../components/Dropdown/DropdownItem");

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _utils = require("../../../../utils");

var _checkableIndicatorUrl = _interopRequireDefault(require("./checkableIndicatorUrl"));

var dropdownItemStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    return Object.assign({
      display: 'flex',
      alignItems: 'center',
      minHeight: 0,
      padding: "".concat((0, _utils.pxToRem)(4), " ").concat((0, _utils.pxToRem)(11)),
      whiteSpace: 'nowrap',
      border: "".concat(v.listItemFocusBorderWidth, " solid transparent"),
      backgroundColor: v.listItemBackgroundColor
    }, p.selected && {
      fontWeight: v.listItemSelectedFontWeight,
      color: v.listItemSelectedColor
    }, {
      position: 'relative'
    }, p.active && Object.assign({}, p.isFromKeyboard && (0, _getBorderFocusStyles.default)({
      variables: siteVariables,
      borderRadius: 0
    })[':focus-visible'], {}, !p.isFromKeyboard && Object.assign({
      color: v.listItemColorHover,
      backgroundColor: v.listItemBackgroundColorHover
    }, p.hasHeader && (0, _defineProperty2.default)({}, "& .".concat(_DropdownItem.dropdownItemSlotClassNames.header), {
      color: v.listItemColorHover
    }), {}, p.hasContent && (0, _defineProperty2.default)({}, "& .".concat(_DropdownItem.dropdownItemSlotClassNames.content), {
      color: v.listItemColorHover
    }))));
  },
  image: function image() {
    return {
      margin: "".concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(12), " ").concat((0, _utils.pxToRem)(3), " ").concat((0, _utils.pxToRem)(4))
    };
  },
  header: function header(_ref4) {
    var p = _ref4.props,
        v = _ref4.variables;
    return Object.assign({
      flexGrow: 1,
      lineHeight: v.listItemHeaderLineHeight,
      fontSize: v.listItemHeaderFontSize,
      // if the item doesn't have content - i.e. it is header only - then it should use the content color
      color: v.listItemContentColor
    }, p.hasContent && {
      // if there is content it needs to be "tightened up" to the header
      marginBottom: (0, _utils.pxToRem)(-1),
      color: v.listItemHeaderColor
    }, {}, p.selected && {
      fontWeight: v.listItemSelectedFontWeight,
      color: v.listItemSelectedColor
    }, {
      whiteSpace: 'normal'
    });
  },
  content: function content(_ref5) {
    var v = _ref5.variables;
    return {
      flexGrow: 1,
      lineHeight: v.listItemContentLineHeight,
      fontSize: v.listItemContentFontSize,
      color: v.listItemContentColor
    };
  },
  checkableIndicator: function checkableIndicator(_ref6) {
    var v = _ref6.variables;
    return {
      backgroundImage: (0, _checkableIndicatorUrl.default)(v.listItemSelectedColor),
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      position: 'absolute',
      width: (0, _utils.pxToRem)(24),
      height: (0, _utils.pxToRem)(24),
      right: (0, _utils.pxToRem)(7),
      top: (0, _utils.pxToRem)(-3)
    };
  },
  endMedia: function endMedia() {
    return {
      flexShrink: 0,
      lineHeight: (0, _utils.pxToRem)(16)
    };
  },
  main: function main() {
    return {
      display: 'flex',
      flexDirection: 'column',
      flexGrow: 1,
      minWidth: 0 // needed for the truncate styles to work

    };
  }
};
var _default = dropdownItemStyles;
exports.default = _default;
