// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Table/tableRowStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var getBorderFocusStyles_1 = require("../../getBorderFocusStyles");
exports.default = {
    root: function (_a) {
        var _b = _a.props, header = _b.header, compact = _b.compact, v = _a.variables, siteVariables = _a.theme.siteVariables;
        var borderFocusStyles = getBorderFocusStyles_1.default({
            variables: siteVariables,
        });
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ display: 'flex', flexDirection: 'row', alignItems: 'center', fontSize: v.bodyFontSize, height: v.defaultRowHeight, color: v.color, backgroundColor: v.backgroundColor, borderWidth: v.borderWidth, borderStyle: 'solid', borderColor: 'transparent', borderBottomColor: v.rowBorderColor, padding: v.rowPadding, position: 'relative', width: '100%', ':hover': {
                color: v.hoverColor,
                backgroundColor: v.backgroundHoverColor,
                borderColor: v.rowBorderHoverColor,
            } }, borderFocusStyles), (header && {
            fontSize: v.headerFontSize,
            ':hover': {
                color: v.color,
                backgroundColor: v.backgroundColor,
            },
        })), (compact && {
            height: v.compactRowHeight,
        }));
    },
};
