// @fluentui/react-northstar/dist/commonjs/components/Reaction/ReactionGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.reactionGroupClassName = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _map2 = _interopRequireDefault(require("lodash/map"));

var _isNil2 = _interopRequireDefault(require("lodash/isNil"));

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var React = _interopRequireWildcard(require("react"));

var _types = require("../../types");

var _utils = require("../../utils");

var _Reaction = _interopRequireDefault(require("./Reaction"));

var reactionGroupClassName = 'ui-reactions';
exports.reactionGroupClassName = reactionGroupClassName;

var ReactionGroup =
/*#__PURE__*/
function (_UIComponent) {
  (0, _inherits2.default)(ReactionGroup, _UIComponent);

  function ReactionGroup() {
    (0, _classCallCheck2.default)(this, ReactionGroup);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ReactionGroup).apply(this, arguments));
  }

  (0, _createClass2.default)(ReactionGroup, [{
    key: "renderComponent",
    value: function renderComponent(_ref) {
      var ElementType = _ref.ElementType,
          classes = _ref.classes,
          accessibility = _ref.accessibility,
          styles = _ref.styles,
          unhandledProps = _ref.unhandledProps;
      var _this$props = this.props,
          children = _this$props.children,
          items = _this$props.items,
          content = _this$props.content;

      if ((0, _isNil2.default)(items)) {
        return React.createElement(ElementType, (0, _extends2.default)({}, accessibility.attributes.root, _utils.rtlTextContainer.getAttributes({
          forElements: [children, content]
        }), unhandledProps, {
          className: classes.root
        }), (0, _utils.childrenExist)(children) ? children : content);
      }

      return React.createElement(ElementType, (0, _extends2.default)({}, unhandledProps, {
        className: classes.root
      }), (0, _map2.default)(items, function (reaction) {
        return _Reaction.default.create(reaction, {
          defaultProps: function defaultProps() {
            return {
              styles: styles.reaction
            };
          }
        });
      }));
    }
  }]);
  return ReactionGroup;
}(_utils.UIComponent);

(0, _defineProperty2.default)(ReactionGroup, "create", void 0);
(0, _defineProperty2.default)(ReactionGroup, "displayName", 'ReactionGroup');
(0, _defineProperty2.default)(ReactionGroup, "deprecated_className", reactionGroupClassName);
(0, _defineProperty2.default)(ReactionGroup, "propTypes", Object.assign({}, _utils.commonPropTypes.createCommon(), {
  items: customPropTypes.collectionShorthand
}));
ReactionGroup.create = (0, _utils.createShorthandFactory)({
  Component: ReactionGroup,
  mappedProp: 'content',
  mappedArrayProp: 'items'
});
/**
 * A ReactionGroup groups multiple Reaction elements.
 */

var _default = (0, _types.withSafeTypeForAs)(ReactionGroup);

exports.default = _default;
