// @fluentui/react-northstar/dist/es/themes/teams/components/Table/tableRowStyles.js
import getBorderFocusStyles from '../../getBorderFocusStyles';
export default {
  root: function root(_ref) {
    var _ref$props = _ref.props,
        header = _ref$props.header,
        compact = _ref$props.compact,
        v = _ref.variables,
        siteVariables = _ref.theme.siteVariables;
    var borderFocusStyles = getBorderFocusStyles({
      variables: siteVariables
    });
    return Object.assign({
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      fontSize: v.bodyFontSize,
      height: v.defaultRowHeight,
      color: v.color,
      backgroundColor: v.backgroundColor,
      borderWidth: v.borderWidth,
      borderStyle: 'solid',
      borderColor: 'transparent',
      borderBottomColor: v.rowBorderColor,
      padding: v.rowPadding,
      position: 'relative',
      width: '100%',
      ':hover': {
        color: v.hoverColor,
        backgroundColor: v.backgroundHoverColor,
        borderColor: v.rowBorderHoverColor
      }
    }, borderFocusStyles, {}, header && {
      fontSize: v.headerFontSize,
      ':hover': {
        color: v.color,
        backgroundColor: v.backgroundColor
      }
    }, {}, compact && {
      height: v.compactRowHeight
    });
  }
};
