// @fluentui/accessibility/dist/commonjs/behaviors/Table/gridCellMultipleFocusableBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _attributes = require("../../attributes");

var _types = require("../../focusZone/types");

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

/**
 * @specification
 * Adds role='gridcell'.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 * Provides arrow key navigation in bidirectional direction.
 * Triggers 'focusCell' action with 'Escape' on 'root'.
 */
var gridCellMultipleFocusableBehavior = function gridCellMultipleFocusableBehavior(props) {
  return {
    attributes: {
      root: (0, _defineProperty2.default)({
        role: 'gridcell'
      }, _attributes.IS_FOCUSABLE_ATTRIBUTE, true)
    },
    focusZone: {
      props: {
        direction: _types.FocusZoneDirection.bidirectional
      }
    },
    keyActions: {
      root: {
        focusCell: {
          keyCombinations: [{
            keyCode: keyboardKey.Escape
          }]
        }
      }
    }
  };
};

var _default = gridCellMultipleFocusableBehavior;
exports.default = _default;
