// @fluentui/accessibility/dist/commonjs/behaviors/Table/gridHeaderRowBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _attributes = require("../../attributes");

var _types = require("../../focusZone/types");

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

var _gridHeaderCellBehavior = _interopRequireDefault(require("./gridHeaderCellBehavior"));

/**
 * @specification
 * Adds role='row'.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 * Provides arrow key navigation in horizontal direction.
 * Focused active element of the component is reset when TAB from the component.
 * When component's container element receives focus, focus will be set to the default focusable child element of the component.
 * Triggers 'unsetRowTabbable' action using 'shiftKey' + 'Tab' key on 'root'.
 * Applies 'gridHeaderCellBehavior' for 'cell' child component.
 */
var gridHeaderRowBehavior = function gridHeaderRowBehavior(props) {
  var _root;

  return {
    attributes: {
      root: (_root = {}, (0, _defineProperty2.default)(_root, _attributes.IS_FOCUSABLE_ATTRIBUTE, true), (0, _defineProperty2.default)(_root, "role", 'row'), _root)
    },
    focusZone: {
      props: {
        direction: _types.FocusZoneDirection.horizontal,
        shouldFocusInnerElementWhenReceivedFocus: true,
        shouldResetActiveElementWhenTabFromZone: true
      }
    },
    keyActions: {
      root: {
        unsetRowTabbable: {
          keyCombinations: [{
            keyCode: keyboardKey.Tab,
            shiftKey: true
          }]
        }
      }
    },
    childBehaviors: {
      cell: _gridHeaderCellBehavior.default
    }
  };
};

var _default = gridHeaderRowBehavior;
exports.default = _default;
