// @fluentui/react-bindings/dist/commonjs/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  unstable_getAccessibility: true,
  AutoFocusZone: true,
  FocusTrapZone: true,
  FocusZone: true,
  useAccessibility: true,
  useAutoControlled: true,
  unstable_useDispatchEffect: true,
  useIsomorphicLayoutEffect: true,
  useStateManager: true,
  useStyles: true,
  useUnhandledProps: true,
  unstable_createAnimationStyles: true,
  unstable_calculateAnimationTimeout: true,
  unstable_getStyles: true,
  deprecated_getTelemetry: true,
  useTelemetry: true,
  getElementType: true,
  getUnhandledProps: true
};
Object.defineProperty(exports, "unstable_getAccessibility", {
  enumerable: true,
  get: function get() {
    return _getAccessibility.default;
  }
});
Object.defineProperty(exports, "AutoFocusZone", {
  enumerable: true,
  get: function get() {
    return _AutoFocusZone.default;
  }
});
Object.defineProperty(exports, "FocusTrapZone", {
  enumerable: true,
  get: function get() {
    return _FocusTrapZone.default;
  }
});
Object.defineProperty(exports, "FocusZone", {
  enumerable: true,
  get: function get() {
    return _FocusZone.default;
  }
});
Object.defineProperty(exports, "useAccessibility", {
  enumerable: true,
  get: function get() {
    return _useAccessibility.default;
  }
});
Object.defineProperty(exports, "useAutoControlled", {
  enumerable: true,
  get: function get() {
    return _useAutoControlled.default;
  }
});
Object.defineProperty(exports, "unstable_useDispatchEffect", {
  enumerable: true,
  get: function get() {
    return _useDispatchEffect.default;
  }
});
Object.defineProperty(exports, "useIsomorphicLayoutEffect", {
  enumerable: true,
  get: function get() {
    return _useIsomorphicLayoutEffect.default;
  }
});
Object.defineProperty(exports, "useStateManager", {
  enumerable: true,
  get: function get() {
    return _useStateManager.default;
  }
});
Object.defineProperty(exports, "useStyles", {
  enumerable: true,
  get: function get() {
    return _useStyles.default;
  }
});
Object.defineProperty(exports, "useUnhandledProps", {
  enumerable: true,
  get: function get() {
    return _useUnhandledProps.default;
  }
});
Object.defineProperty(exports, "unstable_createAnimationStyles", {
  enumerable: true,
  get: function get() {
    return _createAnimationStyles.default;
  }
});
Object.defineProperty(exports, "unstable_calculateAnimationTimeout", {
  enumerable: true,
  get: function get() {
    return _calculateAnimationTimeout.default;
  }
});
Object.defineProperty(exports, "unstable_getStyles", {
  enumerable: true,
  get: function get() {
    return _getStyles.default;
  }
});
Object.defineProperty(exports, "deprecated_getTelemetry", {
  enumerable: true,
  get: function get() {
    return _useTelemetry.getTelemetry;
  }
});
Object.defineProperty(exports, "useTelemetry", {
  enumerable: true,
  get: function get() {
    return _useTelemetry.default;
  }
});
Object.defineProperty(exports, "getElementType", {
  enumerable: true,
  get: function get() {
    return _getElementType.default;
  }
});
Object.defineProperty(exports, "getUnhandledProps", {
  enumerable: true,
  get: function get() {
    return _getUnhandledProps.default;
  }
});

var _getAccessibility = _interopRequireDefault(require("./accessibility/getAccessibility"));

var _types = require("./accessibility/types");

Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _types[key];
    }
  });
});

var _reactCompose = require("@fluentui/react-compose");

Object.keys(_reactCompose).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactCompose[key];
    }
  });
});

var _AutoFocusZone = _interopRequireDefault(require("./FocusZone/AutoFocusZone"));

var _AutoFocusZone2 = require("./FocusZone/AutoFocusZone.types");

Object.keys(_AutoFocusZone2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _AutoFocusZone2[key];
    }
  });
});

var _FocusTrapZone = _interopRequireDefault(require("./FocusZone/FocusTrapZone"));

var _FocusTrapZone2 = require("./FocusZone/FocusTrapZone.types");

Object.keys(_FocusTrapZone2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _FocusTrapZone2[key];
    }
  });
});

var _FocusZone = _interopRequireDefault(require("./FocusZone/FocusZone"));

var _FocusZone2 = require("./FocusZone/FocusZone.types");

Object.keys(_FocusZone2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _FocusZone2[key];
    }
  });
});

var _focusUtilities = require("./FocusZone/focusUtilities");

Object.keys(_focusUtilities).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _focusUtilities[key];
    }
  });
});

var _useAccessibility = _interopRequireDefault(require("./hooks/useAccessibility"));

var _useAutoControlled = _interopRequireDefault(require("./hooks/useAutoControlled"));

var _useDispatchEffect = _interopRequireDefault(require("./hooks/useDispatchEffect"));

var _useIsomorphicLayoutEffect = _interopRequireDefault(require("./hooks/useIsomorphicLayoutEffect"));

var _useStateManager = _interopRequireDefault(require("./hooks/useStateManager"));

var _useStyles = _interopRequireDefault(require("./hooks/useStyles"));

var _useUnhandledProps = _interopRequireDefault(require("./hooks/useUnhandledProps"));

var _createAnimationStyles = _interopRequireDefault(require("./styles/createAnimationStyles"));

var _calculateAnimationTimeout = _interopRequireDefault(require("./styles/calculateAnimationTimeout"));

var _getStyles = _interopRequireDefault(require("./styles/getStyles"));

var _types2 = require("./styles/types");

Object.keys(_types2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _types2[key];
    }
  });
});

var _useTelemetry = _interopRequireWildcard(require("./telemetry/useTelemetry"));

var _types3 = require("./telemetry/types");

Object.keys(_types3).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _types3[key];
    }
  });
});

var _getElementType = _interopRequireDefault(require("./utils/getElementType"));

var _getUnhandledProps = _interopRequireDefault(require("./utils/getUnhandledProps"));
