// @fluentui/react-northstar/dist/commonjs/components/Toolbar/ToolbarMenuDivider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.toolbarMenuDividerClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var _styles = require("@fluentui/styles");

var React = _interopRequireWildcard(require("react"));

var _reactFela = require("react-fela");

var _utils = require("../../utils");

var _types = require("../../types");

var _toolbarVariablesContext = require("./toolbarVariablesContext");

// @ts-ignore
var toolbarMenuDividerClassName = 'ui-toolbar__menudivider';
exports.toolbarMenuDividerClassName = toolbarMenuDividerClassName;

var ToolbarMenuDivider = function ToolbarMenuDivider(props) {
  var context = React.useContext(_reactFela.ThemeContext);

  var _useTelemetry = (0, _reactBindings.useTelemetry)(ToolbarMenuDivider.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var accessibility = props.accessibility,
      className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables;
  var parentVariables = React.useContext(_toolbarVariablesContext.ToolbarVariablesContext);
  var getA11yProps = (0, _reactBindings.useAccessibility)(accessibility, {
    debugName: ToolbarMenuDivider.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(ToolbarMenuDivider.displayName, {
    className: toolbarMenuDividerClassName,
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: (0, _styles.mergeComponentVariables)(parentVariables, variables)
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(ToolbarMenuDivider.handledProps, props);
  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({}, unhandledProps, {
    className: classes.root
  })));
  setEnd();
  return element;
};

ToolbarMenuDivider.displayName = 'ToolbarMenuDivider';
ToolbarMenuDivider.propTypes = _utils.commonPropTypes.createCommon();
ToolbarMenuDivider.defaultProps = {
  as: 'li'
};
ToolbarMenuDivider.handledProps = Object.keys(ToolbarMenuDivider.propTypes);
ToolbarMenuDivider.create = (0, _utils.createShorthandFactory)({
  Component: ToolbarMenuDivider,
  mappedProp: 'content'
});
/**
 * A ToolbarMenuDivider adds non-actionable separator between items of ToolbarMenu.
 */

var _default = (0, _types.withSafeTypeForAs)(ToolbarMenuDivider);

exports.default = _default;
