// @fluentui/react-northstar/dist/es/components/Toolbar/ToolbarMenuItem.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _some from "lodash/some";
import _invoke from "lodash/invoke";
import { toolbarMenuItemBehavior, indicatorBehavior } from '@fluentui/accessibility';
import * as React from 'react';
import cx from 'classnames';
import * as PropTypes from 'prop-types';
import { EventListener } from '@fluentui/react-component-event-listener';
import { Ref } from '@fluentui/react-component-ref';
import * as customPropTypes from '@fluentui/react-proptypes';
import { focusAsync, useTelemetry, useStyles, useAutoControlled, getElementType, useUnhandledProps, useAccessibility } from '@fluentui/react-bindings';
import { mergeComponentVariables } from '@fluentui/styles'; // @ts-ignore

import { ThemeContext } from 'react-fela';
import { Unstable_NestingAuto } from '@fluentui/react-component-nesting-registry';
import { commonPropTypes, createShorthandFactory, childrenExist, doesNodeContainClick } from '../../utils';
import { withSafeTypeForAs } from '../../types';
import { getPopperPropsFromShorthand, Popper } from '../../utils/positioner';
import Box from '../Box/Box';
import Popup from '../Popup/Popup';
import ToolbarMenu from './ToolbarMenu';
import { ToolbarVariablesContext, ToolbarVariablesProvider } from './toolbarVariablesContext';
export var toolbarMenuItemClassName = 'ui-toolbar__menuitem';
export var toolbarMenuItemSlotClassNames = {
  activeIndicator: "".concat(toolbarMenuItemClassName, "__activeIndicator"),
  wrapper: "".concat(toolbarMenuItemClassName, "__wrapper"),
  submenu: "".concat(toolbarMenuItemClassName, "__submenu"),
  submenuIndicator: "".concat(toolbarMenuItemClassName, "__submenuIndicator")
};

var ToolbarMenuItem = function ToolbarMenuItem(props) {
  var context = React.useContext(ThemeContext);

  var _useTelemetry = useTelemetry(ToolbarMenuItem.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var active = props.active,
      activeIndicator = props.activeIndicator,
      children = props.children,
      content = props.content,
      disabled = props.disabled,
      submenuIndicator = props.submenuIndicator,
      icon = props.icon,
      menu = props.menu,
      popup = props.popup,
      wrapper = props.wrapper,
      inSubmenu = props.inSubmenu,
      className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables;

  var _useAutoControlled = useAutoControlled({
    defaultValue: props.defaultMenuOpen,
    value: props.menuOpen,
    initialValue: false
  }),
      _useAutoControlled2 = _slicedToArray(_useAutoControlled, 2),
      menuOpen = _useAutoControlled2[0],
      setMenuOpen = _useAutoControlled2[1];

  var itemRef = React.useRef();
  var menuRef = React.useRef();
  var parentVariables = React.useContext(ToolbarVariablesContext);
  var mergedVariables = mergeComponentVariables(parentVariables, variables);
  var ElementType = getElementType(props);
  var unhandledProps = useUnhandledProps(ToolbarMenuItem.handledProps, props);
  var getA11yProps = useAccessibility(props.accessibility, {
    debugName: ToolbarMenuItem.displayName,
    mapPropsToBehavior: function mapPropsToBehavior() {
      return {
        menu: menu,
        active: active,
        menuOpen: menuOpen,
        disabled: disabled,
        'aria-label': props['aria-label'],
        'aria-labelledby': props['aria-labelledby'],
        'aria-describedby': props['aria-describedby']
      };
    },
    actionHandlers: {
      performClick: function performClick(event) {
        event.preventDefault();
        handleClick(event);
      },
      openMenu: function openMenu(event) {
        return _openMenu(event);
      },
      closeAllMenusAndFocusNextParentItem: function closeAllMenusAndFocusNextParentItem(event) {
        return _closeAllMenus(event);
      },
      closeMenu: function closeMenu(event) {
        return _closeMenu(event);
      },
      closeMenuAndFocusTrigger: function closeMenuAndFocusTrigger(event) {
        return _closeMenu(event);
      },
      doNotNavigateNextParentItem: function doNotNavigateNextParentItem(event) {
        event.stopPropagation();
      },
      closeAllMenus: function closeAllMenus(event) {
        return _closeAllMenus(event);
      }
    },
    rtl: context.rtl
  });

  var _useStyles = useStyles(ToolbarMenuItem.displayName, {
    className: toolbarMenuItemClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        disabled: disabled,
        hasContent: !!content
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: mergedVariables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes,
      resolvedStyles = _useStyles.styles;

  var _openMenu = function _openMenu(e) {
    if (menu && !menuOpen) {
      trySetMenuOpen(true, e);
      e.stopPropagation();
      e.preventDefault();
    }
  };

  var _closeMenu = function _closeMenu(e) {
    if (!isSubmenuOpen()) {
      return;
    }

    trySetMenuOpen(false, e, function () {
      focusAsync(itemRef.current);
    });
    e.stopPropagation();
  };

  var _closeAllMenus = function _closeAllMenus(e) {
    if (!isSubmenuOpen()) {
      return;
    }

    trySetMenuOpen(false, e, function () {
      if (!inSubmenu) {
        focusAsync(itemRef.current);
      }
    }); // avoid spacebar scrolling the page

    if (!inSubmenu) {
      e.preventDefault();
    }
  };

  var isSubmenuOpen = function isSubmenuOpen() {
    return !!(menu && menuOpen);
  };

  var trySetMenuOpen = function trySetMenuOpen(newValue, e, onStateChanged) {
    setMenuOpen(newValue); // The reason why post-effect is not passed as callback to trySetState method
    // is that in 'controlled' mode the post-effect is applied before final re-rendering
    // which cause a broken behavior: for e.g. when it is needed to focus submenu trigger on ESC.
    // TODO: all DOM post-effects should be applied at componentDidMount & componentDidUpdated stages.

    onStateChanged && onStateChanged();

    _invoke(props, 'onMenuOpenChange', e, Object.assign({}, props, {
      menuOpen: newValue
    }));
  };

  var outsideClickHandler = function outsideClickHandler(getRefs) {
    return function (e) {
      var isItemClick = doesNodeContainClick(itemRef.current, e, context.target);

      var isNestedClick = _some(getRefs(), function (childRef) {
        return doesNodeContainClick(childRef.current, e, context.target);
      });

      var isInside = isItemClick || isNestedClick;

      if (!isInside) {
        trySetMenuOpen(false, e);
      }
    };
  };

  var handleMenuOverrides = function handleMenuOverrides(predefinedProps) {
    return {
      onItemClick: function onItemClick(e, itemProps) {
        var popup = itemProps.popup,
            menuOpen = itemProps.menuOpen;

        _invoke(predefinedProps, 'onItemClick', e, itemProps);

        if (popup) {
          return;
        }

        trySetMenuOpen(menuOpen, e);

        if (!menuOpen) {
          _invoke(itemRef.current, 'focus');
        }
      }
    };
  };

  var handleClick = function handleClick(e) {
    if (disabled) {
      e.preventDefault();
      return;
    }

    if (menu) {
      // the menuItem element was clicked => toggle the open/close and stop propagation
      trySetMenuOpen(!menuOpen, e);
      e.stopPropagation();
      e.preventDefault();
    }

    if (popup) {
      e.stopPropagation();
      e.preventDefault();
      return;
    }

    _invoke(props, 'onClick', e, props);
  };

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root,
    onClick: handleClick,
    disabled: disabled
  }, unhandledProps)), childrenExist(children) ? children : React.createElement(React.Fragment, null, Box.create(icon, {
    defaultProps: function defaultProps() {
      return {
        styles: resolvedStyles.icon
      };
    }
  }), content, active && Box.create(activeIndicator, {
    defaultProps: function defaultProps() {
      return {
        as: 'span',
        className: toolbarMenuItemSlotClassNames.activeIndicator,
        styles: resolvedStyles.activeIndicator,
        accessibility: indicatorBehavior
      };
    }
  }), menu && Box.create(submenuIndicator, {
    defaultProps: function defaultProps() {
      return {
        as: 'span',
        className: toolbarMenuItemSlotClassNames.submenuIndicator,
        styles: resolvedStyles.submenuIndicator,
        accessibility: indicatorBehavior
      };
    }
  })));
  var hasChildren = childrenExist(children);

  if (popup && !hasChildren) {
    var popupElement = Popup.create(popup, {
      defaultProps: function defaultProps() {
        return {
          trapFocus: true,
          onOpenChange: function onOpenChange(e) {
            e.stopPropagation();
          }
        };
      },
      overrideProps: {
        trigger: element,
        children: undefined // force-reset `children` defined for `Popup` as it collides with the `trigger`

      }
    });
    setEnd();
    return popupElement;
  }

  var menuItemInner = hasChildren ? children : React.createElement(Ref, {
    innerRef: itemRef
  }, element);
  var maybeSubmenu = menu && menuOpen ? React.createElement(Unstable_NestingAuto, null, function (getRefs, nestingRef) {
    return React.createElement(React.Fragment, null, React.createElement(Ref, {
      innerRef: function innerRef(node) {
        nestingRef.current = node;
        menuRef.current = node;
      }
    }, React.createElement(Popper, _extends({
      align: "top",
      position: context.rtl ? 'before' : 'after',
      targetRef: itemRef
    }, getPopperPropsFromShorthand(menu)), React.createElement(ToolbarVariablesProvider, {
      value: mergedVariables
    }, ToolbarMenu.create(menu, {
      defaultProps: function defaultProps() {
        return {
          className: toolbarMenuItemSlotClassNames.submenu,
          styles: resolvedStyles.menu,
          submenu: true,
          submenuIndicator: submenuIndicator
        };
      },
      overrideProps: handleMenuOverrides
    })))), React.createElement(EventListener, {
      listener: outsideClickHandler(getRefs),
      target: context.target,
      type: "click"
    }));
  }) : null;

  if (!wrapper) {
    setEnd();
    return menuItemInner;
  }

  var wrapperElement = Box.create(wrapper, {
    defaultProps: function defaultProps() {
      return getA11yProps('wrapper', {
        className: cx(toolbarMenuItemSlotClassNames.wrapper, classes.wrapper)
      });
    },
    overrideProps: function overrideProps() {
      return {
        children: React.createElement(React.Fragment, null, menuItemInner, maybeSubmenu)
      };
    }
  });
  setEnd();
  return wrapperElement;
};

ToolbarMenuItem.displayName = 'ToolbarMenuItem';
ToolbarMenuItem.propTypes = Object.assign({}, commonPropTypes.createCommon(), {
  active: PropTypes.bool,
  activeIndicator: customPropTypes.shorthandAllowingChildren,
  defaultMenuOpen: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: customPropTypes.shorthandAllowingChildren,
  index: PropTypes.number,
  submenuIndicator: customPropTypes.shorthandAllowingChildren,
  inSubmenu: PropTypes.bool,
  menu: PropTypes.oneOfType([customPropTypes.itemShorthand, customPropTypes.collectionShorthand]),
  menuOpen: PropTypes.bool,
  onClick: PropTypes.func,
  onMenuOpenChange: PropTypes.func,
  popup: PropTypes.oneOfType([PropTypes.shape(Object.assign({}, Popup.propTypes, {
    trigger: customPropTypes.never,
    children: customPropTypes.never
  })), PropTypes.string]),
  wrapper: customPropTypes.itemShorthand
});
ToolbarMenuItem.handledProps = Object.keys(ToolbarMenuItem.propTypes);
ToolbarMenuItem.defaultProps = {
  as: 'button',
  accessibility: toolbarMenuItemBehavior,
  wrapper: {
    as: 'li'
  },
  activeIndicator: {},
  submenuIndicator: {}
};
ToolbarMenuItem.create = createShorthandFactory({
  Component: ToolbarMenuItem,
  mappedProp: 'content'
});
/**
 * A ToolbarMenuItem renders ToolbarMenu item as button.
 */

export default withSafeTypeForAs(ToolbarMenuItem);
