// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Tooltip/tooltipContentStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _getPointerStyles = require("../../getPointerStyles");

var _pointerSvgUrl = _interopRequireDefault(require("../../pointerSvgUrl"));

var tooltipContentStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables;
    return Object.assign({
      display: 'block',
      position: 'absolute',
      maxWidth: v.maxWidth,
      zIndex: v.zIndex
    }, p.pointing && Object.assign({
      pointerEvents: 'all'
    }, (0, _getPointerStyles.getContainerStyles)({
      placement: p.basePlacement,
      margin: v.pointerMargin
    })), {}, !p.open && {
      opacity: 0
    });
  },
  pointer: function pointer(_ref2) {
    var p = _ref2.props,
        v = _ref2.variables,
        rtl = _ref2.rtl;
    return Object.assign({
      display: 'block',
      position: 'absolute',
      width: v.pointerWidth,
      height: v.pointerHeight
    }, (0, _getPointerStyles.getPointerStyles)({
      backgroundColor: v.backgroundColor,
      borderSize: v.borderSize,
      borderColor: 'transparent',
      gap: v.pointerGap,
      height: v.pointerHeight,
      width: v.pointerWidth,
      placement: p.basePlacement,
      rtl: rtl,
      svg: (0, _pointerSvgUrl.default)(v.backgroundColor)
    }));
  },
  content: function content(_ref3) {
    var p = _ref3.props,
        v = _ref3.variables;
    return Object.assign({
      display: 'block',
      padding: v.padding,
      textAlign: 'left',
      color: v.color,
      background: v.backgroundColor,
      borderRadius: v.borderRadius,
      boxShadow: v.boxShadow
    }, p.pointing && {
      pointerEvents: 'all'
    });
  }
};
var _default = tooltipContentStyles;
exports.default = _default;
