// @fluentui/react-component-ref/dist/dts/src/RefForward.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var PropTypes = require("prop-types");
var React = require("react");
var handleRef_1 = require("./handleRef");
var types_1 = require("./types");
var RefForward = /** @class */ (function (_super) {
    tslib_1.__extends(RefForward, _super);
    function RefForward() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.currentNode = null;
        _this.handleRefOverride = function (node) {
            var _a = _this.props, children = _a.children, innerRef = _a.innerRef;
            handleRef_1.default(children.ref, node);
            handleRef_1.default(innerRef, node);
            _this.currentNode = node;
        };
        return _this;
    }
    RefForward.prototype.componentDidUpdate = function (prevProps) {
        if (prevProps.innerRef !== this.props.innerRef) {
            handleRef_1.default(this.props.innerRef, this.currentNode);
        }
    };
    RefForward.prototype.componentWillUnmount = function () {
        delete this.currentNode;
    };
    RefForward.prototype.render = function () {
        var children = this.props.children;
        return React.cloneElement(children, {
            ref: this.handleRefOverride,
        });
    };
    RefForward.displayName = 'RefForward';
    // TODO: use Babel plugin for this
    RefForward.propTypes = process.env.NODE_ENV !== 'production'
        ? {
            children: PropTypes.element.isRequired,
            innerRef: types_1.refPropType.isRequired,
        }
        : {};
    return RefForward;
}(React.Component));
exports.default = RefForward;
