// @fluentui/accessibility/dist/dts/src/behaviors/SplitButton/splitButtonBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var keyboardKey = require("keyboard-key");
var _ = require("lodash");
var menuButtonBehavior_1 = require("../MenuButton/menuButtonBehavior");
/**
 * @description
 * Adds attribute 'tabIndex=-1' based on the property 'open' to 'trigger' slot.
 * Adds attribute 'aria-controls=menu-id' based on the property 'open' to 'trigger' slot.
 * Adds attribute 'aria-expanded=true' based on the property 'open' to 'trigger' slot.
 * Adds attribute 'id=trigger-id' based on the property 'triggerId' to 'trigger' slot.
 * Adds attribute 'id=menu-id' based on the property 'menuId' to 'menu' slot.
 * Adds attribute 'aria-labelledby=trigger-id' based on the property 'triggerId' to 'menu' slot.
 * Triggers 'closeAndFocusTrigger' action with 'Escape' or 'altKey'+'ArrowUp'.
 * Triggers 'openAndFocusFirst' action with 'altKey'+'ArrowDown' on 'root' slot.
 *
 * @specification
 * Adds attribute 'tabIndex=-1' to 'toggleButton' slot.
 * Adds attribute 'aria-haspopup=true' to 'toggleButton' slot.
 */
var splitButtonBehavior = function (props) {
    var splitButtonMenuButtonBehavior = function () {
        var menuButtonBehaviorData = menuButtonBehavior_1.default(props);
        menuButtonBehaviorData.attributes.trigger['aria-haspopup'] = undefined;
        return _.merge(menuButtonBehaviorData, {
            keyActions: {
                popup: {
                    closeAndFocusTrigger: {
                        keyCombinations: [{ keyCode: keyboardKey.Escape }, { keyCode: keyboardKey.ArrowUp, altKey: true }],
                    },
                },
                root: tslib_1.__assign({}, (!props.open && {
                    openAndFocusFirst: {
                        keyCombinations: [{ keyCode: keyboardKey.ArrowDown, altKey: true }],
                    },
                })),
            },
        });
    };
    return {
        attributes: {
            root: {},
            toggleButton: {
                tabIndex: -1,
                'aria-haspopup': true,
            },
        },
        childBehaviors: {
            menuButton: splitButtonMenuButtonBehavior,
        },
    };
};
exports.default = splitButtonBehavior;
