// @fluentui/react-northstar/dist/commonjs/components/Card/CardColumn.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.cardColumnClassName = void 0;

var React = _interopRequireWildcard(require("react"));

var _types = require("../../types");

var _utils = require("../../utils");

var _reactBindings = require("@fluentui/react-bindings");

var _reactFela = require("react-fela");

// @ts-ignore
var cardColumnClassName = 'ui-card__column';
exports.cardColumnClassName = cardColumnClassName;

var CardColumn = function CardColumn(props) {
  var context = React.useContext(_reactFela.ThemeContext);

  var _useTelemetry = (0, _reactBindings.useTelemetry)(CardColumn.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children;
  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(CardColumn.handledProps, props);
  var getA11yProps = (0, _reactBindings.useAccessibility)(props.accessibility, {
    debugName: CardColumn.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(CardColumn.displayName, {
    className: cardColumnClassName,
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardColumn.displayName = 'CardColumn';
CardColumn.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon());
CardColumn.handledProps = Object.keys(CardColumn.propTypes);
CardColumn.create = (0, _utils.createShorthandFactory)({
  Component: CardColumn
});
/**
 * A CardColumn is used to display content in card as column
 */

var _default = (0, _types.withSafeTypeForAs)(CardColumn);

exports.default = _default;
