// @fluentui/react-northstar/dist/commonjs/components/Card/CardBody.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.cardBodyClassName = void 0;

var React = _interopRequireWildcard(require("react"));

var _types = require("../../types");

var _utils = require("../../utils");

var _reactBindings = require("@fluentui/react-bindings");

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _reactFela = require("react-fela");

// @ts-ignore
var cardBodyClassName = 'ui-card__body';
exports.cardBodyClassName = cardBodyClassName;

var CardBody = function CardBody(props) {
  var context = React.useContext(_reactFela.ThemeContext);

  var _useTelemetry = (0, _reactBindings.useTelemetry)(CardBody.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children,
      fitted = props.fitted;
  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(CardBody.handledProps, props);
  var getA11yProps = (0, _reactBindings.useAccessibility)(props.accessibility, {
    debugName: CardBody.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(CardBody.displayName, {
    className: cardBodyClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        fitted: fitted
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardBody.displayName = 'CardBody';
CardBody.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  fitted: PropTypes.bool
});
CardBody.handledProps = Object.keys(CardBody.propTypes);
CardBody.create = (0, _utils.createShorthandFactory)({
  Component: CardBody
});
/**
 * A CardBody is used to display data in Card body.
 */

var _default = (0, _types.withSafeTypeForAs)(CardBody);

exports.default = _default;
