// @fluentui/react-northstar/dist/commonjs/components/Card/CardTopControls.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.cardTopControlsClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var React = _interopRequireWildcard(require("react"));

var _reactFela = require("react-fela");

var _types = require("../../types");

var _utils = require("../../utils");

// @ts-ignore
var cardTopControlsClassName = 'ui-card__topcontrols';
exports.cardTopControlsClassName = cardTopControlsClassName;

var CardTopControls = function CardTopControls(props) {
  var context = React.useContext(_reactFela.ThemeContext);

  var _useTelemetry = (0, _reactBindings.useTelemetry)(CardTopControls.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children;
  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(CardTopControls.handledProps, props);
  var getA11yProps = (0, _reactBindings.useAccessibility)(props.accessibility, {
    debugName: CardTopControls.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(CardTopControls.displayName, {
    className: cardTopControlsClassName,
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardTopControls.displayName = 'CardTopControls';
CardTopControls.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon());
CardTopControls.handledProps = Object.keys(CardTopControls.propTypes);
CardTopControls.create = (0, _utils.createShorthandFactory)({
  Component: CardTopControls
});
/**
 * A CardTopControls is used to render control elements in the top of a Card component.
 */

var _default = (0, _types.withSafeTypeForAs)(CardTopControls);

exports.default = _default;
