// @fluentui/react-northstar/dist/commonjs/components/Card/CardFooter.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.cardFooterClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var PropTypes = _interopRequireWildcard(require("prop-types"));

var React = _interopRequireWildcard(require("react"));

var _reactFela = require("react-fela");

var _types = require("../../types");

var _utils = require("../../utils");

// @ts-ignore
var cardFooterClassName = 'ui-card__footer';
exports.cardFooterClassName = cardFooterClassName;

var CardFooter = function CardFooter(props) {
  var context = React.useContext(_reactFela.ThemeContext);

  var _useTelemetry = (0, _reactBindings.useTelemetry)(CardFooter.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      children = props.children,
      fitted = props.fitted;
  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(CardFooter.handledProps, props);
  var getA11yProps = (0, _reactBindings.useAccessibility)(props.accessibility, {
    debugName: CardFooter.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(CardFooter.displayName, {
    className: cardFooterClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        fitted: fitted
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var element = React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root
  }, unhandledProps)), children);
  setEnd();
  return element;
};

CardFooter.displayName = 'CardFooter';
CardFooter.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  fitted: PropTypes.bool
});
CardFooter.handledProps = Object.keys(CardFooter.propTypes);
CardFooter.create = (0, _utils.createShorthandFactory)({
  Component: CardFooter
});
/**
 * A CardFooter is used to display data in Card component footer
 */

var _default = (0, _types.withSafeTypeForAs)(CardFooter);

exports.default = _default;
