// @fluentui/accessibility/dist/commonjs/behaviors/Carousel/carouselItemBehavior.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var keyboardKey = _interopRequireWildcard(require("keyboard-key"));

/**
 * @description
 * Adds attribute 'tabIndex=0' to 'root' slot if 'active' property and 'navigation' property is true. Sets the attribute to '-1' otherwise.
 * @specification
 * Adds attribute 'role=tabpanel' based on the property 'navigation' to 'root' slot.
 * Adds attribute 'aria-hidden=false' to 'root' slot if 'active' property is true. Sets the attribute to 'true' otherwise.
 * Triggers 'arrowKeysNavigationStopPropagation' action with 'ArrowRight' or 'ArrowLeft' on 'root'.
 */
var carouselItemBehavior = function carouselItemBehavior(props) {
  return {
    attributes: {
      root: {
        role: props.navigation ? 'tabpanel' : undefined,
        'aria-hidden': props.active ? 'false' : 'true',
        tabIndex: props.navigation ? props.active ? 0 : -1 : undefined
      }
    },
    keyActions: {
      root: {
        arrowKeysNavigationStopPropagation: {
          keyCombinations: [{
            keyCode: keyboardKey.ArrowRight
          }, {
            keyCode: keyboardKey.ArrowLeft
          }]
        }
      }
    }
  };
};

var _default = carouselItemBehavior;
exports.default = _default;
