// @fluentui/react-northstar/dist/dts/test/utils/withProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var styles_1 = require("@fluentui/styles");
var enzyme_1 = require("enzyme");
var React = require("react");
var react_fela_1 = require("react-fela");
var utils_1 = require("src/utils");
exports.EmptyThemeProvider = function (_a) {
    var children = _a.children, _b = _a.renderer, renderer = _b === void 0 ? utils_1.felaRenderer : _b, telemetry = _a.telemetry, _c = _a.theme, theme = _c === void 0 ? styles_1.emptyTheme : _c;
    var value = {
        renderer: renderer,
        target: document,
        disableAnimations: false,
        rtl: false,
        theme: theme,
        telemetry: telemetry,
        performance: {},
    };
    return React.createElement(react_fela_1.ThemeProvider, { theme: value }, children);
};
exports.mountWithProvider = function (node, options) {
    return enzyme_1.mount(node, tslib_1.__assign({ wrappingComponent: exports.EmptyThemeProvider }, options));
};
exports.mountWithProviderAndGetComponent = function (Component, elementToMount, options) {
    return exports.mountWithProvider(elementToMount, options).find(Component);
};
