// @fluentui/react-northstar/dist/es/components/Animation/Animation.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _isNil from "lodash/isNil";
import _noop from "lodash/noop";
import _invoke from "lodash/invoke";
import { useUnhandledProps, unstable_createAnimationStyles as createAnimationStyles, unstable_calculateAnimationTimeout as calculateAnimationTimeout, unstable_getStyles as getStyles, useTelemetry } from '@fluentui/react-bindings';
import cx from 'classnames';
import * as PropTypes from 'prop-types';
import * as React from 'react'; // @ts-ignore

import { ThemeContext } from 'react-fela';
import { Transition } from 'react-transition-group';
import { childrenExist, commonPropTypes } from '../../utils';
export var animationClassName = 'ui-animation';
/**
 * An Animation provides animation effects to rendered elements.
 */

var Animation = function Animation(props) {
  var _ref, _ref$props;

  var context = React.useContext(ThemeContext);

  var _useTelemetry = useTelemetry(Animation.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var appear = props.appear,
      children = props.children,
      className = props.className,
      delay = props.delay,
      direction = props.direction,
      duration = props.duration,
      fillMode = props.fillMode,
      iterationCount = props.iterationCount,
      keyframeParams = props.keyframeParams,
      mountOnEnter = props.mountOnEnter,
      name = props.name,
      playState = props.playState,
      timeout = props.timeout,
      timingFunction = props.timingFunction,
      visible = props.visible,
      unmountOnExit = props.unmountOnExit;

  var handleAnimationEvent = function handleAnimationEvent(event) {
    return function () {
      _invoke(props, event, null, props);
    };
  };

  var _React$useMemo = React.useMemo(function () {
    var animation = {
      name: name,
      keyframeParams: keyframeParams,
      duration: duration,
      delay: delay,
      iterationCount: iterationCount,
      direction: direction,
      fillMode: fillMode,
      playState: playState,
      timingFunction: timingFunction
    };
    return getStyles({
      className: animationClassName,
      displayNames: [Animation.displayName],
      props: {
        className: className,
        styles: createAnimationStyles(animation, context.theme)
      },
      disableAnimations: context.disableAnimations,
      renderer: context.renderer,
      rtl: context.rtl,
      performance: {
        enableSanitizeCssPlugin: false,
        enableStylesCaching: false,
        enableVariablesCaching: false,
        enableBooleanVariablesCaching: false
      },
      saveDebug: _noop,
      theme: context.theme
    });
  }, [className, context, name, delay, direction, duration, fillMode, iterationCount, keyframeParams, playState, timingFunction]),
      classes = _React$useMemo.classes,
      animationStyles = _React$useMemo.styles;

  if (_isNil(children)) {
    setEnd();
    return null;
  }

  var _animationStyles$root = animationStyles.root,
      animationDuration = _animationStyles$root.animationDuration,
      animationDelay = _animationStyles$root.animationDelay;
  var timeoutResult = timeout || calculateAnimationTimeout(animationDuration, animationDelay) || 0;
  var unhandledProps = useUnhandledProps(Animation.handledProps, props);
  var isChildrenFunction = typeof children === 'function';
  var child = childrenExist(children) && !isChildrenFunction && React.Children.only(children);
  var element = React.createElement(Transition, _extends({
    in: visible,
    appear: appear,
    mountOnEnter: mountOnEnter,
    unmountOnExit: unmountOnExit,
    timeout: timeoutResult,
    onEnter: handleAnimationEvent('onEnter'),
    onEntering: handleAnimationEvent('onEntering'),
    onEntered: handleAnimationEvent('onEntered'),
    onExit: handleAnimationEvent('onExit'),
    onExiting: handleAnimationEvent('onExiting'),
    onExited: handleAnimationEvent('onExited')
  }, unhandledProps, {
    className: !isChildrenFunction ? cx(classes.root, (_ref = child) === null || _ref === void 0 ? void 0 : (_ref$props = _ref.props) === null || _ref$props === void 0 ? void 0 : _ref$props.className) : ''
  }), isChildrenFunction ? function () {
    return children({
      classes: classes.root
    });
  } : child);
  setEnd();
  return element;
};

Animation.displayName = 'Animation';
Animation.propTypes = Object.assign({}, commonPropTypes.createCommon({
  accessibility: false,
  as: false,
  content: false,
  children: false
}), {
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
  name: PropTypes.string,
  delay: PropTypes.string,
  direction: PropTypes.string,
  duration: PropTypes.string,
  fillMode: PropTypes.string,
  iterationCount: PropTypes.string,
  keyframeParams: PropTypes.object,
  playState: PropTypes.string,
  timingFunction: PropTypes.string,
  visible: PropTypes.bool,
  appear: PropTypes.bool,
  mountOnEnter: PropTypes.bool,
  unmountOnExit: PropTypes.bool,
  timeout: PropTypes.oneOfType([PropTypes.number, PropTypes.shape({
    appear: PropTypes.number,
    enter: PropTypes.number,
    exit: PropTypes.number
  })]),
  onEnter: PropTypes.func,
  onEntering: PropTypes.func,
  onEntered: PropTypes.func,
  onExit: PropTypes.func,
  onExiting: PropTypes.func,
  onExited: PropTypes.func
});
Animation.handledProps = Object.keys(Animation.propTypes);
export default Animation;
