// @fluentui/react-northstar/dist/commonjs/components/Button/ButtonContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.buttonContentClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var _utils = require("../../utils");

var _Box = _interopRequireDefault(require("../Box/Box"));

var buttonContentClassName = 'ui-button__content';
/**
 * A ButtonContent allows a user to have a dedicated component that can be targeted from the theme.
 */

exports.buttonContentClassName = buttonContentClassName;
var ButtonContent = (0, _reactBindings.compose)(_Box.default, {
  className: buttonContentClassName,
  displayName: 'ButtonContent',
  mapPropsToStylesProps: function mapPropsToStylesProps(props) {
    return {
      size: props.size
    };
  },
  handledProps: ['size'],
  overrideStyles: true
});
ButtonContent.defaultProps = {
  as: 'span'
};
ButtonContent.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  size: customPropTypes.size
});
ButtonContent.shorthandConfig = {
  mappedProp: 'content'
};
var _default = ButtonContent;
exports.default = _default;
