// @fluentui/react-northstar/dist/dts/src/components/Status/Status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var accessibility_1 = require("@fluentui/accessibility");
var react_bindings_1 = require("@fluentui/react-bindings");
var customPropTypes = require("@fluentui/react-proptypes");
var PropTypes = require("prop-types");
var React = require("react");
// @ts-ignore
var react_fela_1 = require("react-fela");
var utils_1 = require("../../utils");
var types_1 = require("../../types");
var Box_1 = require("../Box/Box");
exports.statusClassName = 'ui-status';
var Status = function (props) {
    var context = React.useContext(react_fela_1.ThemeContext);
    var _a = react_bindings_1.useTelemetry(Status.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var className = props.className, color = props.color, icon = props.icon, size = props.size, state = props.state, design = props.design, styles = props.styles, variables = props.variables;
    var _b = react_bindings_1.useStyles(Status.displayName, {
        className: exports.statusClassName,
        mapPropsToStyles: function () { return ({
            color: color,
            size: size,
            state: state,
        }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }), classes = _b.classes, resolvedStyles = _b.styles;
    var getA11Props = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: Status.displayName,
        rtl: context.rtl,
    });
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(Status.handledProps, props);
    var iconElement = Box_1.default.create(icon, {
        defaultProps: function () {
            return getA11Props('icon', {
                styles: resolvedStyles.icon,
                as: 'span',
            });
        },
    });
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), iconElement));
    setEnd();
    return element;
};
Status.displayName = 'Status';
Status.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({
    children: false,
    content: false,
})), { color: PropTypes.string, icon: customPropTypes.shorthandAllowingChildren, size: customPropTypes.size, state: PropTypes.oneOf(['success', 'info', 'warning', 'error', 'unknown']) });
Status.handledProps = Object.keys(Status.propTypes);
Status.defaultProps = {
    accessibility: accessibility_1.statusBehavior,
    as: 'span',
    size: 'medium',
    state: 'unknown',
};
Status.create = utils_1.createShorthandFactory({ Component: Status, mappedProp: 'state' });
/**
 * A Status represents someone's or something's state.
 *
 * @accessibility
 * Implements [ARIA img](https://www.w3.org/TR/wai-aria-1.1/#img) role.
 */
exports.default = types_1.withSafeTypeForAs(Status);
