// @fluentui/accessibility/dist/commonjs/behaviors/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  alertBehavior: true,
  alertWarningBehavior: true,
  alertBaseBehavior: true,
  attachmentBehavior: true,
  buttonBehavior: true,
  buttonGroupBehavior: true,
  toggleButtonBehavior: true,
  imageBehavior: true,
  menuBehavior: true,
  menuItemBehavior: true,
  menuDividerBehavior: true,
  submenuBehavior: true,
  basicListBehavior: true,
  listBehavior: true,
  basicListItemBehavior: true,
  listItemBehavior: true,
  navigableListBehavior: true,
  navigableListItemBehavior: true,
  selectableListBehavior: true,
  selectableListItemBehavior: true,
  loaderBehavior: true,
  inputBehavior: true,
  iconBehavior: true,
  indicatorBehavior: true,
  tabBehavior: true,
  tabListBehavior: true,
  menuAsToolbarBehavior: true,
  toolbarMenuBehavior: true,
  toolbarMenuItemBehavior: true,
  menuItemAsToolbarButtonBehavior: true,
  toolbarBehavior: true,
  toolbarItemBehavior: true,
  toolbarMenuItemCheckboxBehavior: true,
  toolbarMenuItemRadioBehavior: true,
  toolbarMenuRadioGroupBehavior: true,
  toolbarRadioGroupBehavior: true,
  toolbarRadioGroupItemBehavior: true,
  radioGroupBehavior: true,
  radioGroupItemBehavior: true,
  popupBehavior: true,
  chatBehavior: true,
  chatMessageBehavior: true,
  gridBehavior: true,
  gridHorizontalBehavior: true,
  hierarchicalTreeBehavior: true,
  hierarchicalTreeItemBehavior: true,
  hierarchicalTreeTitleBehavior: true,
  hierarchicalSubtreeBehavior: true,
  dialogBehavior: true,
  statusBehavior: true,
  embedBehavior: true,
  accordionBehavior: true,
  accordionTitleBehavior: true,
  accordionContentBehavior: true,
  checkboxBehavior: true,
  tooltipAsDescriptionBehavior: true,
  tooltipAsLabelBehavior: true,
  sliderBehavior: true,
  menuButtonBehavior: true,
  splitButtonBehavior: true,
  treeBehavior: true,
  treeItemBehavior: true,
  treeTitleBehavior: true,
  textAreaBehavior: true,
  treeAsListBehavior: true,
  treeItemAsListItemBehavior: true,
  treeTitleAsListItemTitleBehavior: true,
  carouselBehavior: true,
  carouselItemBehavior: true,
  tableBehavior: true,
  tableRowBehavior: true,
  tableCellBehavior: true,
  tableHeaderCellBehavior: true,
  gridNestedBehavior: true,
  gridHeaderRowBehavior: true,
  gridHeaderCellBehavior: true,
  gridRowBehavior: true,
  gridRowNestedBehavior: true,
  gridCellBehavior: true,
  gridCellMultipleFocusableBehavior: true,
  gridCellWithFocusableElementBehavior: true,
  cardBehavior: true,
  cardFocusableBehavior: true,
  cardChildrenFocusableBehavior: true,
  cardsContainerBehavior: true,
  videoBehavior: true
};
Object.defineProperty(exports, "alertBehavior", {
  enumerable: true,
  get: function get() {
    return _alertBehavior.default;
  }
});
Object.defineProperty(exports, "alertWarningBehavior", {
  enumerable: true,
  get: function get() {
    return _alertWarningBehavior.default;
  }
});
Object.defineProperty(exports, "alertBaseBehavior", {
  enumerable: true,
  get: function get() {
    return _alertBaseBehavior.default;
  }
});
Object.defineProperty(exports, "attachmentBehavior", {
  enumerable: true,
  get: function get() {
    return _attachmentBehavior.default;
  }
});
Object.defineProperty(exports, "buttonBehavior", {
  enumerable: true,
  get: function get() {
    return _buttonBehavior.default;
  }
});
Object.defineProperty(exports, "buttonGroupBehavior", {
  enumerable: true,
  get: function get() {
    return _buttonGroupBehavior.default;
  }
});
Object.defineProperty(exports, "toggleButtonBehavior", {
  enumerable: true,
  get: function get() {
    return _toggleButtonBehavior.default;
  }
});
Object.defineProperty(exports, "imageBehavior", {
  enumerable: true,
  get: function get() {
    return _imageBehavior.default;
  }
});
Object.defineProperty(exports, "menuBehavior", {
  enumerable: true,
  get: function get() {
    return _menuBehavior.default;
  }
});
Object.defineProperty(exports, "menuItemBehavior", {
  enumerable: true,
  get: function get() {
    return _menuItemBehavior.default;
  }
});
Object.defineProperty(exports, "menuDividerBehavior", {
  enumerable: true,
  get: function get() {
    return _menuDividerBehavior.default;
  }
});
Object.defineProperty(exports, "submenuBehavior", {
  enumerable: true,
  get: function get() {
    return _submenuBehavior.default;
  }
});
Object.defineProperty(exports, "basicListBehavior", {
  enumerable: true,
  get: function get() {
    return _listBehavior.default;
  }
});
Object.defineProperty(exports, "listBehavior", {
  enumerable: true,
  get: function get() {
    return _listBehavior.default;
  }
});
Object.defineProperty(exports, "basicListItemBehavior", {
  enumerable: true,
  get: function get() {
    return _basicListItemBehavior.default;
  }
});
Object.defineProperty(exports, "listItemBehavior", {
  enumerable: true,
  get: function get() {
    return _listItemBehavior.default;
  }
});
Object.defineProperty(exports, "navigableListBehavior", {
  enumerable: true,
  get: function get() {
    return _navigableListBehavior.default;
  }
});
Object.defineProperty(exports, "navigableListItemBehavior", {
  enumerable: true,
  get: function get() {
    return _navigableListItemBehavior.default;
  }
});
Object.defineProperty(exports, "selectableListBehavior", {
  enumerable: true,
  get: function get() {
    return _selectableListBehavior.default;
  }
});
Object.defineProperty(exports, "selectableListItemBehavior", {
  enumerable: true,
  get: function get() {
    return _selectableListItemBehavior.default;
  }
});
Object.defineProperty(exports, "loaderBehavior", {
  enumerable: true,
  get: function get() {
    return _loaderBehavior.default;
  }
});
Object.defineProperty(exports, "inputBehavior", {
  enumerable: true,
  get: function get() {
    return _inputBehavior.default;
  }
});
Object.defineProperty(exports, "iconBehavior", {
  enumerable: true,
  get: function get() {
    return _iconBehavior.default;
  }
});
Object.defineProperty(exports, "indicatorBehavior", {
  enumerable: true,
  get: function get() {
    return _indicatorBehavior.default;
  }
});
Object.defineProperty(exports, "tabBehavior", {
  enumerable: true,
  get: function get() {
    return _tabBehavior.default;
  }
});
Object.defineProperty(exports, "tabListBehavior", {
  enumerable: true,
  get: function get() {
    return _tabListBehavior.default;
  }
});
Object.defineProperty(exports, "menuAsToolbarBehavior", {
  enumerable: true,
  get: function get() {
    return _menuAsToolbarBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarMenuBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarMenuItemBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuItemBehavior.default;
  }
});
Object.defineProperty(exports, "menuItemAsToolbarButtonBehavior", {
  enumerable: true,
  get: function get() {
    return _menuItemAsToolbarButtonBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarItemBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarItemBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarMenuItemCheckboxBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuItemCheckboxBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarMenuItemRadioBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuItemRadioBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarMenuRadioGroupBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarMenuRadioGroupBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarRadioGroupBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarRadioGroupBehavior.default;
  }
});
Object.defineProperty(exports, "toolbarRadioGroupItemBehavior", {
  enumerable: true,
  get: function get() {
    return _toolbarRadioGroupItemBehavior.default;
  }
});
Object.defineProperty(exports, "radioGroupBehavior", {
  enumerable: true,
  get: function get() {
    return _radioGroupBehavior.default;
  }
});
Object.defineProperty(exports, "radioGroupItemBehavior", {
  enumerable: true,
  get: function get() {
    return _radioGroupItemBehavior.default;
  }
});
Object.defineProperty(exports, "popupBehavior", {
  enumerable: true,
  get: function get() {
    return _popupBehavior.default;
  }
});
Object.defineProperty(exports, "chatBehavior", {
  enumerable: true,
  get: function get() {
    return _chatBehavior.default;
  }
});
Object.defineProperty(exports, "chatMessageBehavior", {
  enumerable: true,
  get: function get() {
    return _chatMessageBehavior.default;
  }
});
Object.defineProperty(exports, "gridBehavior", {
  enumerable: true,
  get: function get() {
    return _gridBehavior.default;
  }
});
Object.defineProperty(exports, "gridHorizontalBehavior", {
  enumerable: true,
  get: function get() {
    return _gridHorizontalBehavior.default;
  }
});
Object.defineProperty(exports, "hierarchicalTreeBehavior", {
  enumerable: true,
  get: function get() {
    return _hierarchicalTreeBehavior.default;
  }
});
Object.defineProperty(exports, "hierarchicalTreeItemBehavior", {
  enumerable: true,
  get: function get() {
    return _hierarchicalTreeItemBehavior.default;
  }
});
Object.defineProperty(exports, "hierarchicalTreeTitleBehavior", {
  enumerable: true,
  get: function get() {
    return _hierarchicalTreeTitleBehavior.default;
  }
});
Object.defineProperty(exports, "hierarchicalSubtreeBehavior", {
  enumerable: true,
  get: function get() {
    return _hierarchicalSubtreeBehavior.default;
  }
});
Object.defineProperty(exports, "dialogBehavior", {
  enumerable: true,
  get: function get() {
    return _dialogBehavior.default;
  }
});
Object.defineProperty(exports, "statusBehavior", {
  enumerable: true,
  get: function get() {
    return _statusBehavior.default;
  }
});
Object.defineProperty(exports, "embedBehavior", {
  enumerable: true,
  get: function get() {
    return _embedBehavior.default;
  }
});
Object.defineProperty(exports, "accordionBehavior", {
  enumerable: true,
  get: function get() {
    return _accordionBehavior.default;
  }
});
Object.defineProperty(exports, "accordionTitleBehavior", {
  enumerable: true,
  get: function get() {
    return _accordionTitleBehavior.default;
  }
});
Object.defineProperty(exports, "accordionContentBehavior", {
  enumerable: true,
  get: function get() {
    return _accordionContentBehavior.default;
  }
});
Object.defineProperty(exports, "checkboxBehavior", {
  enumerable: true,
  get: function get() {
    return _checkboxBehavior.default;
  }
});
Object.defineProperty(exports, "tooltipAsDescriptionBehavior", {
  enumerable: true,
  get: function get() {
    return _tooltipAsDescriptionBehavior.default;
  }
});
Object.defineProperty(exports, "tooltipAsLabelBehavior", {
  enumerable: true,
  get: function get() {
    return _tooltipAsLabelBehavior.default;
  }
});
Object.defineProperty(exports, "sliderBehavior", {
  enumerable: true,
  get: function get() {
    return _sliderBehavior.default;
  }
});
Object.defineProperty(exports, "menuButtonBehavior", {
  enumerable: true,
  get: function get() {
    return _menuButtonBehavior.default;
  }
});
Object.defineProperty(exports, "splitButtonBehavior", {
  enumerable: true,
  get: function get() {
    return _splitButtonBehavior.default;
  }
});
Object.defineProperty(exports, "treeBehavior", {
  enumerable: true,
  get: function get() {
    return _treeBehavior.default;
  }
});
Object.defineProperty(exports, "treeItemBehavior", {
  enumerable: true,
  get: function get() {
    return _treeItemBehavior.default;
  }
});
Object.defineProperty(exports, "treeTitleBehavior", {
  enumerable: true,
  get: function get() {
    return _treeTitleBehavior.default;
  }
});
Object.defineProperty(exports, "textAreaBehavior", {
  enumerable: true,
  get: function get() {
    return _textAreaBehavior.default;
  }
});
Object.defineProperty(exports, "treeAsListBehavior", {
  enumerable: true,
  get: function get() {
    return _treeAsListBehavior.default;
  }
});
Object.defineProperty(exports, "treeItemAsListItemBehavior", {
  enumerable: true,
  get: function get() {
    return _treeItemAsListItemBehavior.default;
  }
});
Object.defineProperty(exports, "treeTitleAsListItemTitleBehavior", {
  enumerable: true,
  get: function get() {
    return _treeTitleAsListItemTitleBehavior.default;
  }
});
Object.defineProperty(exports, "carouselBehavior", {
  enumerable: true,
  get: function get() {
    return _carouselBehavior.default;
  }
});
Object.defineProperty(exports, "carouselItemBehavior", {
  enumerable: true,
  get: function get() {
    return _carouselItemBehavior.default;
  }
});
Object.defineProperty(exports, "tableBehavior", {
  enumerable: true,
  get: function get() {
    return _tableBehavior.default;
  }
});
Object.defineProperty(exports, "tableRowBehavior", {
  enumerable: true,
  get: function get() {
    return _tableRowBehavior.default;
  }
});
Object.defineProperty(exports, "tableCellBehavior", {
  enumerable: true,
  get: function get() {
    return _tableCellBehavior.default;
  }
});
Object.defineProperty(exports, "tableHeaderCellBehavior", {
  enumerable: true,
  get: function get() {
    return _tableHeaderCellBehavior.default;
  }
});
Object.defineProperty(exports, "gridNestedBehavior", {
  enumerable: true,
  get: function get() {
    return _gridNestedBehavior.default;
  }
});
Object.defineProperty(exports, "gridHeaderRowBehavior", {
  enumerable: true,
  get: function get() {
    return _gridHeaderRowBehavior.default;
  }
});
Object.defineProperty(exports, "gridHeaderCellBehavior", {
  enumerable: true,
  get: function get() {
    return _gridHeaderCellBehavior.default;
  }
});
Object.defineProperty(exports, "gridRowBehavior", {
  enumerable: true,
  get: function get() {
    return _gridRowBehavior.default;
  }
});
Object.defineProperty(exports, "gridRowNestedBehavior", {
  enumerable: true,
  get: function get() {
    return _gridRowNestedBehavior.default;
  }
});
Object.defineProperty(exports, "gridCellBehavior", {
  enumerable: true,
  get: function get() {
    return _gridCellBehavior.default;
  }
});
Object.defineProperty(exports, "gridCellMultipleFocusableBehavior", {
  enumerable: true,
  get: function get() {
    return _gridCellMultipleFocusableBehavior.default;
  }
});
Object.defineProperty(exports, "gridCellWithFocusableElementBehavior", {
  enumerable: true,
  get: function get() {
    return _gridCellWithFocusableElementBehavior.default;
  }
});
Object.defineProperty(exports, "cardBehavior", {
  enumerable: true,
  get: function get() {
    return _cardBehavior.default;
  }
});
Object.defineProperty(exports, "cardFocusableBehavior", {
  enumerable: true,
  get: function get() {
    return _cardFocusableBehavior.default;
  }
});
Object.defineProperty(exports, "cardChildrenFocusableBehavior", {
  enumerable: true,
  get: function get() {
    return _cardChildrenFocusableBehavior.default;
  }
});
Object.defineProperty(exports, "cardsContainerBehavior", {
  enumerable: true,
  get: function get() {
    return _cardsContainerBehavior.default;
  }
});
Object.defineProperty(exports, "videoBehavior", {
  enumerable: true,
  get: function get() {
    return _videoBehavior.default;
  }
});

var _alertBehavior = _interopRequireWildcard(require("./Alert/alertBehavior"));

Object.keys(_alertBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _alertBehavior[key];
    }
  });
});

var _alertWarningBehavior = _interopRequireDefault(require("./Alert/alertWarningBehavior"));

var _alertBaseBehavior = _interopRequireDefault(require("./Alert/alertBaseBehavior"));

var _attachmentBehavior = _interopRequireWildcard(require("./Attachment/attachmentBehavior"));

Object.keys(_attachmentBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _attachmentBehavior[key];
    }
  });
});

var _buttonBehavior = _interopRequireWildcard(require("./Button/buttonBehavior"));

Object.keys(_buttonBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _buttonBehavior[key];
    }
  });
});

var _buttonGroupBehavior = _interopRequireWildcard(require("./Button/buttonGroupBehavior"));

Object.keys(_buttonGroupBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _buttonGroupBehavior[key];
    }
  });
});

var _toggleButtonBehavior = _interopRequireDefault(require("./Button/toggleButtonBehavior"));

var _imageBehavior = _interopRequireWildcard(require("./Image/imageBehavior"));

Object.keys(_imageBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _imageBehavior[key];
    }
  });
});

var _menuBehavior = _interopRequireDefault(require("./Menu/menuBehavior"));

var _menuItemBehavior = _interopRequireDefault(require("./Menu/menuItemBehavior"));

var _menuDividerBehavior = _interopRequireDefault(require("./Menu/menuDividerBehavior"));

var _submenuBehavior = _interopRequireDefault(require("./Menu/submenuBehavior"));

var _listBehavior = _interopRequireWildcard(require("./List/listBehavior"));

Object.keys(_listBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _listBehavior[key];
    }
  });
});

var _basicListItemBehavior = _interopRequireDefault(require("./List/basicListItemBehavior"));

var _listItemBehavior = _interopRequireWildcard(require("./List/listItemBehavior"));

Object.keys(_listItemBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _listItemBehavior[key];
    }
  });
});

var _navigableListBehavior = _interopRequireDefault(require("./List/navigableListBehavior"));

var _navigableListItemBehavior = _interopRequireDefault(require("./List/navigableListItemBehavior"));

var _selectableListBehavior = _interopRequireDefault(require("./List/selectableListBehavior"));

var _selectableListItemBehavior = _interopRequireDefault(require("./List/selectableListItemBehavior"));

var _loaderBehavior = _interopRequireDefault(require("./Loader/loaderBehavior"));

var _inputBehavior = _interopRequireDefault(require("./Input/inputBehavior"));

var _iconBehavior = _interopRequireWildcard(require("./Icon/iconBehavior"));

Object.keys(_iconBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _iconBehavior[key];
    }
  });
});

var _indicatorBehavior = _interopRequireWildcard(require("./Box/indicatorBehavior"));

Object.keys(_indicatorBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _indicatorBehavior[key];
    }
  });
});

var _tabBehavior = _interopRequireDefault(require("./Tab/tabBehavior"));

var _tabListBehavior = _interopRequireDefault(require("./Tab/tabListBehavior"));

var _menuAsToolbarBehavior = _interopRequireDefault(require("./Toolbar/menuAsToolbarBehavior"));

var _toolbarMenuBehavior = _interopRequireWildcard(require("./Toolbar/toolbarMenuBehavior"));

Object.keys(_toolbarMenuBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarMenuBehavior[key];
    }
  });
});

var _toolbarMenuItemBehavior = _interopRequireWildcard(require("./Toolbar/toolbarMenuItemBehavior"));

Object.keys(_toolbarMenuItemBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarMenuItemBehavior[key];
    }
  });
});

var _menuItemAsToolbarButtonBehavior = _interopRequireDefault(require("./Toolbar/menuItemAsToolbarButtonBehavior"));

var _toolbarBehavior = _interopRequireWildcard(require("./Toolbar/toolbarBehavior"));

Object.keys(_toolbarBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarBehavior[key];
    }
  });
});

var _toolbarItemBehavior = _interopRequireWildcard(require("./Toolbar/toolbarItemBehavior"));

Object.keys(_toolbarItemBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarItemBehavior[key];
    }
  });
});

var _toolbarMenuItemCheckboxBehavior = _interopRequireDefault(require("./Toolbar/toolbarMenuItemCheckboxBehavior"));

var _toolbarMenuItemRadioBehavior = _interopRequireDefault(require("./Toolbar/toolbarMenuItemRadioBehavior"));

var _toolbarMenuRadioGroupBehavior = _interopRequireWildcard(require("./Toolbar/toolbarMenuRadioGroupBehavior"));

Object.keys(_toolbarMenuRadioGroupBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarMenuRadioGroupBehavior[key];
    }
  });
});

var _toolbarRadioGroupBehavior = _interopRequireWildcard(require("./Toolbar/toolbarRadioGroupBehavior"));

Object.keys(_toolbarRadioGroupBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _toolbarRadioGroupBehavior[key];
    }
  });
});

var _toolbarRadioGroupItemBehavior = _interopRequireDefault(require("./Toolbar/toolbarRadioGroupItemBehavior"));

var _radioGroupBehavior = _interopRequireDefault(require("./Radio/radioGroupBehavior"));

var _radioGroupItemBehavior = _interopRequireDefault(require("./Radio/radioGroupItemBehavior"));

var _popupBehavior = _interopRequireWildcard(require("./Popup/popupBehavior"));

Object.keys(_popupBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _popupBehavior[key];
    }
  });
});

var _chatBehavior = _interopRequireWildcard(require("./Chat/chatBehavior"));

Object.keys(_chatBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _chatBehavior[key];
    }
  });
});

var _chatMessageBehavior = _interopRequireWildcard(require("./Chat/chatMessageBehavior"));

Object.keys(_chatMessageBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _chatMessageBehavior[key];
    }
  });
});

var _gridBehavior = _interopRequireDefault(require("./Grid/gridBehavior"));

var _gridHorizontalBehavior = _interopRequireDefault(require("./Grid/gridHorizontalBehavior"));

var _hierarchicalTreeBehavior = _interopRequireDefault(require("./HierarchicalTree/hierarchicalTreeBehavior"));

var _hierarchicalTreeItemBehavior = _interopRequireDefault(require("./HierarchicalTree/hierarchicalTreeItemBehavior"));

var _hierarchicalTreeTitleBehavior = _interopRequireDefault(require("./HierarchicalTree/hierarchicalTreeTitleBehavior"));

var _hierarchicalSubtreeBehavior = _interopRequireDefault(require("./HierarchicalTree/hierarchicalSubtreeBehavior"));

var _dialogBehavior = _interopRequireDefault(require("./Dialog/dialogBehavior"));

var _statusBehavior = _interopRequireDefault(require("./Status/statusBehavior"));

var _embedBehavior = _interopRequireWildcard(require("./Embed/embedBehavior"));

Object.keys(_embedBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _embedBehavior[key];
    }
  });
});

var _accordionBehavior = _interopRequireWildcard(require("./Accordion/accordionBehavior"));

Object.keys(_accordionBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _accordionBehavior[key];
    }
  });
});

var _accordionTitleBehavior = _interopRequireWildcard(require("./Accordion/accordionTitleBehavior"));

Object.keys(_accordionTitleBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _accordionTitleBehavior[key];
    }
  });
});

var _accordionContentBehavior = _interopRequireWildcard(require("./Accordion/accordionContentBehavior"));

Object.keys(_accordionContentBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _accordionContentBehavior[key];
    }
  });
});

var _checkboxBehavior = _interopRequireWildcard(require("./Checkbox/checkboxBehavior"));

Object.keys(_checkboxBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _checkboxBehavior[key];
    }
  });
});

var _tooltipAsDescriptionBehavior = _interopRequireWildcard(require("./Tooltip/tooltipAsDescriptionBehavior"));

Object.keys(_tooltipAsDescriptionBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tooltipAsDescriptionBehavior[key];
    }
  });
});

var _tooltipAsLabelBehavior = _interopRequireDefault(require("./Tooltip/tooltipAsLabelBehavior"));

var _sliderBehavior = _interopRequireWildcard(require("./Slider/sliderBehavior"));

Object.keys(_sliderBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _sliderBehavior[key];
    }
  });
});

var _menuButtonBehavior = _interopRequireDefault(require("./MenuButton/menuButtonBehavior"));

var _splitButtonBehavior = _interopRequireWildcard(require("./SplitButton/splitButtonBehavior"));

Object.keys(_splitButtonBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _splitButtonBehavior[key];
    }
  });
});

var _treeBehavior = _interopRequireDefault(require("./Tree/treeBehavior"));

var _treeItemBehavior = _interopRequireWildcard(require("./Tree/treeItemBehavior"));

Object.keys(_treeItemBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _treeItemBehavior[key];
    }
  });
});

var _treeTitleBehavior = _interopRequireWildcard(require("./Tree/treeTitleBehavior"));

Object.keys(_treeTitleBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _treeTitleBehavior[key];
    }
  });
});

var _textAreaBehavior = _interopRequireWildcard(require("./TextArea/textAreaBehavior"));

Object.keys(_textAreaBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _textAreaBehavior[key];
    }
  });
});

var _treeAsListBehavior = _interopRequireDefault(require("./Tree/treeAsListBehavior"));

var _treeItemAsListItemBehavior = _interopRequireDefault(require("./Tree/treeItemAsListItemBehavior"));

var _treeTitleAsListItemTitleBehavior = _interopRequireDefault(require("./Tree/treeTitleAsListItemTitleBehavior"));

var _carouselBehavior = _interopRequireWildcard(require("./Carousel/carouselBehavior"));

Object.keys(_carouselBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _carouselBehavior[key];
    }
  });
});

var _carouselItemBehavior = _interopRequireWildcard(require("./Carousel/carouselItemBehavior"));

Object.keys(_carouselItemBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _carouselItemBehavior[key];
    }
  });
});

var _tableBehavior = _interopRequireWildcard(require("./Table/tableBehavior"));

Object.keys(_tableBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tableBehavior[key];
    }
  });
});

var _tableRowBehavior = _interopRequireDefault(require("./Table/tableRowBehavior"));

var _tableCellBehavior = _interopRequireWildcard(require("./Table/tableCellBehavior"));

Object.keys(_tableCellBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tableCellBehavior[key];
    }
  });
});

var _tableHeaderCellBehavior = _interopRequireDefault(require("./Table/tableHeaderCellBehavior"));

var _gridNestedBehavior = _interopRequireDefault(require("./Table/gridNestedBehavior"));

var _gridHeaderRowBehavior = _interopRequireDefault(require("./Table/gridHeaderRowBehavior"));

var _gridHeaderCellBehavior = _interopRequireDefault(require("./Table/gridHeaderCellBehavior"));

var _gridRowBehavior = _interopRequireWildcard(require("./Table/gridRowBehavior"));

Object.keys(_gridRowBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _gridRowBehavior[key];
    }
  });
});

var _gridRowNestedBehavior = _interopRequireDefault(require("./Table/gridRowNestedBehavior"));

var _gridCellBehavior = _interopRequireDefault(require("./Table/gridCellBehavior"));

var _gridCellMultipleFocusableBehavior = _interopRequireDefault(require("./Table/gridCellMultipleFocusableBehavior"));

var _gridCellWithFocusableElementBehavior = _interopRequireDefault(require("./Table/gridCellWithFocusableElementBehavior"));

var _cardBehavior = _interopRequireWildcard(require("./Card/cardBehavior"));

Object.keys(_cardBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _cardBehavior[key];
    }
  });
});

var _cardFocusableBehavior = _interopRequireDefault(require("./Card/cardFocusableBehavior"));

var _cardChildrenFocusableBehavior = _interopRequireDefault(require("./Card/cardChildrenFocusableBehavior"));

var _cardsContainerBehavior = _interopRequireDefault(require("./Card/cardsContainerBehavior"));

var _videoBehavior = _interopRequireWildcard(require("./Video/videoBehavior"));

Object.keys(_videoBehavior).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _videoBehavior[key];
    }
  });
});
