// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Status/statusVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _default = function _default(siteVariables) {
  return {
    borderColor: undefined,
    borderWidth: 2,
    iconSize: (0, _utils.pxToRem)(7),
    successBackgroundColor: siteVariables.colorScheme.green.background,
    successTextColor: siteVariables.colorScheme.green.foreground1,
    infoBackgroundColor: siteVariables.colorScheme.brand.background,
    infoTextColor: siteVariables.colorScheme.default.foreground2,
    warningBackgroundColor: siteVariables.colorScheme.yellow.background,
    warningTextColor: siteVariables.colorScheme.yellow.foreground2,
    errorBackgroundColor: siteVariables.colorScheme.red.background,
    errorTextColor: siteVariables.colorScheme.red.foreground2,
    defaultBackgroundColor: siteVariables.colorScheme.default.background5,
    defaultTextColor: siteVariables.colorScheme.default.foreground4
  };
};

exports.default = _default;
