// @fluentui/react-northstar/dist/commonjs/themes/teams/components/List/listItemVariables.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("../../../../utils");

var _default = function _default(siteVariables) {
  return {
    minHeight: (0, _utils.pxToRem)(48),
    rootPadding: "0 ".concat((0, _utils.pxToRem)(18), " 0 ").concat((0, _utils.pxToRem)(20)),
    zIndex: siteVariables.zIndexes.foreground,
    // Header
    // TODO: prod app uses 17.5px here, it should be 16px per the design guide!
    headerLineHeight: siteVariables.lineHeightSmall,
    headerFontSize: siteVariables.fontSizes.medium,
    // Header Media
    headerMediaFontSize: siteVariables.fontSizes.small,
    // TODO: prod app uses 17.5px here, it should be 16px per the design guide!
    headerMediaLineHeight: siteVariables.lineHeightSmall,
    // Content
    contentFontSize: siteVariables.fontSizes.small,
    contentLineHeight: siteVariables.lineHeightSmall,
    // Content Media
    contentMediaFontSize: siteVariables.fontSizes.small,
    contentMediaLineHeight: siteVariables.lineHeightSmall,
    // Selectable
    selectableFocusHoverColor: siteVariables.colors.white,
    selectableFocusHoverBackgroundColor: siteVariables.colors.brand[500],
    selectedColor: siteVariables.bodyColor,
    selectedBackgroundColor: siteVariables.colors.grey[100]
  };
};

exports.default = _default;
