// @fluentui/accessibility/dist/dts/src/behaviors/Toolbar/toolbarMenuItemCheckboxBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var keyboardKey = require("keyboard-key");
/**
 * @specification
 * Adds attribute 'aria-checked=true' based on the property 'active'.
 * Adds attribute 'aria-disabled=true' based on the property 'disabled'.
 * Adds role='menuitemcheckbox'.
 * Adds role 'presentation' to 'wrapper' slot.
 * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
 */
var toolbarMenuItemCheckboxBehavior = function (props) { return ({
    attributes: {
        wrapper: {
            role: 'presentation',
        },
        root: {
            'aria-checked': props.active,
            'aria-disabled': props.disabled,
            role: 'menuitemcheckbox',
        },
    },
    keyActions: {
        root: {
            performClick: {
                keyCombinations: [{ keyCode: keyboardKey.Enter }, { keyCode: keyboardKey.Spacebar }],
            },
        },
    },
}); };
exports.default = toolbarMenuItemCheckboxBehavior;
