// @fluentui/react-northstar/dist/commonjs/themes/teams/components/Alert/alertDismissActionStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _reactBindings = require("@fluentui/react-bindings");

var _AlertDismissAction = require("../../../../components/Alert/AlertDismissAction");

var _getBorderFocusStyles = _interopRequireDefault(require("../../getBorderFocusStyles"));

var _getIconFillOrOutlineStyles = _interopRequireDefault(require("../../getIconFillOrOutlineStyles"));

var _dismissIndicatorUrl = _interopRequireDefault(require("./dismissIndicatorUrl"));

var _alertStyles = require("./alertStyles");

var getIndicatorStyles = function getIndicatorStyles(color, outline, size) {
  return {
    width: size,
    height: size,
    backgroundImage: (0, _dismissIndicatorUrl.default)(color, outline),
    backgroundRepeat: 'no-repeat'
  };
};

var alertDismissActionStyles = {
  root: function root(_ref) {
    var p = _ref.props,
        v = _ref.variables,
        theme = _ref.theme;
    var siteVariables = theme.siteVariables;
    var borderWidth = siteVariables.borderWidth;

    var _getIntentColorsFromP = (0, _alertStyles.getIntentColorsFromProps)(p, v, siteVariables),
        dismissActionIndicatorColor = _getIntentColorsFromP.color;

    var borderFocusStyles = (0, _getBorderFocusStyles.default)({
      variables: {
        borderRadius: v.focusBorderRadius,
        borderWidth: v.focusBorderWidth,
        focusInnerBorderColor: v.focusInnerBorderColor,
        focusOuterBorderColor: v.focusOuterBorderColor,
        zIndexes: {
          foreground: v.focusBorderZIndex
        }
      },
      borderPadding: borderWidth
    });
    return Object.assign({
      height: v.dismissActionSize,
      minWidth: v.dismissActionSize,
      color: v.dismissActionColor || 'currentColor',
      outline: 0,
      padding: 0,
      border: 0,
      backgroundColor: v.dismissActionBackgroundColor,
      borderRadius: v.borderRadius,
      display: 'inline-flex',
      justifyContent: 'center',
      alignItems: 'center',
      position: 'relative',
      verticalAlign: 'middle',
      cursor: 'pointer'
    }, (0, _getIconFillOrOutlineStyles.default)({
      outline: true
    }), {}, !p.hasContent && (0, _defineProperty2.default)({}, "& .".concat(_AlertDismissAction.alertDismissActionSlotClassNames.content), Object.assign({}, getIndicatorStyles(dismissActionIndicatorColor, true, v.dismissActionIndicatorSize))), {
      ':hover': Object.assign({
        backgroundColor: v.hoverBackgroundColor,
        color: 'currentColor'
      }, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      }), {}, !p.hasContent && (0, _defineProperty2.default)({}, "& .".concat(_AlertDismissAction.alertDismissActionSlotClassNames.content), Object.assign({}, getIndicatorStyles(dismissActionIndicatorColor, false, v.dismissActionIndicatorSize))), {}, (0, _getBorderFocusStyles.default)({
        variables: {
          borderRadius: v.dismissActionHoverBorderRadius,
          borderWidth: v.dismissActionHoverBorderWidth,
          focusInnerBorderColor: v.dismissActionHoverInnerBorderColor,
          focusOuterBorderColor: v.dismissActionHoverOuterBorderColor,
          zIndexes: {
            foreground: v.dismissActionHoverZIndex
          }
        }
      })[':focus-visible']),
      ':active': Object.assign({}, (0, _reactBindings.unstable_createAnimationStyles)('scaleDownSoft', theme), {
        color: v.dismissActionColorActive,
        backgroundColor: v.dismissActionBackgroundColorActive,
        borderColor: v.dismissActionBorderColorActive,
        boxShadow: 'none'
      }),
      ':focus': borderFocusStyles[':focus'],
      ':focus-visible': Object.assign({
        backgroundColor: v.focusBackgroundColor,
        borderColor: v.dismissActionBorderColorFocus,
        color: v.dismissActionColorFocus,
        borderWidth: borderWidth
      }, borderFocusStyles[':focus-visible'], {}, (0, _getIconFillOrOutlineStyles.default)({
        outline: false
      }), {}, !p.hasContent && (0, _defineProperty2.default)({}, "& .".concat(_AlertDismissAction.alertDismissActionSlotClassNames.content), Object.assign({}, getIndicatorStyles(dismissActionIndicatorColor, false, v.dismissActionIndicatorSize))), {
        ':hover': {
          borderColor: v.dismissActionBorderColorHover
        }
      })
    }, p.disabled && {
      cursor: 'default',
      color: v.dismissActionColorDisabled,
      boxShadow: 'none',
      pointerEvents: 'none',
      ':hover': {
        color: v.dismissActionColorDisabled
      },
      backgroundColor: v.dismissActionBackgroundColorDisabled,
      borderColor: v.dismissActionBorderColorDisabled
    });
  },
  content: function content(_ref5) {
    var p = _ref5.props,
        v = _ref5.variables,
        theme = _ref5.theme;
    var siteVariables = theme.siteVariables;

    var _getIntentColorsFromP2 = (0, _alertStyles.getIntentColorsFromProps)(p, v, siteVariables),
        dismissActionIndicatorColor = _getIntentColorsFromP2.color;

    return Object.assign({
      fontWeight: v.dismissActionContentFontWeight
    }, !p.hasContent && Object.assign({}, getIndicatorStyles(dismissActionIndicatorColor, false, v.dismissActionIndicatorSize)));
  }
};
var _default = alertDismissActionStyles;
exports.default = _default;
