// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Menu/menuDividerStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var colors_1 = require("../../colors");
var menuDividerStyles = {
    root: function (_a) {
        var _b;
        var p = _a.props, v = _a.variables;
        var colors = colors_1.getColorScheme(v.colorScheme, null, p.primary);
        var borderColor = p.primary ? v.primaryBorderColor : v.borderColor || colors.border;
        var borderType = p.vertical ? 'borderTop' : 'borderLeft';
        return p.content
            ? {
                display: 'flex',
                justifyContent: 'center',
                flexDirection: 'column',
                textAlign: 'center',
            }
            : tslib_1.__assign(tslib_1.__assign((_b = {}, _b[borderType] = "1px solid " + borderColor, _b), (!p.vertical && {
                alignSelf: 'stretch',
            })), (p.vertical &&
                p.inSubmenu && {
                margin: '8px 0',
            }));
    },
};
exports.default = menuDividerStyles;
