// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Menu/menuStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
var menuItemStyles_1 = require("./menuItemStyles");
var colors_1 = require("../../colors");
exports.default = {
    root: function (_a) {
        var p = _a.props, v = _a.variables;
        var iconOnly = p.iconOnly, fluid = p.fluid, pointing = p.pointing, pills = p.pills, primary = p.primary, underlined = p.underlined, vertical = p.vertical, submenu = p.submenu;
        var colors = colors_1.getColorScheme(v.colorScheme, null, primary);
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ display: 'flex', minHeight: utils_1.pxToRem(24), margin: 0, padding: 0, color: v.color, backgroundColor: v.backgroundColor || 'inherit', listStyleType: 'none' }, (iconOnly && { alignItems: 'center' })), (vertical && tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ flexDirection: 'column', backgroundColor: v.verticalBackgroundColor, padding: utils_1.pxToRem(8) + " 0" }, (submenu && {
            boxShadow: v.verticalBoxShadow,
        })), (!fluid && !submenu && { width: 'fit-content' })), (iconOnly && {
            display: 'inline-block',
            width: 'auto',
        })))), (!pills &&
            !iconOnly &&
            !(pointing && vertical) &&
            !underlined && {
            // primary has hardcoded grey border color
            border: v.borderWidth + " solid " + (primary ? v.primaryBorderColor : v.borderColor || colors.border),
            borderRadius: utils_1.pxToRem(4),
        })), (underlined && {
            borderBottom: v.underlinedBottomBorderWidth + " solid " + v.underlinedBorderColor,
        }));
    },
    divider: function (_a) {
        var _b = _a.props, pointing = _b.pointing, vertical = _b.vertical, pills = _b.pills;
        return (tslib_1.__assign(tslib_1.__assign({}, (pointing &&
            vertical && {
            marginBottom: menuItemStyles_1.verticalPointingBottomMargin,
        })), (pills && tslib_1.__assign({}, (vertical
            ? { margin: "0 0 " + menuItemStyles_1.verticalPillsBottomMargin + " 0" }
            : { margin: "0 " + menuItemStyles_1.horizontalPillsRightMargin + " 0 0" })))));
    },
};
