// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Form/formFieldStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
var formFieldStyles = {
    root: function (_a) {
        var props = _a.props, variables = _a.variables;
        return ({});
    },
    label: function (_a) {
        var props = _a.props;
        var type = props.type, inline = props.inline, required = props.required;
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, ((!type || (type !== 'radio' && type !== 'checkbox')) && {
            display: 'block',
        })), (inline && { marginRight: utils_1.pxToRem(10), display: 'inline' })), (required && {
            '::after': {
                content: '"*"',
            },
        }));
    },
    control: function (_a) {
        var props = _a.props;
        var type = props.type;
        return tslib_1.__assign({}, (type &&
            (type === 'radio' || type === 'checkbox') && {
            marginRight: utils_1.pxToRem(10),
        }));
    },
    message: function () { return ({
        display: 'block',
    }); },
};
exports.default = formFieldStyles;
