// @fluentui/react-northstar/dist/dts/test/specs/utils/felaDisableAnimationsPlugin-test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var felaDisableAnimationsPlugin_1 = require("src/utils/felaDisableAnimationsPlugin");
var disableAnimationsPlugin = felaDisableAnimationsPlugin_1.default();
var stylesWithAnimationShorthand = {
    animation: 'k1',
    margin: '0px 10px',
};
var stylesWithAnimationProps = {
    animationName: 'k1',
    animationDuration: '1s',
    margin: '0px 10px',
};
describe('felaDisableAnimationsPlugin', function () {
    test('does not disable animations if the props are not provided', function () {
        expect(disableAnimationsPlugin(stylesWithAnimationShorthand, 'RULE')).toMatchObject(stylesWithAnimationShorthand);
    });
    test('does not disable animations if the disableAnimations flag is undefined', function () {
        expect(disableAnimationsPlugin(stylesWithAnimationShorthand, 'RULE', undefined, {
            disableAnimations: undefined,
        })).toMatchObject(stylesWithAnimationShorthand);
    });
    test('does not disable animations if the disableAnimations flag is false', function () {
        expect(disableAnimationsPlugin(stylesWithAnimationProps, 'RULE', undefined, {
            disableAnimations: false,
        })).toMatchObject(stylesWithAnimationProps);
    });
    test('disables animations if the disableAnimations flag is true', function () {
        expect(disableAnimationsPlugin(stylesWithAnimationProps, 'RULE', undefined, {
            disableAnimations: true,
        })).toMatchObject({ margin: '0px 10px' });
    });
    test('disables animations if the disableAnimations flag is true and the animation css shorthand is used', function () {
        expect(disableAnimationsPlugin(stylesWithAnimationShorthand, 'RULE', undefined, {
            disableAnimations: true,
        })).toMatchObject({ margin: '0px 10px' });
    });
});
