// @fluentui/react-bindings/dist/dts/src/hooks/useStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var styles_1 = require("@fluentui/styles");
var React = require("react");
// @ts-ignore We have this export in package, but it is not present in typings
var react_fela_1 = require("react-fela");
var getStyles_1 = require("../styles/getStyles");
var defaultContext = {
    rtl: false,
    disableAnimations: false,
    performance: {
        enableSanitizeCssPlugin: process.env.NODE_ENV !== 'production',
        enableStylesCaching: true,
        enableVariablesCaching: true,
        enableBooleanVariablesCaching: false,
    },
    renderer: { renderRule: function () { return ''; } },
    theme: styles_1.emptyTheme,
};
var useStyles = function (displayName, options) {
    var _a, _b, _c, _d;
    var context = React.useContext(react_fela_1.ThemeContext) || defaultContext;
    var _e = options.className, className = _e === void 0 ? process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉' : _e, composeOptions = options.composeOptions, _f = options.mapPropsToStyles, mapPropsToStyles = _f === void 0 ? function () { return ({}); } : _f, _g = options.mapPropsToInlineStyles, mapPropsToInlineStyles = _g === void 0 ? function () { return ({}); } : _g, _h = options.unstable_props, unstable_props = _h === void 0 ? {} : _h, _j = options.rtl, rtl = _j === void 0 ? false : _j;
    var componentStylesProps = mapPropsToStyles();
    // `composeProps` should include all props including stylesProps as they can contain state
    var composeProps = tslib_1.__assign(tslib_1.__assign({}, unstable_props), componentStylesProps);
    var composeStylesProps = (_b = (_a = composeOptions) === null || _a === void 0 ? void 0 : _a.mapPropsToStylesPropsChain) === null || _b === void 0 ? void 0 : _b.reduce(function (acc, fn) { return (tslib_1.__assign(tslib_1.__assign({}, acc), fn(composeProps))); }, {});
    // Stores debug information for component.
    var debug = React.useRef({ fluentUIDebug: null });
    var _k = getStyles_1.default({
        // Input values
        className: ((_c = composeOptions) === null || _c === void 0 ? void 0 : _c.className) || className,
        displayNames: ((_d = composeOptions) === null || _d === void 0 ? void 0 : _d.displayNames) || [displayName],
        props: tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, componentStylesProps), mapPropsToInlineStyles()), composeStylesProps),
        // Context values
        disableAnimations: context.disableAnimations,
        renderer: context.renderer,
        rtl: rtl,
        saveDebug: function (fluentUIDebug) { return (debug.current = { fluentUIDebug: fluentUIDebug }); },
        theme: context.theme,
        performance: context.performance,
    }), classes = _k.classes, resolvedStyles = _k.styles;
    return { classes: classes, styles: resolvedStyles };
};
exports.default = useStyles;
