// @fluentui/accessibility/dist/es/behaviors/Table/gridRowNestedBehavior.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import { IS_FOCUSABLE_ATTRIBUTE } from '../../attributes';
import { FocusZoneDirection } from '../../focusZone/types';
import * as keyboardKey from 'keyboard-key';
import gridCellBehavior from './gridCellBehavior';

/**
 * @specification
 * Adds role='row'.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 * Adds attribute 'aria-selected=true' based on the property 'selected'. Based on this screen readers will recognize the selected state of the item.
 * Focus can be moved inside a child component with embeded inner FocusZone by pressing a specified key.
 * Provides arrow key navigation in horizontal direction.
 * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
 * Triggers 'unsetRowTabbable' action using 'shiftKey' + 'Tab' key on 'root'.
 * Applies 'gridCellBehavior' for 'cell' child component.
 */
var gridRowNestedBehavior = function gridRowNestedBehavior(props) {
  var _root;

  return {
    attributes: {
      root: (_root = {}, _defineProperty(_root, IS_FOCUSABLE_ATTRIBUTE, true), _defineProperty(_root, "role", 'row'), _defineProperty(_root, 'aria-selected', props.selected), _root)
    },
    focusZone: {
      props: {
        shouldEnterInnerZone: function shouldEnterInnerZone(event) {
          return keyboardKey.getCode(event) === keyboardKey.Enter;
        },
        direction: FocusZoneDirection.horizontal
      }
    },
    keyActions: {
      root: {
        unsetRowTabbable: {
          keyCombinations: [{
            keyCode: keyboardKey.Tab,
            shiftKey: true
          }]
        },
        performClick: {
          keyCombinations: [{
            keyCode: keyboardKey.Enter
          }, {
            keyCode: keyboardKey.Spacebar
          }]
        }
      }
    },
    childBehaviors: {
      cell: gridCellBehavior
    }
  };
};

export default gridRowNestedBehavior;
