// @fluentui/accessibility/dist/es/behaviors/Table/gridCellMultipleFocusableBehavior.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import { IS_FOCUSABLE_ATTRIBUTE } from '../../attributes';
import { FocusZoneDirection } from '../../focusZone/types';
import * as keyboardKey from 'keyboard-key';
/**
 * @specification
 * Adds role='gridcell'.
 * Adds attribute 'data-is-focusable=true' to 'root' slot.
 * Provides arrow key navigation in bidirectional direction.
 * Triggers 'focusCell' action with 'Escape' on 'root'.
 */

var gridCellMultipleFocusableBehavior = function gridCellMultipleFocusableBehavior(props) {
  return {
    attributes: {
      root: _defineProperty({
        role: 'gridcell'
      }, IS_FOCUSABLE_ATTRIBUTE, true)
    },
    focusZone: {
      props: {
        direction: FocusZoneDirection.bidirectional
      }
    },
    keyActions: {
      root: {
        focusCell: {
          keyCombinations: [{
            keyCode: keyboardKey.Escape
          }]
        }
      }
    }
  };
};

export default gridCellMultipleFocusableBehavior;
