// @fluentui/accessibility/dist/dts/src/behaviors/Tree/treeBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var keyboardKey = require("keyboard-key");
var types_1 = require("../../focusZone/types");
var treeItemBehavior_1 = require("./treeItemBehavior");
/**
 * @specification
 * Adds role 'tree' to 'root' slot.
 * Adds attribute 'tabIndex=-1' to 'root' slot.
 * Adds attribute 'aria-labelledby' based on the property 'aria-labelledby' to 'root' slot.
 * Provides arrow key navigation in vertical direction.
 * Triggers 'expandSiblings' action with '*' on 'root'.
 */
var treeBehavior = function (props) {
    return {
        attributes: {
            root: {
                role: 'tree',
                'aria-labelledby': props['aria-labelledby'],
                tabIndex: -1,
            },
        },
        keyActions: {
            root: {
                expandSiblings: {
                    keyCombinations: [{ keyCode: keyboardKey['*'] }],
                },
            },
        },
        focusZone: {
            props: {
                direction: types_1.FocusZoneDirection.vertical,
            },
        },
        childBehaviors: {
            item: treeItemBehavior_1.default,
        },
    };
};
exports.default = treeBehavior;
